// @rc-component/steps/dist/meta__docs.1aaec55d.async.js
!(function(){var Me=Object.defineProperty;var fe=Object.getOwnPropertySymbols;var Oe=Object.prototype.hasOwnProperty,he=Object.prototype.propertyIsEnumerable;var ve=(a,n,e)=>n in a?Me(a,n,{enumerable:!0,configurable:!0,writable:!0,value:e}):a[n]=e,Ie=(a,n)=>{for(var e in n||(n={}))Oe.call(n,e)&&ve(a,e,n[e]);if(fe)for(var e of fe(n))he.call(n,e)&&ve(a,e,n[e]);return a};(self.webpackChunk_rc_component_steps=self.webpackChunk_rc_component_steps||[]).push([[904],{65306:function(a,n){"use strict";var e;e={value:!0},n.Z=void 0;const t={MAC_ENTER:3,BACKSPACE:8,TAB:9,NUM_CENTER:12,ENTER:13,SHIFT:16,CTRL:17,ALT:18,PAUSE:19,CAPS_LOCK:20,ESC:27,SPACE:32,PAGE_UP:33,PAGE_DOWN:34,END:35,HOME:36,LEFT:37,UP:38,RIGHT:39,DOWN:40,PRINT_SCREEN:44,INSERT:45,DELETE:46,ZERO:48,ONE:49,TWO:50,THREE:51,FOUR:52,FIVE:53,SIX:54,SEVEN:55,EIGHT:56,NINE:57,QUESTION_MARK:63,A:65,B:66,C:67,D:68,E:69,F:70,G:71,H:72,I:73,J:74,K:75,L:76,M:77,N:78,O:79,P:80,Q:81,R:82,S:83,T:84,U:85,V:86,W:87,X:88,Y:89,Z:90,META:91,WIN_KEY_RIGHT:92,CONTEXT_MENU:93,NUM_ZERO:96,NUM_ONE:97,NUM_TWO:98,NUM_THREE:99,NUM_FOUR:100,NUM_FIVE:101,NUM_SIX:102,NUM_SEVEN:103,NUM_EIGHT:104,NUM_NINE:105,NUM_MULTIPLY:106,NUM_PLUS:107,NUM_MINUS:109,NUM_PERIOD:110,NUM_DIVISION:111,F1:112,F2:113,F3:114,F4:115,F5:116,F6:117,F7:118,F8:119,F9:120,F10:121,F11:122,F12:123,NUMLOCK:144,SEMICOLON:186,DASH:189,EQUALS:187,COMMA:188,PERIOD:190,SLASH:191,APOSTROPHE:192,SINGLE_QUOTE:222,OPEN_SQUARE_BRACKET:219,BACKSLASH:220,CLOSE_SQUARE_BRACKET:221,WIN_KEY:224,MAC_FF_META:224,WIN_IME:229,isTextModifyingKeyEvent:function(i){const{keyCode:m}=i;if(i.altKey&&!i.ctrlKey||i.metaKey||m>=t.F1&&m<=t.F12)return!1;switch(m){case t.ALT:case t.CAPS_LOCK:case t.CONTEXT_MENU:case t.CTRL:case t.DOWN:case t.END:case t.ESC:case t.HOME:case t.INSERT:case t.LEFT:case t.MAC_FF_META:case t.META:case t.NUMLOCK:case t.NUM_CENTER:case t.PAGE_DOWN:case t.PAGE_UP:case t.PAUSE:case t.PRINT_SCREEN:case t.RIGHT:case t.SHIFT:case t.UP:case t.WIN_KEY:case t.WIN_KEY_RIGHT:return!1;default:return!0}},isCharacterKey:function(i){if(i>=t.ZERO&&i<=t.NINE||i>=t.NUM_ZERO&&i<=t.NUM_MULTIPLY||i>=t.A&&i<=t.Z||window.navigator.userAgent.indexOf("WebKit")!==-1&&i===0)return!0;switch(i){case t.SPACE:case t.QUESTION_MARK:case t.NUM_PLUS:case t.NUM_MINUS:case t.NUM_PERIOD:case t.NUM_DIVISION:case t.SEMICOLON:case t.DASH:case t.EQUALS:case t.COMMA:case t.PERIOD:case t.SLASH:case t.APOSTROPHE:case t.SINGLE_QUOTE:case t.OPEN_SQUARE_BRACKET:case t.BACKSLASH:case t.CLOSE_SQUARE_BRACKET:return!0;default:return!1}}};var r=n.Z=t},3921:function(a,n){"use strict";var e;e={value:!0},n.Z=p;const c=`accept acceptCharset accessKey action allowFullScreen allowTransparency
    alt async autoComplete autoFocus autoPlay capture cellPadding cellSpacing challenge
    charSet checked classID className colSpan cols content contentEditable contextMenu
    controls coords crossOrigin data dateTime default defer dir disabled download draggable
    encType form formAction formEncType formMethod formNoValidate formTarget frameBorder
    headers height hidden high href hrefLang htmlFor httpEquiv icon id inputMode integrity
    is keyParams keyType kind label lang list loop low manifest marginHeight marginWidth max maxLength media
    mediaGroup method min minLength multiple muted name noValidate nonce open
    optimum pattern placeholder poster preload radioGroup readOnly rel required
    reversed role rowSpan rows sandbox scope scoped scrolling seamless selected
    shape size sizes span spellCheck src srcDoc srcLang srcSet start step style
    summary tabIndex target title type useMap value width wmode wrap onCopy onCut onPaste onCompositionEnd onCompositionStart onCompositionUpdate onKeyDown
    onKeyPress onKeyUp onFocus onBlur onChange onInput onSubmit onClick onContextMenu onDoubleClick
    onDrag onDragEnd onDragEnter onDragExit onDragLeave onDragOver onDragStart onDrop onMouseDown
    onMouseEnter onMouseLeave onMouseMove onMouseOut onMouseOver onMouseUp onSelect onTouchCancel
    onTouchEnd onTouchMove onTouchStart onScroll onWheel onAbort onCanPlay onCanPlayThrough
    onDurationChange onEmptied onEncrypted onEnded onError onLoadedData onLoadedMetadata
    onLoadStart onPause onPlay onPlaying onProgress onRateChange onSeeked onSeeking onStalled onSuspend onTimeUpdate onVolumeChange onWaiting onLoad onError`.split(/[\s\n]+/),i="aria-",m="data-";function o(l,d){return l.indexOf(d)===0}function p(l,d=!1){let v;d===!1?v={aria:!0,data:!0,attr:!0}:d===!0?v={aria:!0}:v=Ie({},d);const u={};return Object.keys(l).forEach(I=>{(v.aria&&(I==="role"||o(I,i))||v.data&&o(I,m)||v.attr&&c.includes(I))&&(u[I]=l[I])}),u}},67729:function(a,n,e){"use strict";var t;e.r(n),e.d(n,{demos:function(){return v}});var r=e(15009),c=e.n(r),i=e(99289),m=e.n(i),o=e(67294),p=e(92450),l=e(1653),d=e(69945),v={"docs-demo-alternative-label-demo-alternativelabel":{component:o.memo(o.lazy(function(){return e.e(433).then(e.bind(e,48874))})),asset:{type:"BLOCK",id:"docs-demo-alternative-label-demo-alternativelabel",refAtomIds:[],dependencies:{"index.jsx":{type:"FILE",value:e(66328).Z},react:{type:"NPM",value:"18.3.1"},"@rc-component/steps":{type:"NPM",value:"1.2.0"},"../../assets/iconfont.less":{type:"FILE",value:e(5057).Z},"../../assets/index.less":{type:"FILE",value:e(86834).Z}},entry:"index.jsx"},context:{"../../assets/iconfont.less":l,"../../assets/index.less":d,react:t||(t=e.t(o,2)),"@rc-component/steps":p,"/Users/jilin/projects/antd/rc-steps/assets/iconfont.less":l,"/Users/jilin/projects/antd/rc-steps/assets/index.less":d},renderOpts:{compile:function(){var u=m()(c()().mark(function E(){var _,x=arguments;return c()().wrap(function(s){for(;;)switch(s.prev=s.next){case 0:return s.next=2,e.e(19).then(e.bind(e,4019));case 2:return s.abrupt("return",(_=s.sent).default.apply(_,x));case 3:case"end":return s.stop()}},E)}));function I(){return u.apply(this,arguments)}return I}()}}}},74645:function(a,n,e){"use strict";var t;e.r(n),e.d(n,{demos:function(){return v}});var r=e(15009),c=e.n(r),i=e(99289),m=e.n(i),o=e(67294),p=e(92450),l=e(1653),d=e(69945),v={"docs-demo-composable-demo-composable":{component:o.memo(o.lazy(function(){return e.e(433).then(e.bind(e,52648))})),asset:{type:"BLOCK",id:"docs-demo-composable-demo-composable",refAtomIds:[],dependencies:{"index.jsx":{type:"FILE",value:e(39296).Z},react:{type:"NPM",value:"18.3.1"},"@rc-component/steps":{type:"NPM",value:"1.2.0"},"../../assets/iconfont.less":{type:"FILE",value:e(5057).Z},"../../assets/index.less":{type:"FILE",value:e(86834).Z}},entry:"index.jsx"},context:{"../../assets/iconfont.less":l,"../../assets/index.less":d,react:t||(t=e.t(o,2)),"@rc-component/steps":p,"/Users/jilin/projects/antd/rc-steps/assets/iconfont.less":l,"/Users/jilin/projects/antd/rc-steps/assets/index.less":d},renderOpts:{compile:function(){var u=m()(c()().mark(function E(){var _,x=arguments;return c()().wrap(function(s){for(;;)switch(s.prev=s.next){case 0:return s.next=2,e.e(19).then(e.bind(e,4019));case 2:return s.abrupt("return",(_=s.sent).default.apply(_,x));case 3:case"end":return s.stop()}},E)}));function I(){return u.apply(this,arguments)}return I}()}}}},71005:function(a,n,e){"use strict";var t;e.r(n),e.d(n,{demos:function(){return v}});var r=e(15009),c=e.n(r),i=e(99289),m=e.n(i),o=e(67294),p=e(92450),l=e(69945),d=e(1653),v={"docs-demo-custom-svg-icon-demo-custom-svg-icon":{component:o.memo(o.lazy(function(){return e.e(433).then(e.bind(e,66179))})),asset:{type:"BLOCK",id:"docs-demo-custom-svg-icon-demo-custom-svg-icon",refAtomIds:[],dependencies:{"index.jsx":{type:"FILE",value:e(4927).Z},react:{type:"NPM",value:"18.3.1"},"@rc-component/steps":{type:"NPM",value:"1.2.0"},"../../assets/index.less":{type:"FILE",value:e(86834).Z},"../../assets/iconfont.less":{type:"FILE",value:e(5057).Z}},entry:"index.jsx"},context:{"../../assets/index.less":l,"../../assets/iconfont.less":d,react:t||(t=e.t(o,2)),"@rc-component/steps":p,"/Users/jilin/projects/antd/rc-steps/assets/index.less":l,"/Users/jilin/projects/antd/rc-steps/assets/iconfont.less":d},renderOpts:{compile:function(){var u=m()(c()().mark(function E(){var _,x=arguments;return c()().wrap(function(s){for(;;)switch(s.prev=s.next){case 0:return s.next=2,e.e(19).then(e.bind(e,4019));case 2:return s.abrupt("return",(_=s.sent).default.apply(_,x));case 3:case"end":return s.stop()}},E)}));function I(){return u.apply(this,arguments)}return I}()}}}},66157:function(a,n,e){"use strict";var t;e.r(n),e.d(n,{demos:function(){return v}});var r=e(15009),c=e.n(r),i=e(99289),m=e.n(i),o=e(67294),p=e(92450),l=e(1653),d=e(69945),v={"docs-demo-custom-icon-demo-customicon":{component:o.memo(o.lazy(function(){return e.e(433).then(e.bind(e,5501))})),asset:{type:"BLOCK",id:"docs-demo-custom-icon-demo-customicon",refAtomIds:[],dependencies:{"index.jsx":{type:"FILE",value:e(36393).Z},react:{type:"NPM",value:"18.3.1"},"@rc-component/steps":{type:"NPM",value:"1.2.0"},"../../assets/iconfont.less":{type:"FILE",value:e(5057).Z},"../../assets/index.less":{type:"FILE",value:e(86834).Z}},entry:"index.jsx"},context:{"../../assets/iconfont.less":l,"../../assets/index.less":d,react:t||(t=e.t(o,2)),"@rc-component/steps":p,"/Users/jilin/projects/antd/rc-steps/assets/iconfont.less":l,"/Users/jilin/projects/antd/rc-steps/assets/index.less":d},renderOpts:{compile:function(){var u=m()(c()().mark(function E(){var _,x=arguments;return c()().wrap(function(s){for(;;)switch(s.prev=s.next){case 0:return s.next=2,e.e(19).then(e.bind(e,4019));case 2:return s.abrupt("return",(_=s.sent).default.apply(_,x));case 3:case"end":return s.stop()}},E)}));function I(){return u.apply(this,arguments)}return I}()}}}},29665:function(a,n,e){"use strict";var t;e.r(n),e.d(n,{demos:function(){return v}});var r=e(15009),c=e.n(r),i=e(99289),m=e.n(i),o=e(67294),p=e(92450),l=e(1653),d=e(69945),v={"docs-demo-dynamic-demo-dynamic":{component:o.memo(o.lazy(function(){return e.e(433).then(e.bind(e,63665))})),asset:{type:"BLOCK",id:"docs-demo-dynamic-demo-dynamic",refAtomIds:[],dependencies:{"index.jsx":{type:"FILE",value:e(31626).Z},react:{type:"NPM",value:"18.3.1"},"@rc-component/steps":{type:"NPM",value:"1.2.0"},"../../assets/iconfont.less":{type:"FILE",value:e(5057).Z},"../../assets/index.less":{type:"FILE",value:e(86834).Z}},entry:"index.jsx"},context:{"../../assets/iconfont.less":l,"../../assets/index.less":d,react:t||(t=e.t(o,2)),"@rc-component/steps":p,"/Users/jilin/projects/antd/rc-steps/assets/iconfont.less":l,"/Users/jilin/projects/antd/rc-steps/assets/index.less":d},renderOpts:{compile:function(){var u=m()(c()().mark(function E(){var _,x=arguments;return c()().wrap(function(s){for(;;)switch(s.prev=s.next){case 0:return s.next=2,e.e(19).then(e.bind(e,4019));case 2:return s.abrupt("return",(_=s.sent).default.apply(_,x));case 3:case"end":return s.stop()}},E)}));function I(){return u.apply(this,arguments)}return I}()}}}},48900:function(a,n,e){"use strict";var t;e.r(n),e.d(n,{demos:function(){return v}});var r=e(15009),c=e.n(r),i=e(99289),m=e.n(i),o=e(67294),p=e(92450),l=e(1653),d=e(69945),v={"docs-demo-error-step-demo-errorstep":{component:o.memo(o.lazy(function(){return e.e(433).then(e.bind(e,2297))})),asset:{type:"BLOCK",id:"docs-demo-error-step-demo-errorstep",refAtomIds:[],dependencies:{"index.jsx":{type:"FILE",value:e(75787).Z},react:{type:"NPM",value:"18.3.1"},"@rc-component/steps":{type:"NPM",value:"1.2.0"},"../../assets/iconfont.less":{type:"FILE",value:e(5057).Z},"../../assets/index.less":{type:"FILE",value:e(86834).Z}},entry:"index.jsx"},context:{"../../assets/iconfont.less":l,"../../assets/index.less":d,react:t||(t=e.t(o,2)),"@rc-component/steps":p,"/Users/jilin/projects/antd/rc-steps/assets/iconfont.less":l,"/Users/jilin/projects/antd/rc-steps/assets/index.less":d},renderOpts:{compile:function(){var u=m()(c()().mark(function E(){var _,x=arguments;return c()().wrap(function(s){for(;;)switch(s.prev=s.next){case 0:return s.next=2,e.e(19).then(e.bind(e,4019));case 2:return s.abrupt("return",(_=s.sent).default.apply(_,x));case 3:case"end":return s.stop()}},E)}));function I(){return u.apply(this,arguments)}return I}()}}}},52208:function(a,n,e){"use strict";var t;e.r(n),e.d(n,{demos:function(){return d}});var r=e(15009),c=e.n(r),i=e(99289),m=e.n(i),o=e(67294),p=e(92450),l=e(69945),d={"docs-demo-inline-demo-inline":{component:o.memo(o.lazy(function(){return e.e(433).then(e.bind(e,94947))})),asset:{type:"BLOCK",id:"docs-demo-inline-demo-inline",refAtomIds:[],dependencies:{"index.jsx":{type:"FILE",value:e(32149).Z},react:{type:"NPM",value:"18.3.1"},"@rc-component/steps":{type:"NPM",value:"1.2.0"},"../../assets/index.less":{type:"FILE",value:e(86834).Z}},entry:"index.jsx"},context:{"../../assets/index.less":l,react:t||(t=e.t(o,2)),"@rc-component/steps":p,"/Users/jilin/projects/antd/rc-steps/assets/index.less":l},renderOpts:{compile:function(){var v=m()(c()().mark(function I(){var E,_=arguments;return c()().wrap(function(j){for(;;)switch(j.prev=j.next){case 0:return j.next=2,e.e(19).then(e.bind(e,4019));case 2:return j.abrupt("return",(E=j.sent).default.apply(E,_));case 3:case"end":return j.stop()}},I)}));function u(){return v.apply(this,arguments)}return u}()}}}},85439:function(a,n,e){"use strict";var t;e.r(n),e.d(n,{demos:function(){return v}});var r=e(15009),c=e.n(r),i=e(99289),m=e.n(i),o=e(67294),p=e(92450),l=e(1653),d=e(69945),v={"docs-demo-nav-base-demo-nav-base":{component:o.memo(o.lazy(function(){return e.e(433).then(e.bind(e,47165))})),asset:{type:"BLOCK",id:"docs-demo-nav-base-demo-nav-base",refAtomIds:[],dependencies:{"index.jsx":{type:"FILE",value:e(70258).Z},react:{type:"NPM",value:"18.3.1"},"@rc-component/steps":{type:"NPM",value:"1.2.0"},"../../assets/iconfont.less":{type:"FILE",value:e(5057).Z},"../../assets/index.less":{type:"FILE",value:e(86834).Z}},entry:"index.jsx"},context:{"../../assets/iconfont.less":l,"../../assets/index.less":d,react:t||(t=e.t(o,2)),"@rc-component/steps":p,"/Users/jilin/projects/antd/rc-steps/assets/iconfont.less":l,"/Users/jilin/projects/antd/rc-steps/assets/index.less":d},renderOpts:{compile:function(){var u=m()(c()().mark(function E(){var _,x=arguments;return c()().wrap(function(s){for(;;)switch(s.prev=s.next){case 0:return s.next=2,e.e(19).then(e.bind(e,4019));case 2:return s.abrupt("return",(_=s.sent).default.apply(_,x));case 3:case"end":return s.stop()}},E)}));function I(){return u.apply(this,arguments)}return I}()}}}},55539:function(a,n,e){"use strict";var t;e.r(n),e.d(n,{demos:function(){return u}});var r=e(15009),c=e.n(r),i=e(99289),m=e.n(i),o=e(67294),p=e(92450),l=e(76803),d=e(69945),v=e(1653),u={"docs-demo-next-step-demo-nextstep":{component:o.memo(o.lazy(function(){return e.e(433).then(e.bind(e,68608))})),asset:{type:"BLOCK",id:"docs-demo-next-step-demo-nextstep",refAtomIds:[],dependencies:{"index.jsx":{type:"FILE",value:e(74167).Z},react:{type:"NPM",value:"18.3.1"},"@rc-component/steps":{type:"NPM",value:"1.2.0"},"./nextStep.css":{type:"FILE",value:e(74723).Z},"../../assets/index.less":{type:"FILE",value:e(86834).Z},"../../assets/iconfont.less":{type:"FILE",value:e(5057).Z}},entry:"index.jsx"},context:{"./nextStep.css":l,"../../assets/index.less":d,"../../assets/iconfont.less":v,react:t||(t=e.t(o,2)),"@rc-component/steps":p,"/Users/jilin/projects/antd/rc-steps/docs/examples/nextStep.css":l,"/Users/jilin/projects/antd/rc-steps/assets/index.less":d,"/Users/jilin/projects/antd/rc-steps/assets/iconfont.less":v},renderOpts:{compile:function(){var I=m()(c()().mark(function _(){var x,j=arguments;return c()().wrap(function(L){for(;;)switch(L.prev=L.next){case 0:return L.next=2,e.e(19).then(e.bind(e,4019));case 2:return L.abrupt("return",(x=L.sent).default.apply(x,j));case 3:case"end":return L.stop()}},_)}));function E(){return I.apply(this,arguments)}return E}()}}}},73704:function(a,n,e){"use strict";var t;e.r(n),e.d(n,{demos:function(){return v}});var r=e(15009),c=e.n(r),i=e(99289),m=e.n(i),o=e(67294),p=e(92450),l=e(1653),d=e(69945),v={"docs-demo-progress-dot-demo-progressdot":{component:o.memo(o.lazy(function(){return e.e(433).then(e.bind(e,95439))})),asset:{type:"BLOCK",id:"docs-demo-progress-dot-demo-progressdot",refAtomIds:[],dependencies:{"index.jsx":{type:"FILE",value:e(79818).Z},react:{type:"NPM",value:"18.3.1"},"@rc-component/steps":{type:"NPM",value:"1.2.0"},"../../assets/iconfont.less":{type:"FILE",value:e(5057).Z},"../../assets/index.less":{type:"FILE",value:e(86834).Z}},entry:"index.jsx"},context:{"../../assets/iconfont.less":l,"../../assets/index.less":d,react:t||(t=e.t(o,2)),"@rc-component/steps":p,"/Users/jilin/projects/antd/rc-steps/assets/iconfont.less":l,"/Users/jilin/projects/antd/rc-steps/assets/index.less":d},renderOpts:{compile:function(){var u=m()(c()().mark(function E(){var _,x=arguments;return c()().wrap(function(s){for(;;)switch(s.prev=s.next){case 0:return s.next=2,e.e(19).then(e.bind(e,4019));case 2:return s.abrupt("return",(_=s.sent).default.apply(_,x));case 3:case"end":return s.stop()}},E)}));function I(){return u.apply(this,arguments)}return I}()}}}},19128:function(a,n,e){"use strict";var t;e.r(n),e.d(n,{demos:function(){return v}});var r=e(15009),c=e.n(r),i=e(99289),m=e.n(i),o=e(67294),p=e(92450),l=e(1653),d=e(69945),v={"docs-demo-simple-demo-simple":{component:o.memo(o.lazy(function(){return e.e(433).then(e.bind(e,49187))})),asset:{type:"BLOCK",id:"docs-demo-simple-demo-simple",refAtomIds:[],dependencies:{"index.jsx":{type:"FILE",value:e(39755).Z},react:{type:"NPM",value:"18.3.1"},"@rc-component/steps":{type:"NPM",value:"1.2.0"},"../../assets/iconfont.less":{type:"FILE",value:e(5057).Z},"../../assets/index.less":{type:"FILE",value:e(86834).Z}},entry:"index.jsx"},context:{"../../assets/iconfont.less":l,"../../assets/index.less":d,react:t||(t=e.t(o,2)),"@rc-component/steps":p,"/Users/jilin/projects/antd/rc-steps/assets/iconfont.less":l,"/Users/jilin/projects/antd/rc-steps/assets/index.less":d},renderOpts:{compile:function(){var u=m()(c()().mark(function E(){var _,x=arguments;return c()().wrap(function(s){for(;;)switch(s.prev=s.next){case 0:return s.next=2,e.e(19).then(e.bind(e,4019));case 2:return s.abrupt("return",(_=s.sent).default.apply(_,x));case 3:case"end":return s.stop()}},E)}));function I(){return u.apply(this,arguments)}return I}()}}}},36095:function(a,n,e){"use strict";var t;e.r(n),e.d(n,{demos:function(){return v}});var r=e(15009),c=e.n(r),i=e(99289),m=e.n(i),o=e(67294),p=e(92450),l=e(1653),d=e(69945),v={"docs-demo-small-size-demo-smallsize":{component:o.memo(o.lazy(function(){return e.e(433).then(e.bind(e,22957))})),asset:{type:"BLOCK",id:"docs-demo-small-size-demo-smallsize",refAtomIds:[],dependencies:{"index.jsx":{type:"FILE",value:e(72578).Z},react:{type:"NPM",value:"18.3.1"},"@rc-component/steps":{type:"NPM",value:"1.2.0"},"../../assets/iconfont.less":{type:"FILE",value:e(5057).Z},"../../assets/index.less":{type:"FILE",value:e(86834).Z}},entry:"index.jsx"},context:{"../../assets/iconfont.less":l,"../../assets/index.less":d,react:t||(t=e.t(o,2)),"@rc-component/steps":p,"/Users/jilin/projects/antd/rc-steps/assets/iconfont.less":l,"/Users/jilin/projects/antd/rc-steps/assets/index.less":d},renderOpts:{compile:function(){var u=m()(c()().mark(function E(){var _,x=arguments;return c()().wrap(function(s){for(;;)switch(s.prev=s.next){case 0:return s.next=2,e.e(19).then(e.bind(e,4019));case 2:return s.abrupt("return",(_=s.sent).default.apply(_,x));case 3:case"end":return s.stop()}},E)}));function I(){return u.apply(this,arguments)}return I}()}}}},31447:function(a,n,e){"use strict";var t;e.r(n),e.d(n,{demos:function(){return v}});var r=e(15009),c=e.n(r),i=e(99289),m=e.n(i),o=e(67294),p=e(92450),l=e(1653),d=e(69945),v={"docs-demo-step-icon-demo-stepicon":{component:o.memo(o.lazy(function(){return e.e(433).then(e.bind(e,29932))})),asset:{type:"BLOCK",id:"docs-demo-step-icon-demo-stepicon",refAtomIds:[],dependencies:{"index.jsx":{type:"FILE",value:e(63042).Z},react:{type:"NPM",value:"18.3.1"},"@rc-component/steps":{type:"NPM",value:"1.2.0"},"../../assets/iconfont.less":{type:"FILE",value:e(5057).Z},"../../assets/index.less":{type:"FILE",value:e(86834).Z}},entry:"index.jsx"},context:{"../../assets/iconfont.less":l,"../../assets/index.less":d,react:t||(t=e.t(o,2)),"@rc-component/steps":p,"/Users/jilin/projects/antd/rc-steps/assets/iconfont.less":l,"/Users/jilin/projects/antd/rc-steps/assets/index.less":d},renderOpts:{compile:function(){var u=m()(c()().mark(function E(){var _,x=arguments;return c()().wrap(function(s){for(;;)switch(s.prev=s.next){case 0:return s.next=2,e.e(19).then(e.bind(e,4019));case 2:return s.abrupt("return",(_=s.sent).default.apply(_,x));case 3:case"end":return s.stop()}},E)}));function I(){return u.apply(this,arguments)}return I}()}}}},87250:function(a,n,e){"use strict";var t;e.r(n),e.d(n,{demos:function(){return v}});var r=e(15009),c=e.n(r),i=e(99289),m=e.n(i),o=e(67294),p=e(92450),l=e(1653),d=e(69945),v={"docs-demo-vertical-demo-vertical":{component:o.memo(o.lazy(function(){return e.e(433).then(e.bind(e,84866))})),asset:{type:"BLOCK",id:"docs-demo-vertical-demo-vertical",refAtomIds:[],dependencies:{"index.jsx":{type:"FILE",value:e(79053).Z},react:{type:"NPM",value:"18.3.1"},"@rc-component/steps":{type:"NPM",value:"1.2.0"},"../../assets/iconfont.less":{type:"FILE",value:e(5057).Z},"../../assets/index.less":{type:"FILE",value:e(86834).Z}},entry:"index.jsx"},context:{"../../assets/iconfont.less":l,"../../assets/index.less":d,react:t||(t=e.t(o,2)),"@rc-component/steps":p,"/Users/jilin/projects/antd/rc-steps/assets/iconfont.less":l,"/Users/jilin/projects/antd/rc-steps/assets/index.less":d},renderOpts:{compile:function(){var u=m()(c()().mark(function E(){var _,x=arguments;return c()().wrap(function(s){for(;;)switch(s.prev=s.next){case 0:return s.next=2,e.e(19).then(e.bind(e,4019));case 2:return s.abrupt("return",(_=s.sent).default.apply(_,x));case 3:case"end":return s.stop()}},E)}));function I(){return u.apply(this,arguments)}return I}()}}}},77943:function(a,n,e){"use strict";var t;e.r(n),e.d(n,{demos:function(){return v}});var r=e(15009),c=e.n(r),i=e(99289),m=e.n(i),o=e(67294),p=e(92450),l=e(1653),d=e(69945),v={"docs-demo-vertical-small-demo-verticalsmall":{component:o.memo(o.lazy(function(){return e.e(433).then(e.bind(e,42697))})),asset:{type:"BLOCK",id:"docs-demo-vertical-small-demo-verticalsmall",refAtomIds:[],dependencies:{"index.jsx":{type:"FILE",value:e(80532).Z},react:{type:"NPM",value:"18.3.1"},"@rc-component/steps":{type:"NPM",value:"1.2.0"},"../../assets/iconfont.less":{type:"FILE",value:e(5057).Z},"../../assets/index.less":{type:"FILE",value:e(86834).Z}},entry:"index.jsx"},context:{"../../assets/iconfont.less":l,"../../assets/index.less":d,react:t||(t=e.t(o,2)),"@rc-component/steps":p,"/Users/jilin/projects/antd/rc-steps/assets/iconfont.less":l,"/Users/jilin/projects/antd/rc-steps/assets/index.less":d},renderOpts:{compile:function(){var u=m()(c()().mark(function E(){var _,x=arguments;return c()().wrap(function(s){for(;;)switch(s.prev=s.next){case 0:return s.next=2,e.e(19).then(e.bind(e,4019));case 2:return s.abrupt("return",(_=s.sent).default.apply(_,x));case 3:case"end":return s.stop()}},E)}));function I(){return u.apply(this,arguments)}return I}()}}}},11171:function(a,n,e){"use strict";e.r(n),e.d(n,{demos:function(){return r}});var t=e(67294),r={}},92450:function(a,n,e){"use strict";e.r(n),e.d(n,{Step:function(){return ue},UnstableContext:function(){return E},default:function(){return Pe}});var t=e(97857),r=e.n(t),c=e(13769),i=e.n(c),m=e(93967),o=e.n(m),p=e(67294),l=e(9783),d=e.n(l),v=e(65306),u=e(85893);function I(f){var A=f.prefixCls,M=f.className,h=f.style,P=f.status,b="".concat(A,"-rail");return(0,u.jsx)("div",{className:o()(b,"".concat(b,"-").concat(P),M),style:h})}var E=p.createContext({}),_=p.createContext(null),x=e(3921),j=["className","style","children"],s=p.createContext({}),L=p.forwardRef(function(f,A){var M=f.className,h=f.style,P=f.children,b=i()(f,j),R=p.useContext(_),N=R.prefixCls,y=R.classNames,z=R.styles,W=p.useContext(s),$=W.className,S=W.style,O="".concat(N,"-item");return(0,u.jsx)("div",r()(r()({},(0,x.Z)(b,!1)),{},{ref:A,className:o()("".concat(O,"-icon"),y.itemIcon,$,M),style:r()(r()(r()({},z.itemIcon),S),h),children:P}))}),de=L,Ee=["onClick","title","subTitle","content","description","disabled","icon","status","className","style","classNames","styles"];function te(f){return f!=null}function ue(f){var A=f.prefixCls,M=f.classNames,h=f.styles,P=f.data,b=f.last,R=f.nextStatus,N=f.active,y=f.index,z=f.itemRender,W=f.iconRender,$=f.itemWrapperRender,S=f.onClick,O="".concat(A,"-item"),X=p.useContext(E),B=X.railFollowPrevStatus,F=P.onClick,G=P.title,K=P.subTitle,H=P.content,se=P.description,J=P.disabled,oe=P.icon,w=P.status,k=P.className,re=P.style,q=P.classNames,D=q===void 0?{}:q,Z=P.styles,C=Z===void 0?{}:Z,ae=i()(P,Ee),V=H!=null?H:se,ee={item:r()(r()({},P),{},{content:V}),index:y,active:N},Q=!!(S||F)&&!J,U={};Q&&(U.role="button",U.tabIndex=0,U.onClick=function(le){F==null||F(le),S(y)},U.onKeyDown=function(le){var _e=le.which;(_e===v.Z.ENTER||_e===v.Z.SPACE)&&S(y)});var g=w||"wait",T=te(G),ne=te(K),ce=o()(O,"".concat(O,"-").concat(g),d()(d()(d()(d()({},"".concat(O,"-custom"),oe),"".concat(O,"-active"),N),"".concat(O,"-disabled"),J===!0),"".concat(O,"-empty-header"),!T&&!ne),k,M.item,D.root),Y=(0,u.jsx)(de,{});W&&(Y=W(Y,r()(r()({},ee),{},{components:{Icon:de}})));var me=(0,u.jsxs)("div",{className:o()("".concat(O,"-wrapper"),M.itemWrapper,D.wrapper),style:r()(r()({},h.itemWrapper),C.wrapper),children:[(0,u.jsx)(s.Provider,{value:{className:D.icon,style:C.icon},children:Y}),(0,u.jsxs)("div",{className:o()("".concat(O,"-section"),M.itemSection,D.section),style:r()(r()({},h.itemSection),C.section),children:[(0,u.jsxs)("div",{className:o()("".concat(O,"-header"),M.itemHeader,D.header),style:r()(r()({},h.itemHeader),C.header),children:[T&&(0,u.jsx)("div",{className:o()("".concat(O,"-title"),M.itemTitle,D.title),style:r()(r()({},h.itemTitle),C.title),children:G}),ne&&(0,u.jsx)("div",{title:typeof K=="string"?K:void 0,className:o()("".concat(O,"-subtitle"),M.itemSubtitle,D.subtitle),style:r()(r()({},h.itemSubtitle),C.subtitle),children:K}),!b&&(0,u.jsx)(I,{prefixCls:O,className:o()(M.itemRail,D.rail),style:r()(r()({},h.itemRail),C.rail),status:B?w:R})]}),te(V)&&(0,u.jsx)("div",{className:o()("".concat(O,"-content"),M.itemContent,D.content),style:r()(r()({},h.itemContent),C.content),children:V})]})]}),ie=(0,u.jsx)("li",r()(r()(r()({},ae),U),{},{className:ce,style:r()(r()(r()({},h.item),C.root),re),children:$?$(me):me}));return z&&(ie=z(ie,ee)||null),ie}var xe=["prefixCls","style","className","classNames","styles","rootClassName","orientation","titlePlacement","status","current","initial","onChange","items","iconRender","itemRender","itemWrapperRender"],pe={};function je(f){var A=f.prefixCls,M=A===void 0?"rc-steps":A,h=f.style,P=f.className,b=f.classNames,R=b===void 0?pe:b,N=f.styles,y=N===void 0?pe:N,z=f.rootClassName,W=f.orientation,$=f.titlePlacement,S=f.status,O=S===void 0?"process":S,X=f.current,B=X===void 0?0:X,F=f.initial,G=F===void 0?0:F,K=f.onChange,H=f.items,se=f.iconRender,J=f.itemRender,oe=f.itemWrapperRender,w=i()(f,xe),k=W==="vertical",re=k?"vertical":"horizontal",q=!k&&$==="vertical"?"vertical":"horizontal",D=o()(M,"".concat(M,"-").concat(re),"".concat(M,"-title-").concat(q),z,P,R.root),Z=p.useMemo(function(){return(H||[]).filter(Boolean)},[H]),C=p.useMemo(function(){return Z.map(function(Q,U){var g=Q.status,T=G+U;return g||(T===B?O:T<B?"finish":"wait")})},[Z,O,B,G]),ae=function(U){K&&B!==U&&K(U)},V=p.useMemo(function(){return{prefixCls:M,classNames:R,styles:y}},[M,R,y]),ee=function(U,g){var T=G+g,ne=C[g],ce=C[g+1],Y=r()(r()({},U),{},{status:ne});return(0,u.jsx)(ue,{prefixCls:M,classNames:R,styles:y,data:Y,nextStatus:ce,active:T===B,index:T,last:Z.length-1===g,iconRender:se,itemRender:J,itemWrapperRender:oe,onClick:K&&ae},T)};return(0,u.jsx)("ol",r()(r()({className:D,style:r()(r()({},h),y==null?void 0:y.root)},w),{},{children:(0,u.jsx)(_.Provider,{value:V,children:Z.map(ee)})}))}var Pe=je},76803:function(a,n,e){"use strict";e.r(n)},1653:function(a,n,e){"use strict";e.r(n)},69945:function(a,n,e){"use strict";e.r(n)},14418:function(a,n,e){"use strict";e.r(n),e.d(n,{texts:function(){return t}});const t=[]},55907:function(a,n,e){"use strict";e.r(n),e.d(n,{texts:function(){return t}});const t=[]},91122:function(a,n,e){"use strict";e.r(n),e.d(n,{texts:function(){return t}});const t=[]},37200:function(a,n,e){"use strict";e.r(n),e.d(n,{texts:function(){return t}});const t=[]},49428:function(a,n,e){"use strict";e.r(n),e.d(n,{texts:function(){return t}});const t=[]},67829:function(a,n,e){"use strict";e.r(n),e.d(n,{texts:function(){return t}});const t=[]},59408:function(a,n,e){"use strict";e.r(n),e.d(n,{texts:function(){return t}});const t=[]},63829:function(a,n,e){"use strict";e.r(n),e.d(n,{texts:function(){return t}});const t=[]},71933:function(a,n,e){"use strict";e.r(n),e.d(n,{texts:function(){return t}});const t=[]},60500:function(a,n,e){"use strict";e.r(n),e.d(n,{texts:function(){return t}});const t=[]},49726:function(a,n,e){"use strict";e.r(n),e.d(n,{texts:function(){return t}});const t=[]},65081:function(a,n,e){"use strict";e.r(n),e.d(n,{texts:function(){return t}});const t=[]},86364:function(a,n,e){"use strict";e.r(n),e.d(n,{texts:function(){return t}});const t=[]},15013:function(a,n,e){"use strict";e.r(n),e.d(n,{texts:function(){return t}});const t=[]},79465:function(a,n,e){"use strict";e.r(n),e.d(n,{texts:function(){return t}});const t=[]},51446:function(a,n,e){"use strict";e.r(n),e.d(n,{texts:function(){return t}});const t=[{value:"React steps component.",paraId:0,tocIndex:0},{value:`npm install @rc-component/steps
`,paraId:1,tocIndex:1},{value:`<Steps current={1}>
  <Steps.Step title="first" />
  <Steps.Step title="second" />
  <Steps.Step title="third" />
</Steps>
`,paraId:2,tocIndex:1},{value:"https://steps.vercel.app/",paraId:3,tocIndex:2},{value:`
  `,paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"name",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"type",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"default",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"description",paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
  `,paraId:4,tocIndex:3},{value:`
  `,paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"type",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"string",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"default",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"diretypetion of Steps, could be `default` `navigation` `inline`",paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"direction",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"string",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"horizontal",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"direction of Steps, enum: `horizontal` or `vertical`",paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"current",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"number",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"0",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"index of current step",paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"initial",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"number",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"0",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"index initial",paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"size",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"string",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"size of Steps, could be `small`",paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"titlePlacement",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"string",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"placement of step title, could be `vertical`",paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"status",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"string",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"wait",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"status of current Steps, could be `error` `process` `finish` `wait`",paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"icons",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"{ finish: ReactNode, error: ReactNode }",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"specify the default finish icon and error icon",paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"itemRender",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"(item: StepProps, stepItem: React.ReactNode) => React.ReactNode",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"custom step item renderer",paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"onChange",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"(current: number) => void",paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:`
      `,paraId:4,tocIndex:3},{value:"Trigger when Step changed",paraId:4,tocIndex:3},{value:`
    `,paraId:4,tocIndex:3},{value:`
  `,paraId:4,tocIndex:3},{value:`
  `,paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"name",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"type",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"default",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"description",paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
  `,paraId:5,tocIndex:4},{value:`
  `,paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"title",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"ReactNode",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"title of step item",paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"subTitle",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"ReactNode",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"subTitle of step item",paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"description",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"ReactNode",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"description of step item",paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"icon",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"ReactNode",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"set icon of step item",paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"status",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"string",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"status of current Steps, could be `error` `process` `finish` `wait`",paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"tailContent",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"ReactNode",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"content above tail",paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"disabled",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"bool",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"false",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"disabled step when onChange exist",paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"render",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"(stepItem: React.ReactNode) => React.ReactNode",paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:`
      `,paraId:5,tocIndex:4},{value:"custom step item renderer",paraId:5,tocIndex:4},{value:`
    `,paraId:5,tocIndex:4},{value:`
  `,paraId:5,tocIndex:4},{value:`npm install
npm start
`,paraId:6,tocIndex:5},{value:"@rc-component/steps is released under the MIT license.",paraId:7,tocIndex:6}]},5057:function(a,n){"use strict";n.Z=`@icon-url               : "//at.alicdn.com/t/font_1434092639_4910953";
.ie-rotate(@rotation) {
  filter: progid:DXImageTransform.Microsoft.BasicImage(rotation=@rotation);
}
.rotate(@degrees) {
  -webkit-transform: rotate(@degrees);
  -ms-transform: rotate(@degrees); // IE9 only
  -o-transform: rotate(@degrees);
  transform: rotate(@degrees);
}
.animation(@animation) {
  -webkit-animation: @animation;
  -o-animation: @animation;
  animation: @animation;
}
// font-face
// @icon-url\uFF1A \u5B57\u4F53\u6E90\u6587\u4EF6\u7684\u5730\u5740
@font-face {
  font-family: 'anticon';
  src: url('@{icon-url}.eot'); /* IE9*/
  src: url('@{icon-url}.eot?#iefix') format('embedded-opentype'), /* IE6-IE8 */
  url('@{icon-url}.woff') format('woff'), /* chrome\u3001firefox */
  url('@{icon-url}.ttf') format('truetype'), /* chrome\u3001firefox\u3001opera\u3001Safari, Android, iOS 4.2+*/
  url('@{icon-url}.svg#iconfont') format('svg'); /* iOS 4.1- */
}

.rcicon {
  position: relative;
  display: inline-block;
  font-style: normal;
  vertical-align: baseline;
  text-align: center;
  text-transform: none;
  text-rendering: auto;
  // \u66F4\u597D\u5730\u6E32\u67D3\u5B57\u4F53
  -webkit-font-smoothing: antialiased;
  -webkit-text-stroke-width: 0px;
  -moz-osx-font-smoothing: grayscale;

  &:before {
    display: block;
    font-family: "anticon" !important;
  }
}
// \u65B9\u5411\u6027\u56FE\u6807
.rcicon-step-backward:before          {content:"\\e662";}
.rcicon-step-forward                  {.ie-rotate(2);}
.rcicon-step-forward:before           {content:"\\e662";.rotate(180deg);}
.rcicon-fast-backward:before          {content:"\\e62a";}
.rcicon-fast-forward                  {.ie-rotate(2);}
.rcicon-fast-forward:before           {content:"\\e62a";.rotate(180deg);}
.rcicon-shrink:before                 {content:"\\e65f";}
.rcicon-arrow-salt:before             {content:"\\e608";}
.rcicon-caret-down:before             {content:"\\e60f";}
.rcicon-caret-left                    {.ie-rotate(1);}
.rcicon-caret-left:before             {content:"\\e60f";.rotate(90deg);}
.rcicon-caret-up                      {.ie-rotate(2);}
.rcicon-caret-up:before               {content:"\\e60f";.rotate(180deg);}
.rcicon-caret-right                   {.ie-rotate(3);}
.rcicon-caret-right:before            {content:"\\e60f";.rotate(270deg);}
.rcicon-caret-circle-right:before     {content:"\\e60d";}
.rcicon-caret-circle-left             {.ie-rotate(2);}
.rcicon-caret-circle-left:before      {content:"\\e60d";.rotate(180deg);}
.rcicon-caret-circle-o-right:before   {content:"\\e60e";}
.rcicon-caret-circle-o-left           {.ie-rotate(2);}
.rcicon-caret-circle-o-left:before    {content:"\\e60e";.rotate(180deg);}
.rcicon-circle-right:before           {content:"\\e602";}
.rcicon-circle-left                   {.ie-rotate(2);}
.rcicon-circle-left:before            {content:"\\e602";.rotate(180deg);}
.rcicon-circle-o-right:before         {content:"\\e603";}
.rcicon-circle-o-left                 {.ie-rotate(2);}
.rcicon-circle-o-left:before          {content:"\\e603";.rotate(180deg);}
.rcicon-double-right:before           {content:"\\e604";}
.rcicon-double-left                   {.ie-rotate(2);}
.rcicon-double-left:before            {content:"\\e604";.rotate(180deg);}
.rcicon-verticle-right:before         {content:"\\e605";}
.rcicon-verticle-left                 {.ie-rotate(2);}
.rcicon-verticle-left:before          {content:"\\e605";.rotate(180deg);}
.rcicon-forward:before                {content:"\\e630";}
.rcicon-backward                      {.ie-rotate(2);}
.rcicon-backward:before               {content:"\\e630";.rotate(180deg);}
.rcicon-rollback:before               {content:"\\e65a";}
.rcicon-retweet:before                {content:"\\e659";}

.rcicon-right:before                  {content:"\\e611";}
.rcicon-down                          {.ie-rotate(1);}
.rcicon-down:before                   {content:"\\e611";.rotate(90deg);}
.rcicon-left                          {.ie-rotate(2);}
.rcicon-left:before                   {content:"\\e611";.rotate(180deg);}
.rcicon-up                            {.ie-rotate(3);}
.rcicon-up:before                     {content:"\\e611";.rotate(270deg);}

// \u63D0\u793A\u6027\u56FE\u6807
.rcicon-question:before               {content:"\\e655";}
.rcicon-question-circle:before        {content:"\\e656";}
.rcicon-question-circle-o:before      {content:"\\e657";}
.rcicon-plus:before                   {content:"\\e651";}
.rcicon-plus-circle:before            {content:"\\e652";}
.rcicon-plus-circle-o:before          {content:"\\e653";}
.rcicon-pause:before                  {content:"\\e64c";}
.rcicon-pause-circle:before           {content:"\\e64d";}
.rcicon-pause-circle-o:before         {content:"\\e64e";}
.rcicon-minus:before                  {content:"\\e646";}
.rcicon-minus-circle:before           {content:"\\e647";}
.rcicon-minus-circle-o:before         {content:"\\e648";}
.rcicon-info-circle:before            {content:"\\e637";}
.rcicon-info-circle-o:before          {content:"\\e638";}
.rcicon-info:before                   {content:"\\e63a";}
.rcicon-exclamation:before            {content:"\\e627";}
.rcicon-exclamation-circle:before     {content:"\\e628";}
.rcicon-exclamation-circle-o:before   {content:"\\e629";}
.rcicon-cross:before                  {content:"\\e61e";}
.rcicon-cross-circle:before           {content:"\\e61f";}
.rcicon-cross-circle-o:before         {content:"\\e620";}
.rcicon-check:before                  {content:"\\e613";}
.rcicon-check-circle:before           {content:"\\e614";}
.rcicon-check-circle-o:before         {content:"\\e615";}
.rcicon-clock-circle:before           {content:"\\e616";}
.rcicon-clock-circle-o:before         {content:"\\e617";}

// \u7F51\u7AD9\u901A\u7528\u56FE\u6807
.rcicon-lock:before                   {content:"\\e641";}
.rcicon-android:before                {content:"\\e601";}
.rcicon-apple:before                  {content:"\\e606";}
.rcicon-area-chart:before             {content:"\\e607";}
.rcicon-bar-chart:before              {content:"\\e609";}
.rcicon-bars:before                   {content:"\\e60a";}
.rcicon-book:before                   {content:"\\e60b";}
.rcicon-calendar:before               {content:"\\e60c";}
.rcicon-cloud:before                  {content:"\\e618";}
.rcicon-cloud-download:before         {content:"\\e619";}
.rcicon-code:before                   {content:"\\e61a";}
.rcicon-copy:before                   {content:"\\e61c";}
.rcicon-credit-card:before            {content:"\\e61d";}
.rcicon-delete:before                 {content:"\\e621";}
.rcicon-desktop:before                {content:"\\e622";}
.rcicon-download-line:before          {content:"\\e623";}
.rcicon-edit:before                   {content:"\\e624";}
.rcicon-ellipsis:before               {content:"\\e625";}
.rcicon-environment:before            {content:"\\e626";}
.rcicon-file:before                   {content:"\\e62c";}
.rcicon-file-text:before              {content:"\\e62d";}
.rcicon-folder:before                 {content:"\\e62e";}
.rcicon-folder-open:before            {content:"\\e62f";}
.rcicon-github:before                 {content:"\\e631";}
.rcicon-hdd:before                    {content:"\\e632";}
.rcicon-frown:before                  {content:"\\e633";}
.rcicon-meh:before                    {content:"\\e634";}
.rcicon-inbox:before                  {content:"\\e635";}
.rcicon-laptop:before                 {content:"\\e63d";}
.rcicon-large:before                  {content:"\\e63e";}
.rcicon-line-chart:before             {content:"\\e63f";}
.rcicon-link:before                   {content:"\\e640";}
.rcicon-logout:before                 {content:"\\e642";}
.rcicon-mail:before                   {content:"\\e643";}
.rcicon-menu-fold:before              {content:"\\e644";}
.rcicon-menu-unfold:before            {content:"\\e645";}
.rcicon-mobile:before                 {content:"\\e649";}
.rcicon-notification:before           {content:"\\e64a";}
.rcicon-paper-clip:before             {content:"\\e64b";}
.rcicon-picture:before                {content:"\\e64f";}
.rcicon-pie-chart:before              {content:"\\e650";}
.rcicon-poweroff:before               {content:"\\e654";}
.rcicon-reload:before                 {content:"\\e658";}
.rcicon-search:before                 {content:"\\e65b";}
.rcicon-setting:before                {content:"\\e65c";}
.rcicon-share-alt:before              {content:"\\e65d";}
.rcicon-shopping-cart:before          {content:"\\e65e";}
.rcicon-smile:before                  {content:"\\e661";}
.rcicon-tablet:before                 {content:"\\e664";}
.rcicon-tag:before                    {content:"\\e665";}
.rcicon-tags:before                   {content:"\\e666";}
.rcicon-to-top:before                  {content:"\\e667";}
.rcicon-unlock:before                 {content:"\\e668";}
.rcicon-upload:before                 {content:"\\e669";}
.rcicon-user:before                   {content:"\\e66a";}
.rcicon-video-camera:before           {content:"\\e66b";}
.rcicon-windows:before                {content:"\\e66c";}
.rcicon-loading:before {
  display: inline-block;
  .animation(loadingCircle 1.0s infinite linear);
  content:"\\e610";
}

:root {
  .rcicon-step-forward,
  .rcicon-fast-forward,
  .rcicon-left,
  .rcicon-up,
  .rcicon-down,
  .rcicon-caret-left,
  .rcicon-caret-up,
  .rcicon-caret-right,
  .rcicon-caret-circle-left,
  .rcicon-caret-circle-o-left,
  .rcicon-circle-left,
  .rcicon-circle-o-left,
  .rcicon-double-left,
  .rcicon-verticle-left,
  .rcicon-backward {
    filter: none;
  }
}
`},86834:function(a,n){"use strict";n.Z=`@import 'variables';

.@{stepsPrefixClass} {
  font-size: 0;
  width: 100%;
  line-height: 1.5;
  display: flex;

  &,
  * {
    box-sizing: border-box;
  }
}

.@{stepsPrefixClass}-item {
  position: relative;
  display: inline-block;
  vertical-align: top;
  flex: 1;
  // overflow: hidden;

  &:last-child {
    flex: none;
  }

  &:last-child &-title:after {
    display: none;
  }

  &-icon,
  &-section {
    display: inline-block;
    vertical-align: top;
  }

  &-icon {
    flex: none;
    border: 1px solid @wait-icon-color;
    width: 26px;
    height: 26px;
    line-height: 26px;
    text-align: center;
    border-radius: 26px;
    font-size: 14px;
    transition:
      background-color 0.3s,
      border-color 0.3s;

    > .@{stepsPrefixClass}-icon {
      line-height: 1;
      top: -1px;
      color: @primary-color;
      position: relative;

      &.rcicon {
        font-size: 12px;
        position: relative;
        top: -2px;
      }
    }
  }

  &-section {
    margin-top: 3px;
  }
  &-title {
    font-size: 14px;
    color: #666;
    font-weight: bold;
    display: inline-block;
    position: relative;
  }
  &-subtitle {
    font-size: 12px;
    display: inline-block;
    color: #999;
  }
  &-description {
    font-size: 12px;
    color: #999;
  }
  .step-item-status(wait);
  .step-item-status(process);
  &-process &-icon {
    background: @process-icon-color;
    > .@{stepsPrefixClass}-icon {
      color: #fff;
    }
  }
  .step-item-status(finish);
  .step-item-status(error);

  &.@{stepsPrefixClass}-next-error .@{stepsPrefixClass}-item-title:after {
    background: @error-icon-color;
  }
}

.@{stepsPrefixClass}-horizontal:not(.@{stepsPrefixClass}-label-vertical) {
  .@{stepsPrefixClass}-item {
    &-description {
      max-width: @stepDescriptionMaxWidth;
    }
  }
}

.step-item-status(@status) {
  @icon-color: '@{status}-icon-color';
  @title-color: '@{status}-title-color';
  @description-color: '@{status}-description-color';
  @tail-color: '@{status}-tail-color';
  &-@{status} &-icon {
    border-color: @@icon-color;
    background-color: #fff;
    > .@{stepsPrefixClass}-icon {
      color: @@icon-color;
      .@{stepsPrefixClass}-icon-dot {
        background: @@icon-color;
      }
    }
  }
  &-@{status} &-title {
    color: @@title-color;
  }
  &-@{status} &-description {
    color: @@description-color;
  }
}

// @import 'custom-icon';
// @import 'small';
// @import 'vertical';
// @import 'label-placement';
// @import 'progress-dot';
// @import 'nav';
// @import 'inline';

// ======================= Horizontal =======================
.verticalFlex() {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.@{stepsPrefixClass} {
  .@{stepsPrefixClass}-item {
    &-section {
      min-width: 0;
    }

    &-header {
      display: flex;
      gap: 8px;
      align-items: center;
    }

    // Ellipsis
    &-title,
    &-subtitle,
    &-description {
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
    }
  }
}

.@{stepsPrefixClass}-horizontal {
  .@{stepsPrefixClass}-item {
    flex: 1;
    position: relative;
    min-width: 0;

    &-rail {
      height: 1px;
      background: @process-tail-color;
    }
  }

  // Label Vertical
  &.@{stepsPrefixClass}-label-vertical {
    .@{stepsPrefixClass}-item {
      .verticalFlex();
      padding-inline: 8px;

      &-section {
        .verticalFlex();
      }

      &-rail {
        position: absolute;
        top: 13px;
        left: calc(50% + 13px);
        width: 100%;
      }
    }
  }

  // Label Horizontal
  &.@{stepsPrefixClass}-label-horizontal {
    .@{stepsPrefixClass}-item {
      display: flex;

      &:last-child {
        flex: none;
      }

      &-section {
        flex: 1;
      }

      &-rail {
        flex: 1;
        min-width: 0;
      }
    }
  }
}

// ======================== Vertical ========================
.@{stepsPrefixClass}-vertical {
}
`},66328:function(a,n){"use strict";n.Z=`import '../../assets/index.less';
import '../../assets/iconfont.less';
import React from 'react';
import Steps from '@rc-component/steps';

const description =
  '\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A';

export default () => (
  <Steps
    titlePlacement="vertical"
    current={1}
    items={[
      {
        title: '\u5DF2\u5B8C\u6210',
        description,
        status: 'wait',
      },
      {
        title: '\u8FDB\u884C\u4E2D',
        description,
        status: 'wait',
        subTitle: '\u5269\u4F59 00:00:07',
      },
      undefined,
      {
        title: '\u5F85\u8FD0\u884C',
        description,
        status: 'process',
      },
      false,
      {
        title: '\u5F85\u8FD0\u884C',
        description,
        status: 'finish',
        disabled: true,
      },
      null,
    ]}
  />
);
`},39296:function(a,n){"use strict";n.Z=`import '../../assets/index.less';
import '../../assets/iconfont.less';
import React from 'react';
import Steps from '@rc-component/steps';

const description =
  '\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036';

export default () => (
  <Steps
    current={1}
    items={[
      {
        title: '\u5DF2\u5B8C\u6210',
        description,
      },
      {
        title: '\u8FDB\u884C\u4E2D',
        description,
      },
      {
        title: '\u8FDB\u884C\u4E2D',
        description,
        style: { fontWeight: 'bold', fontStyle: 'italic' },
      },
      {
        title: '\u5F85\u8FD0\u884C',
        description,
      },
    ]}
  />
);
`},4927:function(a,n){"use strict";n.Z=`import '../../assets/index.less';
import '../../assets/iconfont.less';
import React from 'react';
import Steps from '@rc-component/steps';

function getFinishIcon() {
  const path =
    'M923 283.6c-13.4-31.1-32.6-58.9-56.9-82.8-24.3-23.8-52.' +
    '5-42.4-84-55.5-32.5-13.5-66.9-20.3-102.4-20.3-49.3 0-97.4 13.5-139' +
    '.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5' +
    '-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 ' +
    '55.5-24.4 23.9-43.5 51.7-56.9 82.8-13.9 32.3-21 66.6-21 101.9 0 33' +
    '.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99' +
    '.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.' +
    '7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 10' +
    '1.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 ' +
    '20.3-103.3 0.1-35.3-7-69.6-20.9-101.9z';
  return (
    <svg
      width="1em"
      height="1em"
      fill="currentColor"
      viewBox="0 0 1024 1024"
      style={{ verticalAlign: '-.125em' }}
    >
      <path d={path} />
    </svg>
  );
}

function getErrorIcon() {
  const path1 =
    'M512 0C229.2 0 0 229.2 0 512s229.2 512 512 512 512-229' +
    '.2 512-512S794.8 0 512 0zm311.1 823.1c-40.4 40.4-87.5 72.2-139.9 9' +
    '4.3C629 940.4 571.4 952 512 952s-117-11.6-171.2-34.5c-52.4-22.2-99' +
    '.4-53.9-139.9-94.3-40.4-40.4-72.2-87.5-94.3-139.9C83.6 629 72 571.' +
    '4 72 512s11.6-117 34.5-171.2c22.2-52.4 53.9-99.4 94.3-139.9 40.4-4' +
    '0.4 87.5-72.2 139.9-94.3C395 83.6 452.6 72 512 72s117 11.6 171.2 3' +
    '4.5c52.4 22.2 99.4 53.9 139.9 94.3 40.4 40.4 72.2 87.5 94.3 139.9C' +
    '940.4 395 952 452.6 952 512s-11.6 117-34.5 171.2c-22.2 52.4-53.9 9' +
    '9.5-94.4 139.9z';
  const path2 =
    'M640.3 765.5c-19.9 0-36-16.1-36-36 0-50.9-41.4-92.3-92' +
    '.3-92.3s-92.3 41.4-92.3 92.3c0 19.9-16.1 36-36 36s-36-16.1-36-36c0' +
    '-90.6 73.7-164.3 164.3-164.3s164.3 73.7 164.3 164.3c0 19.9-16.1 36' +
    '-36 36zM194.2 382.4a60 60 0 1 0 120 0 60 60 0 1 0-120 0zM709.5 382' +
    '.4a60 60 0 1 0 120 0 60 60 0 1 0-120 0z';
  return (
    <svg
      width="1em"
      height="1em"
      fill="currentColor"
      viewBox="0 0 1024 1024"
      style={{ verticalAlign: '-.125em' }}
    >
      <path d={path1} />
      <path d={path2} />
    </svg>
  );
}

const icons = {
  finish: getFinishIcon(),
  error: getErrorIcon(),
};

const description = 'This is a description';

export default () => (
  <Steps
    current={1}
    status="error"
    icons={icons}
    items={[
      { title: 'Finished', description },
      { title: 'In Process', description },
      { title: 'Waiting', description },
    ]}
  />
);
`},36393:function(a,n){"use strict";n.Z=`import '../../assets/index.less';
import '../../assets/iconfont.less';
import React from 'react';
import Steps from '@rc-component/steps';

// eslint-disable-next-line react/prop-types
const Icon = ({ type }) => <i className={\`rcicon rcicon-\${type}\`} />;

export default () => (
  <Steps
    current={1}
    items={[
      { title: '\u6B65\u9AA41', icon: <Icon type="cloud" /> },
      { title: '\u6B65\u9AA42', icon: 'apple' },
      { title: '\u6B65\u9AA41', icon: 'github' },
    ]}
  />
);
`},31626:function(a,n){"use strict";n.Z=`import '../../assets/index.less';
import '../../assets/iconfont.less';
import React, { useState } from 'react';
import Steps from '@rc-component/steps';

export default () => {
  const [items, setItems] = useState([
    {
      title: '\u5DF2\u5B8C\u6210',
      description:
        '\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036',
    },
    {
      title: '\u8FDB\u884C\u4E2D',
      description:
        '\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036',
    },
    {
      title: '\u5F85\u8FD0\u884C',
      description:
        '\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036',
    },
    {
      title: '\u5F85\u8FD0\u884C',
      description:
        '\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036',
    },
  ]);

  const addStep = () => {
    const newSteps = [...items];
    newSteps.push({
      title: '\u5F85\u8FD0\u884C',
      description: '\u65B0\u7684\u8282\u70B9',
    });
    setItems(newSteps);
  };
  return (
    <div>
      <button type="button" onClick={addStep}>
        Add new step
      </button>
      <Steps items={items} />
    </div>
  );
};
`},75787:function(a,n){"use strict";n.Z=`import '../../assets/index.less';
import '../../assets/iconfont.less';
import React from 'react';
import Steps from '@rc-component/steps';

const description =
  '\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A';

export default () => (
  <Steps
    current={2}
    status="error"
    items={[
      {
        title: '\u5DF2\u5B8C\u6210',
        description,
      },
      {
        title: '\u8FDB\u884C\u4E2D',
        description,
      },
      {
        title: '\u5F85\u8FD0\u884C',
        description,
      },
      {
        title: '\u5F85\u8FD0\u884C',
        description,
      },
    ]}
  />
);
`},32149:function(a,n){"use strict";n.Z=`import '../../assets/index.less';
import React, { useState } from 'react';
import Steps from '@rc-component/steps';

export default () => {
  const [current, setCurrent] = useState(0);

  return (
    <>
      <button
        onClick={() => {
          setCurrent(0);
        }}
      >
        Current: {current}
      </button>

      <br />

      <Steps
        type="inline"
        current={current}
        onChange={setCurrent}
        items={[
          {
            title: '\u5F00\u53D1',
            description: '\u5F00\u53D1\u9636\u6BB5\uFF1A\u5F00\u53D1\u4E2D',
          },
          {
            title: '\u6D4B\u8BD5',
            description: '\u6D4B\u8BD5\u9636\u6BB5\uFF1A\u6D4B\u8BD5\u4E2D',
          },
          {
            title: '\u9884\u53D1',
            description: '\u9884\u53D1\u9636\u6BB5\uFF1A\u9884\u53D1\u4E2D',
          },
          {
            title: '\u53D1\u5E03',
            description: '\u53D1\u5E03\u9636\u6BB5\uFF1A\u53D1\u5E03\u4E2D',
          },
        ]}
        itemRender={(item, stepItem) => React.cloneElement(stepItem, { title: item.description })}
      />
    </>
  );
};
`},70258:function(a,n){"use strict";n.Z=`import '../../assets/index.less';
import '../../assets/iconfont.less';
import React, { useState } from 'react';
import Steps from '@rc-component/steps';

export default () => {
  const [current, setCurrent] = useState(0);

  const onChange = (current) => {
    // eslint-disable-next-line no-console
    console.log('onChange:', current);
    setCurrent(current);
  };

  const containerStyle = {
    border: '1px solid rgb(235, 237, 240)',
    marginBottom: 24,
  };

  const description = 'This is a description.';

  return (
    <div>
      <Steps
        style={containerStyle}
        type="navigation"
        current={current}
        onChange={onChange}
        items={[
          {
            title: 'Step 1',
            status: 'finish',
            subTitle: '\u5269\u4F59 00:00:05 \u8D85\u957F\u9690\u85CF',
            description,
          },
          {
            title: 'Step 2',
            status: 'process',
            description,
          },
          {
            title: 'Step 3',
            status: 'wait',
            description,
            disabled: true,
          },
        ]}
      />
      <Steps
        style={containerStyle}
        type="navigation"
        current={current}
        onChange={onChange}
        items={[
          {
            title: 'Step 1',
            status: 'finish',
            subTitle: '\u5269\u4F59 00:00:05 \u8D85\u957F\u9690\u85CF',
          },
          {
            title: 'Step 2',
            status: 'process',
          },
          {
            title: 'Step 3',
            status: 'wait',
          },
          {
            title: 'Step 3',
            status: 'wait',
          },
        ]}
      />
    </div>
  );
};
`},74723:function(a,n){"use strict";n.Z=`.my-step-form {
  width: 100%;
}
.my-step-form > div {
  margin-bottom: 20px;
}
.my-step-container {
  width: 100%;
}
`},74167:function(a,n){"use strict";n.Z=`import '../../assets/index.less';
import '../../assets/iconfont.less';
import './nextStep.css';
import React from 'react';
import Steps from '@rc-component/steps';

function generateRandomSteps() {
  const n = Math.floor(Math.random() * 3) + 3;
  const arr = [];
  for (let i = 0; i < n; i++) {
    arr.push({
      title: \`\u6B65\u9AA4\${i + 1}\`,
    });
  }
  return arr;
}
const steps = generateRandomSteps();

class MyForm extends React.Component {
  state = {
    currentStep: Math.floor(Math.random() * steps.length),
  };

  nextStep = () => {
    const { currentStep } = this.state;
    let s = currentStep + 1;
    if (s === steps.length) {
      s = 0;
    }
    this.setState({
      currentStep: s,
    });
  };

  render() {
    const { currentStep: cs } = this.state;
    this.stepsRefs = [];
    return (
      <form className="my-step-form">
        <div>\u8FD9\u4E2Ademo\u968F\u673A\u751F\u62103~6\u4E2A\u6B65\u9AA4\uFF0C\u521D\u59CB\u968F\u673A\u8FDB\u884C\u5230\u5176\u4E2D\u4E00\u4E2A\u6B65\u9AA4</div>
        <div>\u5F53\u524D\u6B63\u5728\u6267\u884C\u7B2C{cs + 1}\u6B65</div>
        <div className="my-step-container">
          <Steps
            current={cs}
            items={steps.map((s, i) => ({
              ref: (c) => {
                this.stepsRefs[i] = c;
              },
              key: i,
              title: s.title,
            }))}
          />
        </div>

        <div>
          <button type="button" onClick={this.nextStep}>
            \u4E0B\u4E00\u6B65
          </button>
        </div>
      </form>
    );
  }
}

export default MyForm;
`},79818:function(a,n){"use strict";n.Z=`import '../../assets/index.less';
import '../../assets/iconfont.less';
import React from 'react';
import Steps from '@rc-component/steps';

const description =
  '\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A';

export default () => (
  <Steps
    progressDot
    size="small"
    current={1}
    items={[
      {
        title: '\u5DF2\u5B8C\u6210',
        description,
      },
      {
        title: '\u8FDB\u884C\u4E2D',
        description,
      },
      {
        title: '\u5F85\u8FD0\u884C',
        description,
      },
      {
        title: '\u5F85\u8FD0\u884C',
        description,
      },
      {
        title: '\u5F85\u8FD0\u884C',
        description,
      },
    ]}
  />
);
`},39755:function(a,n){"use strict";n.Z=`import '../../assets/index.less';
import '../../assets/iconfont.less';
import React from 'react';
import Steps from '@rc-component/steps';

const description =
  '\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u63CF\u8FF0\u554A\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036\u54E6\u8036';

const ControlSteps = () => {
  const [current, setCurrent] = React.useState(0);
  return (
    <Steps
      current={current}
      onChange={(val) => {
        // eslint-disable-next-line no-console
        console.log('Change:', val);
        setCurrent(val);
      }}
      items={[
        {
          title: '\u5DF2\u5B8C\u6210',
        },
        {
          title: '\u8FDB\u884C\u4E2D',
        },
        {
          title: '\u5F85\u8FD0\u884C',
          description: 'Hello World!',
        },
        {
          title: '\u5F85\u8FD0\u884C',
        },
      ]}
    />
  );
};

export default () => (
  <div>
    <Steps
      current={1}
      items={[
        {
          title: '\u5DF2\u5B8C\u6210',
        },
        {
          title: '\u8FDB\u884C\u4E2D',
        },
        {
          title: '\u5F85\u8FD0\u884C',
        },
        {
          title: '\u5F85\u8FD0\u884C',
        },
      ]}
    />
    <Steps
      current={1}
      style={{ marginTop: 40 }}
      items={[
        {
          title: '\u5DF2\u5B8C\u6210',
          description,
        },
        {
          title: '\u8FDB\u884C\u4E2D',
          subTitle: '\u5269\u4F59 00:00:07',
          description,
        },
        {
          title: '\u5F85\u8FD0\u884C',
          description,
        },
        {
          title: '\u5F85\u8FD0\u884C',
          description,
        },
      ]}
    />
    <Steps
      current={1}
      style={{ marginTop: 40 }}
      status="error"
      items={[
        {
          title: '\u5DF2\u5B8C\u6210',
          description,
        },
        {
          title: '\u8FDB\u884C\u4E2D',
          subTitle: '\u5269\u4F59 00:00:07',
          description,
        },
        {
          title: '\u5F85\u8FD0\u884C',
          description,
        },
        {
          title: '\u5F85\u8FD0\u884C',
          description,
        },
      ]}
    />
    <ControlSteps />
  </div>
);
`},72578:function(a,n){"use strict";n.Z=`import '../../assets/index.less';
import '../../assets/iconfont.less';
import React from 'react';
import Steps from '@rc-component/steps';

// eslint-disable-next-line react/prop-types
const Icon = ({ type }) => <i className={\`rcicon rcicon-\${type}\`} />;

export default () => (
  <div>
    <Steps
      size="small"
      current={1}
      items={[
        {
          title: '\u5DF2\u5B8C\u6210',
        },
        {
          title: '\u8FDB\u884C\u4E2D',
        },
        {
          title: '\u5F85\u8FD0\u884C',
        },
        {
          title: '\u5F85\u8FD0\u884C',
        },
      ]}
    />
    <Steps
      size="small"
      current={1}
      style={{ marginTop: 40 }}
      items={[
        {
          title: '\u6B65\u9AA41',
        },
        {
          title: '\u6B65\u9AA42',
          icon: <Icon type="cloud" />,
        },
        {
          title: '\u6B65\u9AA43',
          icon: 'apple',
        },
        {
          title: '\u5F85\u8FD0\u884C',
          icon: 'github',
        },
      ]}
    />
  </div>
);
`},63042:function(a,n){"use strict";n.Z=`import '../../assets/index.less';
import '../../assets/iconfont.less';
import React from 'react';
import Steps from '@rc-component/steps';

function stepIcon({ status, node }) {
  const isProcessing = status === 'process';
  return isProcessing ? <div style={{ backgroundColor: 'blue' }}>{node}</div> : node;
}

export default () => {
  const [current, setCurrent] = React.useState(0);
  return (
    <>
      <button
        type="button"
        onClick={() => {
          setCurrent((current + 1) % 5);
        }}
      >
        loop
      </button>
      <Steps
        stepIcon={stepIcon}
        current={current}
        items={[
          {
            title: '\u5DF2\u5B8C\u6210',
          },
          {
            title: '\u8FDB\u884C\u4E2D',
          },
          {
            title: '\u5F85\u8FD0\u884C',
          },
          {
            title: '\u5F85\u8FD0\u884C',
          },
          {
            title: '\u5F85\u8FD0\u884C',
          },
        ]}
      />
    </>
  );
};
`},79053:function(a,n){"use strict";n.Z=`import '../../assets/index.less';
import '../../assets/iconfont.less';
import React from 'react';
import Steps from '@rc-component/steps';

const description =
  '\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A';

export default () => (
  <Steps
    direction="vertical"
    items={[
      {
        title: '\u5DF2\u5B8C\u6210',
        description,
      },
      {
        title: '\u8FDB\u884C\u4E2D',
        description,
      },
      {
        title: '\u5F85\u8FD0\u884C',
        description,
      },
      {
        title: '\u5F85\u8FD0\u884C',
        description,
      },
    ]}
  />
);
`},80532:function(a,n){"use strict";n.Z=`import '../../assets/index.less';
import '../../assets/iconfont.less';
import React from 'react';
import Steps from '@rc-component/steps';

const description =
  '\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A\u8FD9\u91CC\u662F\u591A\u4FE1\u606F\u7684\u63CF\u8FF0\u554A';

export default () => (
  <Steps
    direction="vertical"
    size="small"
    items={[
      {
        title: '\u5DF2\u5B8C\u6210',
        description,
      },
      {
        title: '\u8FDB\u884C\u4E2D',
        description,
      },
      {
        title: '\u5F85\u8FD0\u884C',
        description,
      },
      {
        title: '\u5F85\u8FD0\u884C',
        description,
      },
    ]}
  />
);
`},13769:function(a,n,e){var t=e(48541);function r(c,i){if(c==null)return{};var m=t(c,i),o,p;if(Object.getOwnPropertySymbols){var l=Object.getOwnPropertySymbols(c);for(p=0;p<l.length;p++)o=l[p],!(i.indexOf(o)>=0)&&Object.prototype.propertyIsEnumerable.call(c,o)&&(m[o]=c[o])}return m}a.exports=r,a.exports.__esModule=!0,a.exports.default=a.exports},48541:function(a){function n(e,t){if(e==null)return{};var r={},c=Object.keys(e),i,m;for(m=0;m<c.length;m++)i=c[m],!(t.indexOf(i)>=0)&&(r[i]=e[i]);return r}a.exports=n,a.exports.__esModule=!0,a.exports.default=a.exports}}]);
}());
