// @rc-component/tree-select/dist/meta__docs.dbf04b66.async.js
"use strict";(self.webpackChunk_rc_component_tree_select=self.webpackChunk_rc_component_tree_select||[]).push([[904],{86420:function(v,t,e){var c;e.r(t),e.d(t,{demos:function(){return G}});var o=e(7557),u=e.n(o),P=e(41498),g=e.n(P),r=e(96162),U=e(80206),E=e(55311),h=e(6701),l=e(8645),s=e(25240),a=e(63484),n=e(56253),_=e(27165),p=e(84670),D=e(28853),f=e(87643),i=e(83238),m=e(95094),M=e(99217),y=e(8399),I=e(47239),O=e(43647),C=e(89957),T=e(64803),j=e(76536),x=e(2971),S=e(41234),R=e(46077),A=e(19811),L=e(87992),V=e(72770),Z=e(43880),$=e(29213),J=e(64887),b=e(1732),d=e(26694),X=e(4216),G={"docs-demo-basic-demo-basic":{component:r.memo(r.lazy(function(){return e.e(433).then(e.bind(e,59558))})),asset:{type:"BLOCK",id:"docs-demo-basic-demo-basic",refAtomIds:[],dependencies:{"index.tsx":{type:"FILE",value:e(45745).Z},"rc-dialog":{type:"NPM",value:"9.6.0"},react:{type:"NPM",value:"16.14.0"},"./utils/dataUtil.ts":{type:"FILE",value:e(37756).Z},"../assets/index.less":{type:"FILE",value:e(25866).Z},"../src.tsx":{type:"FILE",value:e(42103).Z},"./utils/strategyUtil.ts":{type:"FILE",value:e(76742).Z},"./TreeSelect.tsx":{type:"FILE",value:e(51184).Z},"./TreeNode.tsx":{type:"FILE",value:e(41849).Z},"@rc-component/select":{type:"NPM",value:"1.0.0"},"./valueUtil.ts":{type:"FILE",value:e(4370).Z},"rc-tree":{type:"NPM",value:"5.13.1"},"@rc-component/util":{type:"NPM",value:"1.2.1"},"./utils/warningPropsUtil.ts":{type:"FILE",value:e(29162).Z},"./hooks/useCache.ts":{type:"FILE",value:e(95108).Z},"./hooks/useCheckedKeys.ts":{type:"FILE",value:e(7118).Z},"./LegacyContext.tsx":{type:"FILE",value:e(28458).Z},"./hooks/useTreeData.ts":{type:"FILE",value:e(4997).Z},"./hooks/useDataEntities.ts":{type:"FILE",value:e(31367).Z},"./hooks/useFilterTreeData.ts":{type:"FILE",value:e(67685).Z},"./hooks/useRefFunc.ts":{type:"FILE",value:e(60133).Z},"./TreeSelectContext.ts":{type:"FILE",value:e(29285).Z},"./OptionList.tsx":{type:"FILE",value:e(49597).Z},"./utils/legacyUtil.tsx":{type:"FILE",value:e(29035).Z}},entry:"index.tsx"},context:{"./utils/dataUtil.ts":l,"../assets/index.less":s,"../src.tsx":a,"./utils/strategyUtil.ts":n,"./TreeSelect.tsx":_,"./TreeNode.tsx":p,"./valueUtil.ts":f,"./utils/warningPropsUtil.ts":I,"./hooks/useCache.ts":O,"./hooks/useCheckedKeys.ts":C,"./LegacyContext.tsx":T,"./hooks/useTreeData.ts":j,"./hooks/useDataEntities.ts":x,"./hooks/useFilterTreeData.ts":S,"./hooks/useRefFunc.ts":R,"./TreeSelectContext.ts":A,"./OptionList.tsx":L,"./utils/legacyUtil.tsx":V,"rc-dialog":E,"rc-dialog/assets/index.css":h,react:c||(c=e.t(r,2)),"/Users/jilin/projects/antd/rc-tree-select/examples/utils/dataUtil.ts":l,"/Users/jilin/projects/antd/rc-tree-select/assets/index.less":s,"/Users/jilin/projects/antd/rc-tree-select/src/index.tsx":a,"/Users/jilin/projects/antd/rc-tree-select/src/utils/strategyUtil.ts":n,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelect.tsx":_,"/Users/jilin/projects/antd/rc-tree-select/src/TreeNode.tsx":p,"@rc-component/select":D,"/Users/jilin/projects/antd/rc-tree-select/src/utils/valueUtil.ts":f,"@rc-component/select/lib/hooks/useId":i,"rc-tree/lib/utils/conductUtil":m,"@rc-component/util/lib/hooks/useMergedState":M,"@rc-component/util/lib/warning":y,"/Users/jilin/projects/antd/rc-tree-select/src/utils/warningPropsUtil.ts":I,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCache.ts":O,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCheckedKeys.ts":C,"/Users/jilin/projects/antd/rc-tree-select/src/LegacyContext.tsx":T,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useTreeData.ts":j,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useDataEntities.ts":x,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useFilterTreeData.ts":S,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useRefFunc.ts":R,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelectContext.ts":A,"/Users/jilin/projects/antd/rc-tree-select/src/OptionList.tsx":L,"/Users/jilin/projects/antd/rc-tree-select/src/utils/legacyUtil.tsx":V,"rc-tree/lib/utils/treeUtil":Z,"@rc-component/util/lib/Children/toArray":$,"rc-tree":J,"@rc-component/util/lib/KeyCode":b,"@rc-component/util/lib/hooks/useMemo":d,"@rc-component/util":X},renderOpts:{compile:function(){var F=g()(u()().mark(function k(){var K,W=arguments;return u()().wrap(function(N){for(;;)switch(N.prev=N.next){case 0:return N.next=2,e.e(439).then(e.bind(e,56439));case 2:return N.abrupt("return",(K=N.sent).default.apply(K,W));case 3:case"end":return N.stop()}},k)}));function H(){return F.apply(this,arguments)}return H}()}}}},53443:function(v,t,e){var c;e.r(t),e.d(t,{demos:function(){return X}});var o=e(7557),u=e.n(o),P=e(41498),g=e.n(P),r=e(96162),U=e(2934),E=e(31458),h=e(25240),l=e(63484),s=e(8645),a=e(56253),n=e(27165),_=e(84670),p=e(28853),D=e(87643),f=e(83238),i=e(95094),m=e(99217),M=e(8399),y=e(2971),I=e(43647),O=e(89957),C=e(47239),T=e(76536),j=e(64803),x=e(87992),S=e(19811),R=e(72770),A=e(41234),L=e(46077),V=e(43880),Z=e(29213),$=e(64887),J=e(1732),b=e(26694),d=e(4216),X={"docs-demo-big-data-demo-big-data":{component:r.memo(r.lazy(function(){return e.e(433).then(e.bind(e,42681))})),asset:{type:"BLOCK",id:"docs-demo-big-data-demo-big-data",refAtomIds:[],dependencies:{"index.tsx":{type:"FILE",value:e(13371).Z},react:{type:"NPM",value:"16.14.0"},"./utils/big-data-generator.tsx":{type:"FILE",value:e(30390).Z},"../assets/index.less":{type:"FILE",value:e(25866).Z},"../src.tsx":{type:"FILE",value:e(42103).Z},"./dataUtil.ts":{type:"FILE",value:e(37756).Z},"./utils/strategyUtil.ts":{type:"FILE",value:e(76742).Z},"./TreeSelect.tsx":{type:"FILE",value:e(51184).Z},"./TreeNode.tsx":{type:"FILE",value:e(41849).Z},"@rc-component/select":{type:"NPM",value:"1.0.0"},"./valueUtil.ts":{type:"FILE",value:e(4370).Z},"rc-tree":{type:"NPM",value:"5.13.1"},"@rc-component/util":{type:"NPM",value:"1.2.1"},"./hooks/useDataEntities.ts":{type:"FILE",value:e(31367).Z},"./hooks/useCache.ts":{type:"FILE",value:e(95108).Z},"./hooks/useCheckedKeys.ts":{type:"FILE",value:e(7118).Z},"./utils/warningPropsUtil.ts":{type:"FILE",value:e(29162).Z},"./hooks/useTreeData.ts":{type:"FILE",value:e(4997).Z},"./LegacyContext.tsx":{type:"FILE",value:e(28458).Z},"./OptionList.tsx":{type:"FILE",value:e(49597).Z},"./TreeSelectContext.ts":{type:"FILE",value:e(29285).Z},"./utils/legacyUtil.tsx":{type:"FILE",value:e(29035).Z},"./hooks/useFilterTreeData.ts":{type:"FILE",value:e(67685).Z},"./hooks/useRefFunc.ts":{type:"FILE",value:e(60133).Z}},entry:"index.tsx"},context:{"./utils/big-data-generator.tsx":E,"../assets/index.less":h,"../src.tsx":l,"./dataUtil.ts":s,"./utils/strategyUtil.ts":a,"./TreeSelect.tsx":n,"./TreeNode.tsx":_,"./valueUtil.ts":D,"./hooks/useDataEntities.ts":y,"./hooks/useCache.ts":I,"./hooks/useCheckedKeys.ts":O,"./utils/warningPropsUtil.ts":C,"./hooks/useTreeData.ts":T,"./LegacyContext.tsx":j,"./OptionList.tsx":x,"./TreeSelectContext.ts":S,"./utils/legacyUtil.tsx":R,"./hooks/useFilterTreeData.ts":A,"./hooks/useRefFunc.ts":L,react:c||(c=e.t(r,2)),"/Users/jilin/projects/antd/rc-tree-select/examples/utils/big-data-generator.tsx":E,"/Users/jilin/projects/antd/rc-tree-select/assets/index.less":h,"/Users/jilin/projects/antd/rc-tree-select/src/index.tsx":l,"/Users/jilin/projects/antd/rc-tree-select/examples/utils/dataUtil.ts":s,"/Users/jilin/projects/antd/rc-tree-select/src/utils/strategyUtil.ts":a,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelect.tsx":n,"/Users/jilin/projects/antd/rc-tree-select/src/TreeNode.tsx":_,"@rc-component/select":p,"/Users/jilin/projects/antd/rc-tree-select/src/utils/valueUtil.ts":D,"@rc-component/select/lib/hooks/useId":f,"rc-tree/lib/utils/conductUtil":i,"@rc-component/util/lib/hooks/useMergedState":m,"@rc-component/util/lib/warning":M,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useDataEntities.ts":y,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCache.ts":I,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCheckedKeys.ts":O,"/Users/jilin/projects/antd/rc-tree-select/src/utils/warningPropsUtil.ts":C,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useTreeData.ts":T,"/Users/jilin/projects/antd/rc-tree-select/src/LegacyContext.tsx":j,"/Users/jilin/projects/antd/rc-tree-select/src/OptionList.tsx":x,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelectContext.ts":S,"/Users/jilin/projects/antd/rc-tree-select/src/utils/legacyUtil.tsx":R,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useFilterTreeData.ts":A,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useRefFunc.ts":L,"rc-tree/lib/utils/treeUtil":V,"@rc-component/util/lib/Children/toArray":Z,"rc-tree":$,"@rc-component/util/lib/KeyCode":J,"@rc-component/util/lib/hooks/useMemo":b,"@rc-component/util":d},renderOpts:{compile:function(){var G=g()(u()().mark(function H(){var k,K=arguments;return u()().wrap(function(le){for(;;)switch(le.prev=le.next){case 0:return le.next=2,e.e(439).then(e.bind(e,56439));case 2:return le.abrupt("return",(k=le.sent).default.apply(k,K));case 3:case"end":return le.stop()}},H)}));function F(){return G.apply(this,arguments)}return F}()}}}},43345:function(v,t,e){var c;e.r(t),e.d(t,{demos:function(){return d}});var o=e(7557),u=e.n(o),P=e(41498),g=e.n(P),r=e(96162),U=e(86795),E=e(6701),h=e(63484),l=e(25240),s=e(56253),a=e(27165),n=e(84670),_=e(28853),p=e(87643),D=e(83238),f=e(95094),i=e(99217),m=e(8399),M=e(47239),y=e(43647),I=e(89957),O=e(2971),C=e(41234),T=e(46077),j=e(76536),x=e(64803),S=e(87992),R=e(19811),A=e(72770),L=e(43880),V=e(29213),Z=e(64887),$=e(1732),J=e(26694),b=e(4216),d={"docs-demo-controlled-demo-controlled":{component:r.memo(r.lazy(function(){return e.e(433).then(e.bind(e,6190))})),asset:{type:"BLOCK",id:"docs-demo-controlled-demo-controlled",refAtomIds:[],dependencies:{"index.tsx":{type:"FILE",value:e(19760).Z},react:{type:"NPM",value:"16.14.0"},"rc-dialog":{type:"NPM",value:"9.6.0"},"../src.tsx":{type:"FILE",value:e(42103).Z},"../assets/index.less":{type:"FILE",value:e(25866).Z},"./utils/strategyUtil.ts":{type:"FILE",value:e(76742).Z},"./TreeSelect.tsx":{type:"FILE",value:e(51184).Z},"./TreeNode.tsx":{type:"FILE",value:e(41849).Z},"@rc-component/select":{type:"NPM",value:"1.0.0"},"./valueUtil.ts":{type:"FILE",value:e(4370).Z},"rc-tree":{type:"NPM",value:"5.13.1"},"@rc-component/util":{type:"NPM",value:"1.2.1"},"./utils/warningPropsUtil.ts":{type:"FILE",value:e(29162).Z},"./hooks/useCache.ts":{type:"FILE",value:e(95108).Z},"./hooks/useCheckedKeys.ts":{type:"FILE",value:e(7118).Z},"./hooks/useDataEntities.ts":{type:"FILE",value:e(31367).Z},"./hooks/useFilterTreeData.ts":{type:"FILE",value:e(67685).Z},"./hooks/useRefFunc.ts":{type:"FILE",value:e(60133).Z},"./hooks/useTreeData.ts":{type:"FILE",value:e(4997).Z},"./LegacyContext.tsx":{type:"FILE",value:e(28458).Z},"./OptionList.tsx":{type:"FILE",value:e(49597).Z},"./TreeSelectContext.ts":{type:"FILE",value:e(29285).Z},"./utils/legacyUtil.tsx":{type:"FILE",value:e(29035).Z}},entry:"index.tsx"},context:{"../src.tsx":h,"../assets/index.less":l,"./utils/strategyUtil.ts":s,"./TreeSelect.tsx":a,"./TreeNode.tsx":n,"./valueUtil.ts":p,"./utils/warningPropsUtil.ts":M,"./hooks/useCache.ts":y,"./hooks/useCheckedKeys.ts":I,"./hooks/useDataEntities.ts":O,"./hooks/useFilterTreeData.ts":C,"./hooks/useRefFunc.ts":T,"./hooks/useTreeData.ts":j,"./LegacyContext.tsx":x,"./OptionList.tsx":S,"./TreeSelectContext.ts":R,"./utils/legacyUtil.tsx":A,react:c||(c=e.t(r,2)),"rc-dialog/assets/index.css":E,"/Users/jilin/projects/antd/rc-tree-select/src/index.tsx":h,"/Users/jilin/projects/antd/rc-tree-select/assets/index.less":l,"/Users/jilin/projects/antd/rc-tree-select/src/utils/strategyUtil.ts":s,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelect.tsx":a,"/Users/jilin/projects/antd/rc-tree-select/src/TreeNode.tsx":n,"@rc-component/select":_,"/Users/jilin/projects/antd/rc-tree-select/src/utils/valueUtil.ts":p,"@rc-component/select/lib/hooks/useId":D,"rc-tree/lib/utils/conductUtil":f,"@rc-component/util/lib/hooks/useMergedState":i,"@rc-component/util/lib/warning":m,"/Users/jilin/projects/antd/rc-tree-select/src/utils/warningPropsUtil.ts":M,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCache.ts":y,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCheckedKeys.ts":I,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useDataEntities.ts":O,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useFilterTreeData.ts":C,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useRefFunc.ts":T,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useTreeData.ts":j,"/Users/jilin/projects/antd/rc-tree-select/src/LegacyContext.tsx":x,"/Users/jilin/projects/antd/rc-tree-select/src/OptionList.tsx":S,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelectContext.ts":R,"/Users/jilin/projects/antd/rc-tree-select/src/utils/legacyUtil.tsx":A,"rc-tree/lib/utils/treeUtil":L,"@rc-component/util/lib/Children/toArray":V,"rc-tree":Z,"@rc-component/util/lib/KeyCode":$,"@rc-component/util/lib/hooks/useMemo":J,"@rc-component/util":b},renderOpts:{compile:function(){var X=g()(u()().mark(function F(){var H,k=arguments;return u()().wrap(function(W){for(;;)switch(W.prev=W.next){case 0:return W.next=2,e.e(439).then(e.bind(e,56439));case 2:return W.abrupt("return",(H=W.sent).default.apply(H,k));case 3:case"end":return W.stop()}},F)}));function G(){return X.apply(this,arguments)}return G}()}}}},11002:function(v,t,e){var c;e.r(t),e.d(t,{demos:function(){return X}});var o=e(7557),u=e.n(o),P=e(41498),g=e.n(P),r=e(96162),U=e(32445),E=e(6701),h=e(8645),l=e(25240),s=e(63484),a=e(56253),n=e(27165),_=e(84670),p=e(28853),D=e(87643),f=e(83238),i=e(95094),m=e(99217),M=e(8399),y=e(47239),I=e(43647),O=e(89957),C=e(2971),T=e(41234),j=e(46077),x=e(76536),S=e(64803),R=e(87992),A=e(19811),L=e(72770),V=e(43880),Z=e(29213),$=e(64887),J=e(1732),b=e(26694),d=e(4216),X={"docs-demo-custom-icons-demo-custom-icons":{component:r.memo(r.lazy(function(){return e.e(433).then(e.bind(e,54472))})),asset:{type:"BLOCK",id:"docs-demo-custom-icons-demo-custom-icons",refAtomIds:[],dependencies:{"index.tsx":{type:"FILE",value:e(79898).Z},react:{type:"NPM",value:"16.14.0"},"rc-dialog":{type:"NPM",value:"9.6.0"},"./utils/dataUtil.ts":{type:"FILE",value:e(37756).Z},"../assets/index.less":{type:"FILE",value:e(25866).Z},"../src.tsx":{type:"FILE",value:e(42103).Z},"./utils/strategyUtil.ts":{type:"FILE",value:e(76742).Z},"./TreeSelect.tsx":{type:"FILE",value:e(51184).Z},"./TreeNode.tsx":{type:"FILE",value:e(41849).Z},"@rc-component/select":{type:"NPM",value:"1.0.0"},"./valueUtil.ts":{type:"FILE",value:e(4370).Z},"rc-tree":{type:"NPM",value:"5.13.1"},"@rc-component/util":{type:"NPM",value:"1.2.1"},"./utils/warningPropsUtil.ts":{type:"FILE",value:e(29162).Z},"./hooks/useCache.ts":{type:"FILE",value:e(95108).Z},"./hooks/useCheckedKeys.ts":{type:"FILE",value:e(7118).Z},"./hooks/useDataEntities.ts":{type:"FILE",value:e(31367).Z},"./hooks/useFilterTreeData.ts":{type:"FILE",value:e(67685).Z},"./hooks/useRefFunc.ts":{type:"FILE",value:e(60133).Z},"./hooks/useTreeData.ts":{type:"FILE",value:e(4997).Z},"./LegacyContext.tsx":{type:"FILE",value:e(28458).Z},"./OptionList.tsx":{type:"FILE",value:e(49597).Z},"./TreeSelectContext.ts":{type:"FILE",value:e(29285).Z},"./utils/legacyUtil.tsx":{type:"FILE",value:e(29035).Z}},entry:"index.tsx"},context:{"./utils/dataUtil.ts":h,"../assets/index.less":l,"../src.tsx":s,"./utils/strategyUtil.ts":a,"./TreeSelect.tsx":n,"./TreeNode.tsx":_,"./valueUtil.ts":D,"./utils/warningPropsUtil.ts":y,"./hooks/useCache.ts":I,"./hooks/useCheckedKeys.ts":O,"./hooks/useDataEntities.ts":C,"./hooks/useFilterTreeData.ts":T,"./hooks/useRefFunc.ts":j,"./hooks/useTreeData.ts":x,"./LegacyContext.tsx":S,"./OptionList.tsx":R,"./TreeSelectContext.ts":A,"./utils/legacyUtil.tsx":L,react:c||(c=e.t(r,2)),"rc-dialog/assets/index.css":E,"/Users/jilin/projects/antd/rc-tree-select/examples/utils/dataUtil.ts":h,"/Users/jilin/projects/antd/rc-tree-select/assets/index.less":l,"/Users/jilin/projects/antd/rc-tree-select/src/index.tsx":s,"/Users/jilin/projects/antd/rc-tree-select/src/utils/strategyUtil.ts":a,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelect.tsx":n,"/Users/jilin/projects/antd/rc-tree-select/src/TreeNode.tsx":_,"@rc-component/select":p,"/Users/jilin/projects/antd/rc-tree-select/src/utils/valueUtil.ts":D,"@rc-component/select/lib/hooks/useId":f,"rc-tree/lib/utils/conductUtil":i,"@rc-component/util/lib/hooks/useMergedState":m,"@rc-component/util/lib/warning":M,"/Users/jilin/projects/antd/rc-tree-select/src/utils/warningPropsUtil.ts":y,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCache.ts":I,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCheckedKeys.ts":O,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useDataEntities.ts":C,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useFilterTreeData.ts":T,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useRefFunc.ts":j,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useTreeData.ts":x,"/Users/jilin/projects/antd/rc-tree-select/src/LegacyContext.tsx":S,"/Users/jilin/projects/antd/rc-tree-select/src/OptionList.tsx":R,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelectContext.ts":A,"/Users/jilin/projects/antd/rc-tree-select/src/utils/legacyUtil.tsx":L,"rc-tree/lib/utils/treeUtil":V,"@rc-component/util/lib/Children/toArray":Z,"rc-tree":$,"@rc-component/util/lib/KeyCode":J,"@rc-component/util/lib/hooks/useMemo":b,"@rc-component/util":d},renderOpts:{compile:function(){var G=g()(u()().mark(function H(){var k,K=arguments;return u()().wrap(function(le){for(;;)switch(le.prev=le.next){case 0:return le.next=2,e.e(439).then(e.bind(e,56439));case 2:return le.abrupt("return",(k=le.sent).default.apply(k,K));case 3:case"end":return le.stop()}},H)}));function F(){return G.apply(this,arguments)}return F}()}}}},80743:function(v,t,e){var c;e.r(t),e.d(t,{demos:function(){return b}});var o=e(7557),u=e.n(o),P=e(41498),g=e.n(P),r=e(96162),U=e(55145),E=e(25240),h=e(63484),l=e(56253),s=e(27165),a=e(84670),n=e(28853),_=e(83238),p=e(87643),D=e(95094),f=e(99217),i=e(8399),m=e(47239),M=e(43647),y=e(89957),I=e(2971),O=e(41234),C=e(46077),T=e(76536),j=e(64803),x=e(87992),S=e(19811),R=e(72770),A=e(64887),L=e(29213),V=e(43880),Z=e(1732),$=e(26694),J=e(4216),b={"docs-demo-debug-demo-debug":{component:r.memo(r.lazy(function(){return e.e(433).then(e.bind(e,83939))})),asset:{type:"BLOCK",id:"docs-demo-debug-demo-debug",refAtomIds:[],dependencies:{"index.tsx":{type:"FILE",value:e(91938).Z},react:{type:"NPM",value:"16.14.0"},"../assets/index.less":{type:"FILE",value:e(25866).Z},"../src.tsx":{type:"FILE",value:e(42103).Z},"./utils/strategyUtil.ts":{type:"FILE",value:e(76742).Z},"./TreeSelect.tsx":{type:"FILE",value:e(51184).Z},"./TreeNode.tsx":{type:"FILE",value:e(41849).Z},"@rc-component/select":{type:"NPM",value:"1.0.0"},"./valueUtil.ts":{type:"FILE",value:e(4370).Z},"rc-tree":{type:"NPM",value:"5.13.1"},"@rc-component/util":{type:"NPM",value:"1.2.1"},"./utils/warningPropsUtil.ts":{type:"FILE",value:e(29162).Z},"./hooks/useCache.ts":{type:"FILE",value:e(95108).Z},"./hooks/useCheckedKeys.ts":{type:"FILE",value:e(7118).Z},"./hooks/useDataEntities.ts":{type:"FILE",value:e(31367).Z},"./hooks/useFilterTreeData.ts":{type:"FILE",value:e(67685).Z},"./hooks/useRefFunc.ts":{type:"FILE",value:e(60133).Z},"./hooks/useTreeData.ts":{type:"FILE",value:e(4997).Z},"./LegacyContext.tsx":{type:"FILE",value:e(28458).Z},"./OptionList.tsx":{type:"FILE",value:e(49597).Z},"./TreeSelectContext.ts":{type:"FILE",value:e(29285).Z},"./utils/legacyUtil.tsx":{type:"FILE",value:e(29035).Z}},entry:"index.tsx"},context:{"../assets/index.less":E,"../src.tsx":h,"./utils/strategyUtil.ts":l,"./TreeSelect.tsx":s,"./TreeNode.tsx":a,"./valueUtil.ts":p,"./utils/warningPropsUtil.ts":m,"./hooks/useCache.ts":M,"./hooks/useCheckedKeys.ts":y,"./hooks/useDataEntities.ts":I,"./hooks/useFilterTreeData.ts":O,"./hooks/useRefFunc.ts":C,"./hooks/useTreeData.ts":T,"./LegacyContext.tsx":j,"./OptionList.tsx":x,"./TreeSelectContext.ts":S,"./utils/legacyUtil.tsx":R,react:c||(c=e.t(r,2)),"/Users/jilin/projects/antd/rc-tree-select/assets/index.less":E,"/Users/jilin/projects/antd/rc-tree-select/src/index.tsx":h,"/Users/jilin/projects/antd/rc-tree-select/src/utils/strategyUtil.ts":l,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelect.tsx":s,"/Users/jilin/projects/antd/rc-tree-select/src/TreeNode.tsx":a,"@rc-component/select":n,"@rc-component/select/lib/hooks/useId":_,"/Users/jilin/projects/antd/rc-tree-select/src/utils/valueUtil.ts":p,"rc-tree/lib/utils/conductUtil":D,"@rc-component/util/lib/hooks/useMergedState":f,"@rc-component/util/lib/warning":i,"/Users/jilin/projects/antd/rc-tree-select/src/utils/warningPropsUtil.ts":m,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCache.ts":M,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCheckedKeys.ts":y,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useDataEntities.ts":I,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useFilterTreeData.ts":O,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useRefFunc.ts":C,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useTreeData.ts":T,"/Users/jilin/projects/antd/rc-tree-select/src/LegacyContext.tsx":j,"/Users/jilin/projects/antd/rc-tree-select/src/OptionList.tsx":x,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelectContext.ts":S,"/Users/jilin/projects/antd/rc-tree-select/src/utils/legacyUtil.tsx":R,"rc-tree":A,"@rc-component/util/lib/Children/toArray":L,"rc-tree/lib/utils/treeUtil":V,"@rc-component/util/lib/KeyCode":Z,"@rc-component/util/lib/hooks/useMemo":$,"@rc-component/util":J},renderOpts:{compile:function(){var d=g()(u()().mark(function G(){var F,H=arguments;return u()().wrap(function(K){for(;;)switch(K.prev=K.next){case 0:return K.next=2,e.e(439).then(e.bind(e,56439));case 2:return K.abrupt("return",(F=K.sent).default.apply(F,H));case 3:case"end":return K.stop()}},G)}));function X(){return d.apply(this,arguments)}return X}()}}}},1016:function(v,t,e){var c;e.r(t),e.d(t,{demos:function(){return b}});var o=e(7557),u=e.n(o),P=e(41498),g=e.n(P),r=e(96162),U=e(94785),E=e(63484),h=e(25240),l=e(27165),s=e(56253),a=e(84670),n=e(28853),_=e(83238),p=e(87643),D=e(95094),f=e(99217),i=e(8399),m=e(47239),M=e(43647),y=e(89957),I=e(2971),O=e(87992),C=e(19811),T=e(72770),j=e(46077),x=e(41234),S=e(76536),R=e(64803),A=e(64887),L=e(43880),V=e(29213),Z=e(1732),$=e(26694),J=e(4216),b={"docs-demo-disable-demo-disable":{component:r.memo(r.lazy(function(){return e.e(433).then(e.bind(e,38228))})),asset:{type:"BLOCK",id:"docs-demo-disable-demo-disable",refAtomIds:[],dependencies:{"index.tsx":{type:"FILE",value:e(73633).Z},react:{type:"NPM",value:"16.14.0"},"../src.tsx":{type:"FILE",value:e(42103).Z},"../assets/index.less":{type:"FILE",value:e(25866).Z},"./TreeSelect.tsx":{type:"FILE",value:e(51184).Z},"./utils/strategyUtil.ts":{type:"FILE",value:e(76742).Z},"./TreeNode.tsx":{type:"FILE",value:e(41849).Z},"@rc-component/select":{type:"NPM",value:"1.0.0"},"./valueUtil.ts":{type:"FILE",value:e(4370).Z},"rc-tree":{type:"NPM",value:"5.13.1"},"@rc-component/util":{type:"NPM",value:"1.2.1"},"./utils/warningPropsUtil.ts":{type:"FILE",value:e(29162).Z},"./hooks/useCache.ts":{type:"FILE",value:e(95108).Z},"./hooks/useCheckedKeys.ts":{type:"FILE",value:e(7118).Z},"./hooks/useDataEntities.ts":{type:"FILE",value:e(31367).Z},"./OptionList.tsx":{type:"FILE",value:e(49597).Z},"./TreeSelectContext.ts":{type:"FILE",value:e(29285).Z},"./utils/legacyUtil.tsx":{type:"FILE",value:e(29035).Z},"./hooks/useRefFunc.ts":{type:"FILE",value:e(60133).Z},"./hooks/useFilterTreeData.ts":{type:"FILE",value:e(67685).Z},"./hooks/useTreeData.ts":{type:"FILE",value:e(4997).Z},"./LegacyContext.tsx":{type:"FILE",value:e(28458).Z}},entry:"index.tsx"},context:{"../src.tsx":E,"../assets/index.less":h,"./TreeSelect.tsx":l,"./utils/strategyUtil.ts":s,"./TreeNode.tsx":a,"./valueUtil.ts":p,"./utils/warningPropsUtil.ts":m,"./hooks/useCache.ts":M,"./hooks/useCheckedKeys.ts":y,"./hooks/useDataEntities.ts":I,"./OptionList.tsx":O,"./TreeSelectContext.ts":C,"./utils/legacyUtil.tsx":T,"./hooks/useRefFunc.ts":j,"./hooks/useFilterTreeData.ts":x,"./hooks/useTreeData.ts":S,"./LegacyContext.tsx":R,react:c||(c=e.t(r,2)),"/Users/jilin/projects/antd/rc-tree-select/src/index.tsx":E,"/Users/jilin/projects/antd/rc-tree-select/assets/index.less":h,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelect.tsx":l,"/Users/jilin/projects/antd/rc-tree-select/src/utils/strategyUtil.ts":s,"/Users/jilin/projects/antd/rc-tree-select/src/TreeNode.tsx":a,"@rc-component/select":n,"@rc-component/select/lib/hooks/useId":_,"/Users/jilin/projects/antd/rc-tree-select/src/utils/valueUtil.ts":p,"rc-tree/lib/utils/conductUtil":D,"@rc-component/util/lib/hooks/useMergedState":f,"@rc-component/util/lib/warning":i,"/Users/jilin/projects/antd/rc-tree-select/src/utils/warningPropsUtil.ts":m,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCache.ts":M,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCheckedKeys.ts":y,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useDataEntities.ts":I,"/Users/jilin/projects/antd/rc-tree-select/src/OptionList.tsx":O,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelectContext.ts":C,"/Users/jilin/projects/antd/rc-tree-select/src/utils/legacyUtil.tsx":T,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useRefFunc.ts":j,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useFilterTreeData.ts":x,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useTreeData.ts":S,"/Users/jilin/projects/antd/rc-tree-select/src/LegacyContext.tsx":R,"rc-tree":A,"rc-tree/lib/utils/treeUtil":L,"@rc-component/util/lib/Children/toArray":V,"@rc-component/util/lib/KeyCode":Z,"@rc-component/util/lib/hooks/useMemo":$,"@rc-component/util":J},renderOpts:{compile:function(){var d=g()(u()().mark(function G(){var F,H=arguments;return u()().wrap(function(K){for(;;)switch(K.prev=K.next){case 0:return K.next=2,e.e(439).then(e.bind(e,56439));case 2:return K.abrupt("return",(F=K.sent).default.apply(F,H));case 3:case"end":return K.stop()}},G)}));function X(){return d.apply(this,arguments)}return X}()}}}},28156:function(v,t,e){var c;e.r(t),e.d(t,{demos:function(){return d}});var o=e(7557),u=e.n(o),P=e(41498),g=e.n(P),r=e(96162),U=e(547),E=e(25240),h=e(8645),l=e(63484),s=e(56253),a=e(27165),n=e(84670),_=e(28853),p=e(87643),D=e(83238),f=e(95094),i=e(99217),m=e(8399),M=e(46077),y=e(43647),I=e(47239),O=e(89957),C=e(2971),T=e(76536),j=e(41234),x=e(64803),S=e(87992),R=e(19811),A=e(72770),L=e(43880),V=e(64887),Z=e(29213),$=e(1732),J=e(26694),b=e(4216),d={"docs-demo-dynamic-demo-dynamic":{component:r.memo(r.lazy(function(){return e.e(433).then(e.bind(e,19890))})),asset:{type:"BLOCK",id:"docs-demo-dynamic-demo-dynamic",refAtomIds:[],dependencies:{"index.tsx":{type:"FILE",value:e(86272).Z},react:{type:"NPM",value:"16.14.0"},"../assets/index.less":{type:"FILE",value:e(25866).Z},"./utils/dataUtil.ts":{type:"FILE",value:e(37756).Z},"../src.tsx":{type:"FILE",value:e(42103).Z},"./utils/strategyUtil.ts":{type:"FILE",value:e(76742).Z},"./TreeSelect.tsx":{type:"FILE",value:e(51184).Z},"./TreeNode.tsx":{type:"FILE",value:e(41849).Z},"@rc-component/select":{type:"NPM",value:"1.0.0"},"./valueUtil.ts":{type:"FILE",value:e(4370).Z},"rc-tree":{type:"NPM",value:"5.13.1"},"@rc-component/util":{type:"NPM",value:"1.2.1"},"./hooks/useRefFunc.ts":{type:"FILE",value:e(60133).Z},"./hooks/useCache.ts":{type:"FILE",value:e(95108).Z},"./utils/warningPropsUtil.ts":{type:"FILE",value:e(29162).Z},"./hooks/useCheckedKeys.ts":{type:"FILE",value:e(7118).Z},"./hooks/useDataEntities.ts":{type:"FILE",value:e(31367).Z},"./hooks/useTreeData.ts":{type:"FILE",value:e(4997).Z},"./hooks/useFilterTreeData.ts":{type:"FILE",value:e(67685).Z},"./LegacyContext.tsx":{type:"FILE",value:e(28458).Z},"./OptionList.tsx":{type:"FILE",value:e(49597).Z},"./TreeSelectContext.ts":{type:"FILE",value:e(29285).Z},"./utils/legacyUtil.tsx":{type:"FILE",value:e(29035).Z}},entry:"index.tsx"},context:{"../assets/index.less":E,"./utils/dataUtil.ts":h,"../src.tsx":l,"./utils/strategyUtil.ts":s,"./TreeSelect.tsx":a,"./TreeNode.tsx":n,"./valueUtil.ts":p,"./hooks/useRefFunc.ts":M,"./hooks/useCache.ts":y,"./utils/warningPropsUtil.ts":I,"./hooks/useCheckedKeys.ts":O,"./hooks/useDataEntities.ts":C,"./hooks/useTreeData.ts":T,"./hooks/useFilterTreeData.ts":j,"./LegacyContext.tsx":x,"./OptionList.tsx":S,"./TreeSelectContext.ts":R,"./utils/legacyUtil.tsx":A,react:c||(c=e.t(r,2)),"/Users/jilin/projects/antd/rc-tree-select/assets/index.less":E,"/Users/jilin/projects/antd/rc-tree-select/examples/utils/dataUtil.ts":h,"/Users/jilin/projects/antd/rc-tree-select/src/index.tsx":l,"/Users/jilin/projects/antd/rc-tree-select/src/utils/strategyUtil.ts":s,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelect.tsx":a,"/Users/jilin/projects/antd/rc-tree-select/src/TreeNode.tsx":n,"@rc-component/select":_,"/Users/jilin/projects/antd/rc-tree-select/src/utils/valueUtil.ts":p,"@rc-component/select/lib/hooks/useId":D,"rc-tree/lib/utils/conductUtil":f,"@rc-component/util/lib/hooks/useMergedState":i,"@rc-component/util/lib/warning":m,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useRefFunc.ts":M,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCache.ts":y,"/Users/jilin/projects/antd/rc-tree-select/src/utils/warningPropsUtil.ts":I,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCheckedKeys.ts":O,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useDataEntities.ts":C,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useTreeData.ts":T,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useFilterTreeData.ts":j,"/Users/jilin/projects/antd/rc-tree-select/src/LegacyContext.tsx":x,"/Users/jilin/projects/antd/rc-tree-select/src/OptionList.tsx":S,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelectContext.ts":R,"/Users/jilin/projects/antd/rc-tree-select/src/utils/legacyUtil.tsx":A,"rc-tree/lib/utils/treeUtil":L,"rc-tree":V,"@rc-component/util/lib/Children/toArray":Z,"@rc-component/util/lib/KeyCode":$,"@rc-component/util/lib/hooks/useMemo":J,"@rc-component/util":b},renderOpts:{compile:function(){var X=g()(u()().mark(function F(){var H,k=arguments;return u()().wrap(function(W){for(;;)switch(W.prev=W.next){case 0:return W.next=2,e.e(439).then(e.bind(e,56439));case 2:return W.abrupt("return",(H=W.sent).default.apply(H,k));case 3:case"end":return W.stop()}},F)}));function G(){return X.apply(this,arguments)}return G}()}}}},22820:function(v,t,e){var c;e.r(t),e.d(t,{demos:function(){return d}});var o=e(7557),u=e.n(o),P=e(41498),g=e.n(P),r=e(96162),U=e(55230),E=e(6701),h=e(63484),l=e(25240),s=e(56253),a=e(84670),n=e(27165),_=e(28853),p=e(87643),D=e(83238),f=e(95094),i=e(99217),m=e(8399),M=e(47239),y=e(43647),I=e(89957),O=e(2971),C=e(64803),T=e(87992),j=e(41234),x=e(19811),S=e(72770),R=e(76536),A=e(46077),L=e(64887),V=e(43880),Z=e(29213),$=e(1732),J=e(26694),b=e(4216),d={"docs-demo-field-names-demo-fieldnames":{component:r.memo(r.lazy(function(){return e.e(433).then(e.bind(e,80981))})),asset:{type:"BLOCK",id:"docs-demo-field-names-demo-fieldnames",refAtomIds:[],dependencies:{"index.tsx":{type:"FILE",value:e(14610).Z},react:{type:"NPM",value:"16.14.0"},"rc-dialog":{type:"NPM",value:"9.6.0"},"../src.tsx":{type:"FILE",value:e(42103).Z},"../assets/index.less":{type:"FILE",value:e(25866).Z},"./utils/strategyUtil.ts":{type:"FILE",value:e(76742).Z},"./TreeNode.tsx":{type:"FILE",value:e(41849).Z},"./TreeSelect.tsx":{type:"FILE",value:e(51184).Z},"@rc-component/select":{type:"NPM",value:"1.0.0"},"./valueUtil.ts":{type:"FILE",value:e(4370).Z},"rc-tree":{type:"NPM",value:"5.13.1"},"@rc-component/util":{type:"NPM",value:"1.2.1"},"./utils/warningPropsUtil.ts":{type:"FILE",value:e(29162).Z},"./hooks/useCache.ts":{type:"FILE",value:e(95108).Z},"./hooks/useCheckedKeys.ts":{type:"FILE",value:e(7118).Z},"./hooks/useDataEntities.ts":{type:"FILE",value:e(31367).Z},"./LegacyContext.tsx":{type:"FILE",value:e(28458).Z},"./OptionList.tsx":{type:"FILE",value:e(49597).Z},"./hooks/useFilterTreeData.ts":{type:"FILE",value:e(67685).Z},"./TreeSelectContext.ts":{type:"FILE",value:e(29285).Z},"./utils/legacyUtil.tsx":{type:"FILE",value:e(29035).Z},"./hooks/useTreeData.ts":{type:"FILE",value:e(4997).Z},"./hooks/useRefFunc.ts":{type:"FILE",value:e(60133).Z}},entry:"index.tsx"},context:{"../src.tsx":h,"../assets/index.less":l,"./utils/strategyUtil.ts":s,"./TreeNode.tsx":a,"./TreeSelect.tsx":n,"./valueUtil.ts":p,"./utils/warningPropsUtil.ts":M,"./hooks/useCache.ts":y,"./hooks/useCheckedKeys.ts":I,"./hooks/useDataEntities.ts":O,"./LegacyContext.tsx":C,"./OptionList.tsx":T,"./hooks/useFilterTreeData.ts":j,"./TreeSelectContext.ts":x,"./utils/legacyUtil.tsx":S,"./hooks/useTreeData.ts":R,"./hooks/useRefFunc.ts":A,react:c||(c=e.t(r,2)),"rc-dialog/assets/index.css":E,"/Users/jilin/projects/antd/rc-tree-select/src/index.tsx":h,"/Users/jilin/projects/antd/rc-tree-select/assets/index.less":l,"/Users/jilin/projects/antd/rc-tree-select/src/utils/strategyUtil.ts":s,"/Users/jilin/projects/antd/rc-tree-select/src/TreeNode.tsx":a,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelect.tsx":n,"@rc-component/select":_,"/Users/jilin/projects/antd/rc-tree-select/src/utils/valueUtil.ts":p,"@rc-component/select/lib/hooks/useId":D,"rc-tree/lib/utils/conductUtil":f,"@rc-component/util/lib/hooks/useMergedState":i,"@rc-component/util/lib/warning":m,"/Users/jilin/projects/antd/rc-tree-select/src/utils/warningPropsUtil.ts":M,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCache.ts":y,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCheckedKeys.ts":I,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useDataEntities.ts":O,"/Users/jilin/projects/antd/rc-tree-select/src/LegacyContext.tsx":C,"/Users/jilin/projects/antd/rc-tree-select/src/OptionList.tsx":T,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useFilterTreeData.ts":j,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelectContext.ts":x,"/Users/jilin/projects/antd/rc-tree-select/src/utils/legacyUtil.tsx":S,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useTreeData.ts":R,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useRefFunc.ts":A,"rc-tree":L,"rc-tree/lib/utils/treeUtil":V,"@rc-component/util/lib/Children/toArray":Z,"@rc-component/util/lib/KeyCode":$,"@rc-component/util/lib/hooks/useMemo":J,"@rc-component/util":b},renderOpts:{compile:function(){var X=g()(u()().mark(function F(){var H,k=arguments;return u()().wrap(function(W){for(;;)switch(W.prev=W.next){case 0:return W.next=2,e.e(439).then(e.bind(e,56439));case 2:return W.abrupt("return",(H=W.sent).default.apply(H,k));case 3:case"end":return W.stop()}},F)}));function G(){return X.apply(this,arguments)}return G}()}}}},74242:function(v,t,e){var c;e.r(t),e.d(t,{demos:function(){return d}});var o=e(7557),u=e.n(o),P=e(41498),g=e.n(P),r=e(96162),U=e(63166),E=e(25240),h=e(8645),l=e(63484),s=e(27165),a=e(56253),n=e(84670),_=e(28853),p=e(87643),D=e(83238),f=e(95094),i=e(99217),m=e(8399),M=e(47239),y=e(43647),I=e(89957),O=e(2971),C=e(64803),T=e(41234),j=e(46077),x=e(76536),S=e(19811),R=e(87992),A=e(72770),L=e(43880),V=e(64887),Z=e(29213),$=e(1732),J=e(26694),b=e(4216),d={"docs-demo-filter-demo-filter":{component:r.memo(r.lazy(function(){return e.e(433).then(e.bind(e,76595))})),asset:{type:"BLOCK",id:"docs-demo-filter-demo-filter",refAtomIds:[],dependencies:{"index.tsx":{type:"FILE",value:e(33073).Z},react:{type:"NPM",value:"16.14.0"},"../assets/index.less":{type:"FILE",value:e(25866).Z},"./utils/dataUtil.ts":{type:"FILE",value:e(37756).Z},"../src.tsx":{type:"FILE",value:e(42103).Z},"./TreeSelect.tsx":{type:"FILE",value:e(51184).Z},"./utils/strategyUtil.ts":{type:"FILE",value:e(76742).Z},"./TreeNode.tsx":{type:"FILE",value:e(41849).Z},"@rc-component/select":{type:"NPM",value:"1.0.0"},"./valueUtil.ts":{type:"FILE",value:e(4370).Z},"rc-tree":{type:"NPM",value:"5.13.1"},"@rc-component/util":{type:"NPM",value:"1.2.1"},"./utils/warningPropsUtil.ts":{type:"FILE",value:e(29162).Z},"./hooks/useCache.ts":{type:"FILE",value:e(95108).Z},"./hooks/useCheckedKeys.ts":{type:"FILE",value:e(7118).Z},"./hooks/useDataEntities.ts":{type:"FILE",value:e(31367).Z},"./LegacyContext.tsx":{type:"FILE",value:e(28458).Z},"./hooks/useFilterTreeData.ts":{type:"FILE",value:e(67685).Z},"./hooks/useRefFunc.ts":{type:"FILE",value:e(60133).Z},"./hooks/useTreeData.ts":{type:"FILE",value:e(4997).Z},"./TreeSelectContext.ts":{type:"FILE",value:e(29285).Z},"./OptionList.tsx":{type:"FILE",value:e(49597).Z},"./utils/legacyUtil.tsx":{type:"FILE",value:e(29035).Z}},entry:"index.tsx"},context:{"../assets/index.less":E,"./utils/dataUtil.ts":h,"../src.tsx":l,"./TreeSelect.tsx":s,"./utils/strategyUtil.ts":a,"./TreeNode.tsx":n,"./valueUtil.ts":p,"./utils/warningPropsUtil.ts":M,"./hooks/useCache.ts":y,"./hooks/useCheckedKeys.ts":I,"./hooks/useDataEntities.ts":O,"./LegacyContext.tsx":C,"./hooks/useFilterTreeData.ts":T,"./hooks/useRefFunc.ts":j,"./hooks/useTreeData.ts":x,"./TreeSelectContext.ts":S,"./OptionList.tsx":R,"./utils/legacyUtil.tsx":A,react:c||(c=e.t(r,2)),"/Users/jilin/projects/antd/rc-tree-select/assets/index.less":E,"/Users/jilin/projects/antd/rc-tree-select/examples/utils/dataUtil.ts":h,"/Users/jilin/projects/antd/rc-tree-select/src/index.tsx":l,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelect.tsx":s,"/Users/jilin/projects/antd/rc-tree-select/src/utils/strategyUtil.ts":a,"/Users/jilin/projects/antd/rc-tree-select/src/TreeNode.tsx":n,"@rc-component/select":_,"/Users/jilin/projects/antd/rc-tree-select/src/utils/valueUtil.ts":p,"@rc-component/select/lib/hooks/useId":D,"rc-tree/lib/utils/conductUtil":f,"@rc-component/util/lib/hooks/useMergedState":i,"@rc-component/util/lib/warning":m,"/Users/jilin/projects/antd/rc-tree-select/src/utils/warningPropsUtil.ts":M,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCache.ts":y,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCheckedKeys.ts":I,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useDataEntities.ts":O,"/Users/jilin/projects/antd/rc-tree-select/src/LegacyContext.tsx":C,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useFilterTreeData.ts":T,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useRefFunc.ts":j,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useTreeData.ts":x,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelectContext.ts":S,"/Users/jilin/projects/antd/rc-tree-select/src/OptionList.tsx":R,"/Users/jilin/projects/antd/rc-tree-select/src/utils/legacyUtil.tsx":A,"rc-tree/lib/utils/treeUtil":L,"rc-tree":V,"@rc-component/util/lib/Children/toArray":Z,"@rc-component/util/lib/KeyCode":$,"@rc-component/util/lib/hooks/useMemo":J,"@rc-component/util":b},renderOpts:{compile:function(){var X=g()(u()().mark(function F(){var H,k=arguments;return u()().wrap(function(W){for(;;)switch(W.prev=W.next){case 0:return W.next=2,e.e(439).then(e.bind(e,56439));case 2:return W.abrupt("return",(H=W.sent).default.apply(H,k));case 3:case"end":return W.stop()}},F)}));function G(){return X.apply(this,arguments)}return G}()}}}},16748:function(v,t,e){var c;e.r(t),e.d(t,{demos:function(){return G}});var o=e(7557),u=e.n(o),P=e(41498),g=e.n(P),r=e(96162),U=e(29272),E=e(28853),h=e(7866),l=e(70624),s=e(8645),a=e(63484),n=e(25240),_=e(56253),p=e(27165),D=e(84670),f=e(87643),i=e(83238),m=e(95094),M=e(99217),y=e(8399),I=e(47239),O=e(43647),C=e(89957),T=e(64803),j=e(76536),x=e(87992),S=e(2971),R=e(19811),A=e(72770),L=e(41234),V=e(46077),Z=e(43880),$=e(64887),J=e(29213),b=e(1732),d=e(26694),X=e(4216),G={"docs-demo-form-demo-form":{component:r.memo(r.lazy(function(){return e.e(433).then(e.bind(e,23576))})),asset:{type:"BLOCK",id:"docs-demo-form-demo-form",refAtomIds:[],dependencies:{"index.tsx":{type:"FILE",value:e(43335).Z},react:{type:"NPM",value:"16.14.0"},"@rc-component/select":{type:"NPM",value:"1.0.0"},"rc-field-form":{type:"NPM",value:"2.7.0"},"./utils/dataUtil.ts":{type:"FILE",value:e(37756).Z},"../src.tsx":{type:"FILE",value:e(42103).Z},"../assets/index.less":{type:"FILE",value:e(25866).Z},"./utils/strategyUtil.ts":{type:"FILE",value:e(76742).Z},"./TreeSelect.tsx":{type:"FILE",value:e(51184).Z},"./TreeNode.tsx":{type:"FILE",value:e(41849).Z},"./valueUtil.ts":{type:"FILE",value:e(4370).Z},"rc-tree":{type:"NPM",value:"5.13.1"},"@rc-component/util":{type:"NPM",value:"1.2.1"},"./utils/warningPropsUtil.ts":{type:"FILE",value:e(29162).Z},"./hooks/useCache.ts":{type:"FILE",value:e(95108).Z},"./hooks/useCheckedKeys.ts":{type:"FILE",value:e(7118).Z},"./LegacyContext.tsx":{type:"FILE",value:e(28458).Z},"./hooks/useTreeData.ts":{type:"FILE",value:e(4997).Z},"./OptionList.tsx":{type:"FILE",value:e(49597).Z},"./hooks/useDataEntities.ts":{type:"FILE",value:e(31367).Z},"./TreeSelectContext.ts":{type:"FILE",value:e(29285).Z},"./utils/legacyUtil.tsx":{type:"FILE",value:e(29035).Z},"./hooks/useFilterTreeData.ts":{type:"FILE",value:e(67685).Z},"./hooks/useRefFunc.ts":{type:"FILE",value:e(60133).Z}},entry:"index.tsx"},context:{"./utils/dataUtil.ts":s,"../src.tsx":a,"../assets/index.less":n,"./utils/strategyUtil.ts":_,"./TreeSelect.tsx":p,"./TreeNode.tsx":D,"./valueUtil.ts":f,"./utils/warningPropsUtil.ts":I,"./hooks/useCache.ts":O,"./hooks/useCheckedKeys.ts":C,"./LegacyContext.tsx":T,"./hooks/useTreeData.ts":j,"./OptionList.tsx":x,"./hooks/useDataEntities.ts":S,"./TreeSelectContext.ts":R,"./utils/legacyUtil.tsx":A,"./hooks/useFilterTreeData.ts":L,"./hooks/useRefFunc.ts":V,react:c||(c=e.t(r,2)),"@rc-component/select":E,"rc-field-form":h,"@rc-component/select/assets/index.less":l,"/Users/jilin/projects/antd/rc-tree-select/examples/utils/dataUtil.ts":s,"/Users/jilin/projects/antd/rc-tree-select/src/index.tsx":a,"/Users/jilin/projects/antd/rc-tree-select/assets/index.less":n,"/Users/jilin/projects/antd/rc-tree-select/src/utils/strategyUtil.ts":_,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelect.tsx":p,"/Users/jilin/projects/antd/rc-tree-select/src/TreeNode.tsx":D,"/Users/jilin/projects/antd/rc-tree-select/src/utils/valueUtil.ts":f,"@rc-component/select/lib/hooks/useId":i,"rc-tree/lib/utils/conductUtil":m,"@rc-component/util/lib/hooks/useMergedState":M,"@rc-component/util/lib/warning":y,"/Users/jilin/projects/antd/rc-tree-select/src/utils/warningPropsUtil.ts":I,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCache.ts":O,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCheckedKeys.ts":C,"/Users/jilin/projects/antd/rc-tree-select/src/LegacyContext.tsx":T,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useTreeData.ts":j,"/Users/jilin/projects/antd/rc-tree-select/src/OptionList.tsx":x,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useDataEntities.ts":S,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelectContext.ts":R,"/Users/jilin/projects/antd/rc-tree-select/src/utils/legacyUtil.tsx":A,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useFilterTreeData.ts":L,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useRefFunc.ts":V,"rc-tree/lib/utils/treeUtil":Z,"rc-tree":$,"@rc-component/util/lib/Children/toArray":J,"@rc-component/util/lib/KeyCode":b,"@rc-component/util/lib/hooks/useMemo":d,"@rc-component/util":X},renderOpts:{compile:function(){var F=g()(u()().mark(function k(){var K,W=arguments;return u()().wrap(function(N){for(;;)switch(N.prev=N.next){case 0:return N.next=2,e.e(439).then(e.bind(e,56439));case 2:return N.abrupt("return",(K=N.sent).default.apply(K,W));case 3:case"end":return N.stop()}},k)}));function H(){return F.apply(this,arguments)}return H}()}}}},58077:function(v,t,e){var c;e.r(t),e.d(t,{demos:function(){return J}});var o=e(7557),u=e.n(o),P=e(41498),g=e.n(P),r=e(96162),U=e(20189),E=e(63484),h=e(27165),l=e(56253),s=e(84670),a=e(28853),n=e(87643),_=e(83238),p=e(95094),D=e(99217),f=e(8399),i=e(47239),m=e(43647),M=e(89957),y=e(2971),I=e(41234),O=e(46077),C=e(76536),T=e(64803),j=e(87992),x=e(19811),S=e(72770),R=e(64887),A=e(43880),L=e(29213),V=e(1732),Z=e(26694),$=e(4216),J={"docs-demo-mutiple-with-max-count-demo-mutiple-with-maxcount":{component:r.memo(r.lazy(function(){return e.e(433).then(e.bind(e,54129))})),asset:{type:"BLOCK",id:"docs-demo-mutiple-with-max-count-demo-mutiple-with-maxcount",refAtomIds:[],dependencies:{"index.tsx":{type:"FILE",value:e(67959).Z},react:{type:"NPM",value:"16.14.0"},"../src.tsx":{type:"FILE",value:e(42103).Z},"./TreeSelect.tsx":{type:"FILE",value:e(51184).Z},"./utils/strategyUtil.ts":{type:"FILE",value:e(76742).Z},"./TreeNode.tsx":{type:"FILE",value:e(41849).Z},"@rc-component/select":{type:"NPM",value:"1.0.0"},"./valueUtil.ts":{type:"FILE",value:e(4370).Z},"rc-tree":{type:"NPM",value:"5.13.1"},"@rc-component/util":{type:"NPM",value:"1.2.1"},"./utils/warningPropsUtil.ts":{type:"FILE",value:e(29162).Z},"./hooks/useCache.ts":{type:"FILE",value:e(95108).Z},"./hooks/useCheckedKeys.ts":{type:"FILE",value:e(7118).Z},"./hooks/useDataEntities.ts":{type:"FILE",value:e(31367).Z},"./hooks/useFilterTreeData.ts":{type:"FILE",value:e(67685).Z},"./hooks/useRefFunc.ts":{type:"FILE",value:e(60133).Z},"./hooks/useTreeData.ts":{type:"FILE",value:e(4997).Z},"./LegacyContext.tsx":{type:"FILE",value:e(28458).Z},"./OptionList.tsx":{type:"FILE",value:e(49597).Z},"./TreeSelectContext.ts":{type:"FILE",value:e(29285).Z},"./utils/legacyUtil.tsx":{type:"FILE",value:e(29035).Z}},entry:"index.tsx"},context:{"../src.tsx":E,"./TreeSelect.tsx":h,"./utils/strategyUtil.ts":l,"./TreeNode.tsx":s,"./valueUtil.ts":n,"./utils/warningPropsUtil.ts":i,"./hooks/useCache.ts":m,"./hooks/useCheckedKeys.ts":M,"./hooks/useDataEntities.ts":y,"./hooks/useFilterTreeData.ts":I,"./hooks/useRefFunc.ts":O,"./hooks/useTreeData.ts":C,"./LegacyContext.tsx":T,"./OptionList.tsx":j,"./TreeSelectContext.ts":x,"./utils/legacyUtil.tsx":S,react:c||(c=e.t(r,2)),"/Users/jilin/projects/antd/rc-tree-select/src/index.tsx":E,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelect.tsx":h,"/Users/jilin/projects/antd/rc-tree-select/src/utils/strategyUtil.ts":l,"/Users/jilin/projects/antd/rc-tree-select/src/TreeNode.tsx":s,"@rc-component/select":a,"/Users/jilin/projects/antd/rc-tree-select/src/utils/valueUtil.ts":n,"@rc-component/select/lib/hooks/useId":_,"rc-tree/lib/utils/conductUtil":p,"@rc-component/util/lib/hooks/useMergedState":D,"@rc-component/util/lib/warning":f,"/Users/jilin/projects/antd/rc-tree-select/src/utils/warningPropsUtil.ts":i,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCache.ts":m,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCheckedKeys.ts":M,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useDataEntities.ts":y,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useFilterTreeData.ts":I,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useRefFunc.ts":O,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useTreeData.ts":C,"/Users/jilin/projects/antd/rc-tree-select/src/LegacyContext.tsx":T,"/Users/jilin/projects/antd/rc-tree-select/src/OptionList.tsx":j,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelectContext.ts":x,"/Users/jilin/projects/antd/rc-tree-select/src/utils/legacyUtil.tsx":S,"rc-tree":R,"rc-tree/lib/utils/treeUtil":A,"@rc-component/util/lib/Children/toArray":L,"@rc-component/util/lib/KeyCode":V,"@rc-component/util/lib/hooks/useMemo":Z,"@rc-component/util":$},renderOpts:{compile:function(){var b=g()(u()().mark(function X(){var G,F=arguments;return u()().wrap(function(k){for(;;)switch(k.prev=k.next){case 0:return k.next=2,e.e(439).then(e.bind(e,56439));case 2:return k.abrupt("return",(G=k.sent).default.apply(G,F));case 3:case"end":return k.stop()}},X)}));function d(){return b.apply(this,arguments)}return d}()}}}},15371:function(v,t,e){var c;e.r(t),e.d(t,{demos:function(){return b}});var o=e(7557),u=e.n(o),P=e(41498),g=e.n(P),r=e(96162),U=e(67928),E=e(25240),h=e(63484),l=e(56253),s=e(27165),a=e(84670),n=e(28853),_=e(87643),p=e(83238),D=e(95094),f=e(99217),i=e(8399),m=e(47239),M=e(46077),y=e(76536),I=e(64803),O=e(87992),C=e(43647),T=e(19811),j=e(89957),x=e(72770),S=e(2971),R=e(41234),A=e(64887),L=e(29213),V=e(43880),Z=e(1732),$=e(26694),J=e(4216),b={"docs-demo-tree-node-label-prop-demo-treenodelabelprop":{component:r.memo(r.lazy(function(){return e.e(433).then(e.bind(e,89658))})),asset:{type:"BLOCK",id:"docs-demo-tree-node-label-prop-demo-treenodelabelprop",refAtomIds:[],dependencies:{"index.tsx":{type:"FILE",value:e(73007).Z},react:{type:"NPM",value:"16.14.0"},"../assets/index.less":{type:"FILE",value:e(25866).Z},"../src.tsx":{type:"FILE",value:e(42103).Z},"./utils/strategyUtil.ts":{type:"FILE",value:e(76742).Z},"./TreeSelect.tsx":{type:"FILE",value:e(51184).Z},"./TreeNode.tsx":{type:"FILE",value:e(41849).Z},"@rc-component/select":{type:"NPM",value:"1.0.0"},"./valueUtil.ts":{type:"FILE",value:e(4370).Z},"rc-tree":{type:"NPM",value:"5.13.1"},"@rc-component/util":{type:"NPM",value:"1.2.1"},"./utils/warningPropsUtil.ts":{type:"FILE",value:e(29162).Z},"./hooks/useRefFunc.ts":{type:"FILE",value:e(60133).Z},"./hooks/useTreeData.ts":{type:"FILE",value:e(4997).Z},"./LegacyContext.tsx":{type:"FILE",value:e(28458).Z},"./OptionList.tsx":{type:"FILE",value:e(49597).Z},"./hooks/useCache.ts":{type:"FILE",value:e(95108).Z},"./TreeSelectContext.ts":{type:"FILE",value:e(29285).Z},"./hooks/useCheckedKeys.ts":{type:"FILE",value:e(7118).Z},"./utils/legacyUtil.tsx":{type:"FILE",value:e(29035).Z},"./hooks/useDataEntities.ts":{type:"FILE",value:e(31367).Z},"./hooks/useFilterTreeData.ts":{type:"FILE",value:e(67685).Z}},entry:"index.tsx"},context:{"../assets/index.less":E,"../src.tsx":h,"./utils/strategyUtil.ts":l,"./TreeSelect.tsx":s,"./TreeNode.tsx":a,"./valueUtil.ts":_,"./utils/warningPropsUtil.ts":m,"./hooks/useRefFunc.ts":M,"./hooks/useTreeData.ts":y,"./LegacyContext.tsx":I,"./OptionList.tsx":O,"./hooks/useCache.ts":C,"./TreeSelectContext.ts":T,"./hooks/useCheckedKeys.ts":j,"./utils/legacyUtil.tsx":x,"./hooks/useDataEntities.ts":S,"./hooks/useFilterTreeData.ts":R,react:c||(c=e.t(r,2)),"/Users/jilin/projects/antd/rc-tree-select/assets/index.less":E,"/Users/jilin/projects/antd/rc-tree-select/src/index.tsx":h,"/Users/jilin/projects/antd/rc-tree-select/src/utils/strategyUtil.ts":l,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelect.tsx":s,"/Users/jilin/projects/antd/rc-tree-select/src/TreeNode.tsx":a,"@rc-component/select":n,"/Users/jilin/projects/antd/rc-tree-select/src/utils/valueUtil.ts":_,"@rc-component/select/lib/hooks/useId":p,"rc-tree/lib/utils/conductUtil":D,"@rc-component/util/lib/hooks/useMergedState":f,"@rc-component/util/lib/warning":i,"/Users/jilin/projects/antd/rc-tree-select/src/utils/warningPropsUtil.ts":m,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useRefFunc.ts":M,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useTreeData.ts":y,"/Users/jilin/projects/antd/rc-tree-select/src/LegacyContext.tsx":I,"/Users/jilin/projects/antd/rc-tree-select/src/OptionList.tsx":O,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCache.ts":C,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelectContext.ts":T,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCheckedKeys.ts":j,"/Users/jilin/projects/antd/rc-tree-select/src/utils/legacyUtil.tsx":x,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useDataEntities.ts":S,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useFilterTreeData.ts":R,"rc-tree":A,"@rc-component/util/lib/Children/toArray":L,"rc-tree/lib/utils/treeUtil":V,"@rc-component/util/lib/KeyCode":Z,"@rc-component/util/lib/hooks/useMemo":$,"@rc-component/util":J},renderOpts:{compile:function(){var d=g()(u()().mark(function G(){var F,H=arguments;return u()().wrap(function(K){for(;;)switch(K.prev=K.next){case 0:return K.next=2,e.e(439).then(e.bind(e,56439));case 2:return K.abrupt("return",(F=K.sent).default.apply(F,H));case 3:case"end":return K.stop()}},G)}));function X(){return d.apply(this,arguments)}return X}()}}}},43072:function(v,t,e){var c;e.r(t),e.d(t,{demos:function(){return b}});var o=e(7557),u=e.n(o),P=e(41498),g=e.n(P),r=e(96162),U=e(15316),E=e(63484),h=e(25240),l=e(27165),s=e(56253),a=e(84670),n=e(28853),_=e(87643),p=e(83238),D=e(95094),f=e(99217),i=e(8399),m=e(47239),M=e(43647),y=e(89957),I=e(2971),O=e(41234),C=e(46077),T=e(76536),j=e(64803),x=e(87992),S=e(19811),R=e(72770),A=e(43880),L=e(29213),V=e(64887),Z=e(1732),$=e(26694),J=e(4216),b={"docs-demo-width-demo-width":{component:r.memo(r.lazy(function(){return e.e(433).then(e.bind(e,36112))})),asset:{type:"BLOCK",id:"docs-demo-width-demo-width",refAtomIds:[],dependencies:{"index.tsx":{type:"FILE",value:e(33149).Z},react:{type:"NPM",value:"16.14.0"},"../src.tsx":{type:"FILE",value:e(42103).Z},"../assets/index.less":{type:"FILE",value:e(25866).Z},"./TreeSelect.tsx":{type:"FILE",value:e(51184).Z},"./utils/strategyUtil.ts":{type:"FILE",value:e(76742).Z},"./TreeNode.tsx":{type:"FILE",value:e(41849).Z},"@rc-component/select":{type:"NPM",value:"1.0.0"},"./valueUtil.ts":{type:"FILE",value:e(4370).Z},"rc-tree":{type:"NPM",value:"5.13.1"},"@rc-component/util":{type:"NPM",value:"1.2.1"},"./utils/warningPropsUtil.ts":{type:"FILE",value:e(29162).Z},"./hooks/useCache.ts":{type:"FILE",value:e(95108).Z},"./hooks/useCheckedKeys.ts":{type:"FILE",value:e(7118).Z},"./hooks/useDataEntities.ts":{type:"FILE",value:e(31367).Z},"./hooks/useFilterTreeData.ts":{type:"FILE",value:e(67685).Z},"./hooks/useRefFunc.ts":{type:"FILE",value:e(60133).Z},"./hooks/useTreeData.ts":{type:"FILE",value:e(4997).Z},"./LegacyContext.tsx":{type:"FILE",value:e(28458).Z},"./OptionList.tsx":{type:"FILE",value:e(49597).Z},"./TreeSelectContext.ts":{type:"FILE",value:e(29285).Z},"./utils/legacyUtil.tsx":{type:"FILE",value:e(29035).Z}},entry:"index.tsx"},context:{"../src.tsx":E,"../assets/index.less":h,"./TreeSelect.tsx":l,"./utils/strategyUtil.ts":s,"./TreeNode.tsx":a,"./valueUtil.ts":_,"./utils/warningPropsUtil.ts":m,"./hooks/useCache.ts":M,"./hooks/useCheckedKeys.ts":y,"./hooks/useDataEntities.ts":I,"./hooks/useFilterTreeData.ts":O,"./hooks/useRefFunc.ts":C,"./hooks/useTreeData.ts":T,"./LegacyContext.tsx":j,"./OptionList.tsx":x,"./TreeSelectContext.ts":S,"./utils/legacyUtil.tsx":R,react:c||(c=e.t(r,2)),"/Users/jilin/projects/antd/rc-tree-select/src/index.tsx":E,"/Users/jilin/projects/antd/rc-tree-select/assets/index.less":h,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelect.tsx":l,"/Users/jilin/projects/antd/rc-tree-select/src/utils/strategyUtil.ts":s,"/Users/jilin/projects/antd/rc-tree-select/src/TreeNode.tsx":a,"@rc-component/select":n,"/Users/jilin/projects/antd/rc-tree-select/src/utils/valueUtil.ts":_,"@rc-component/select/lib/hooks/useId":p,"rc-tree/lib/utils/conductUtil":D,"@rc-component/util/lib/hooks/useMergedState":f,"@rc-component/util/lib/warning":i,"/Users/jilin/projects/antd/rc-tree-select/src/utils/warningPropsUtil.ts":m,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCache.ts":M,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useCheckedKeys.ts":y,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useDataEntities.ts":I,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useFilterTreeData.ts":O,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useRefFunc.ts":C,"/Users/jilin/projects/antd/rc-tree-select/src/hooks/useTreeData.ts":T,"/Users/jilin/projects/antd/rc-tree-select/src/LegacyContext.tsx":j,"/Users/jilin/projects/antd/rc-tree-select/src/OptionList.tsx":x,"/Users/jilin/projects/antd/rc-tree-select/src/TreeSelectContext.ts":S,"/Users/jilin/projects/antd/rc-tree-select/src/utils/legacyUtil.tsx":R,"rc-tree/lib/utils/treeUtil":A,"@rc-component/util/lib/Children/toArray":L,"rc-tree":V,"@rc-component/util/lib/KeyCode":Z,"@rc-component/util/lib/hooks/useMemo":$,"@rc-component/util":J},renderOpts:{compile:function(){var d=g()(u()().mark(function G(){var F,H=arguments;return u()().wrap(function(K){for(;;)switch(K.prev=K.next){case 0:return K.next=2,e.e(439).then(e.bind(e,56439));case 2:return K.abrupt("return",(F=K.sent).default.apply(F,H));case 3:case"end":return K.stop()}},G)}));function X(){return d.apply(this,arguments)}return X}()}}}},81059:function(v,t,e){e.r(t),e.d(t,{demos:function(){return u}});var c=e(96162),o=e(8210),u={}},31458:function(v,t,e){e.r(t);var c=e(82100),o=e.n(c),u=e(29186),P=e.n(u),g=e(13720),r=e.n(g),U=e(80619),E=e.n(U),h=e(47074),l=e.n(h),s=e(85573),a=e.n(s),n=e(96162),_=e(8645),p=function(D){E()(i,D);var f=l()(i);function i(){var m;o()(this,i);for(var M=arguments.length,y=new Array(M),I=0;I<M;I++)y[I]=arguments[I];return m=f.call.apply(f,[this].concat(y)),a()(r()(m),"xRef",n.createRef()),a()(r()(m),"yRef",n.createRef()),a()(r()(m),"zRef",n.createRef()),a()(r()(m),"state",{nums:""}),a()(r()(m),"onGen",function(O){var C=m.props.onGen;O.preventDefault();var T=m.getVals();C((0,_.generateData)(T.x,T.y,T.z)),m.setState({nums:(0,_.calcTotal)(T.x,T.y,T.z)})}),a()(r()(m),"getVals",function(){return{x:parseInt(m.xRef.current.value,10),y:parseInt(m.yRef.current.value,10),z:parseInt(m.zRef.current.value,10)}}),a()(r()(m),"getDefaultValue",function(O){return O in m.props?String(m.props[O]):null}),m}return P()(i,[{key:"componentDidMount",value:function(){var M=this.props.onGen,y=this.getVals();M((0,_.generateData)(y.x,y.y,y.z))}},{key:"render",value:function(){var M=this.state.nums,y=this.props,I=y.x,O=y.y,C=y.z;return n.createElement("div",{style:{padding:"0 20px"}},n.createElement("h2",null,"big data generator"),n.createElement("form",{onSubmit:this.onGen},n.createElement("span",{style:{marginRight:10}},"x:"," ",n.createElement("input",{ref:this.xRef,defaultValue:this.getDefaultValue("x"),type:"number",min:"1",required:!0,style:{width:50}})),n.createElement("span",{style:{marginRight:10}},"y:"," ",n.createElement("input",{ref:this.yRef,defaultValue:this.getDefaultValue("y"),type:"number",min:"1",required:!0,style:{width:50}})),n.createElement("span",{style:{marginRight:10}},"z:"," ",n.createElement("input",{ref:this.zRef,defaultValue:this.getDefaultValue("z"),type:"number",min:"1",required:!0,style:{width:50}})),n.createElement("button",{type:"submit"},"Generate"),n.createElement("p",null,"total nodes: ",M||(0,_.calcTotal)(I,O,C))),n.createElement("p",{style:{fontSize:12}},"x\uFF1A\u6BCF\u4E00\u7EA7\u4E0B\u7684\u8282\u70B9\u603B\u6570\u3002y\uFF1A\u6BCF\u7EA7\u8282\u70B9\u91CC\u6709y\u4E2A\u8282\u70B9\u3001\u5B58\u5728\u5B50\u8282\u70B9\u3002z\uFF1A\u6811\u7684level\u5C42\u7EA7\u6570\uFF080\u8868\u793A\u4E00\u7EA7\uFF09"))}}]),i}(n.Component);a()(p,"defaultProps",{onGen:function(){},x:20,y:18,z:1}),t.default=p},8645:function(v,t,e){e.r(t),e.d(t,{calcTotal:function(){return o},gData:function(){return u},generateData:function(){return c},generateTreeNodes:function(){return P},getFilterValue:function(){return h},getNewTreeData:function(){return r}});function c(){var l=arguments.length>0&&arguments[0]!==void 0?arguments[0]:3,s=arguments.length>1&&arguments[1]!==void 0?arguments[1]:2,a=arguments.length>2&&arguments[2]!==void 0?arguments[2]:1,n=arguments.length>3&&arguments[3]!==void 0?arguments[3]:[];function _(p,D,f){for(var i=D||"0",m=f||n,M=[],y=0;y<l;y++){var I="".concat(i,"-").concat(y);m.push({label:"".concat(I,"-label"),value:"".concat(I,"-value"),key:I,disabled:I==="0-0-0-1"||!1}),y<s&&M.push(I)}if(p<0)return m;var O=p-1;return M.forEach(function(C,T){return m[T].children=[],_(O,C,m[T].children)}),null}return _(a),n}function o(){var l=arguments.length>0&&arguments[0]!==void 0?arguments[0]:3,s=arguments.length>1&&arguments[1]!==void 0?arguments[1]:2,a=arguments.length>2&&arguments[2]!==void 0?arguments[2]:1,n=function _(p){return p>=0?l*Math.pow(s,p--)+_(p):0};return n(a+1)}console.log("\u603B\u8282\u70B9\u6570\uFF08\u5355\u4E2Atree\uFF09\uFF1A",o());var u=c();function P(l){for(var s=[],a=l.props.eventKey,n=0;n<3;n++)s.push({label:"".concat(a,"-").concat(n,"-label"),value:"".concat(a,"-").concat(n,"-value"),key:"".concat(a,"-").concat(n)});return s}function g(l,s,a){var n=function _(p,D){var f=D-1;p.forEach(function(i){(i.key.length>s.length?i.key.indexOf(s)!==0:s.indexOf(i.key)!==0)||(i.children?_(i.children,f):f<1&&(i.isLeaf=!0))})};n(l,a+1)}function r(l,s,a,n){var _=function p(D){n<1||s.length-3>n*2||D.forEach(function(f){s.indexOf(f.key)===0&&(f.children?p(f.children):f.children=a)})};_(l),g(l,s,n)}function U(l,s){var a=function n(_){var p=arguments.length>1&&arguments[1]!==void 0?arguments[1]:"0";_.forEach(function(D,f){var i="".concat(p,"-").concat(f);D.children&&n(D.children,i),s(D,f,i)})};a(l)}function E(l,s){return s.length<l.length||s.length>l.length&&s.charAt(l.length)!=="-"?!1:s.substr(0,l.length)===l}function h(l,s,a){var n=[],_=[];U(u,function(f,i,m){s.indexOf(f.value)>-1&&n.push(m),a.indexOf(f.value)>-1&&_.push(m)});var p=[];_.forEach(function(f){n.forEach(function(i){E(f,i)||E(i,f)||p.push(i)})});var D=[];return p.length&&U(u,function(f,i,m){p.indexOf(m)>-1&&D.push(f.value)}),D}},64803:function(v,t,e){e.r(t);var c=e(96162),o=c.createContext(null);t.default=o},87992:function(v,t,e){e.r(t);var c=e(95582),o=e.n(c),u=e(91232),P=e.n(u),g=e(37205),r=e.n(g),U=e(79800),E=e.n(U),h=e(28853),l=e(64887),s=e(1732),a=e(26694),n=e(96162),_=e(64803),p=e(19811),D=e(87643),f=e(4216),i={width:0,height:0,display:"flex",overflow:"hidden",opacity:0,border:0,padding:0,margin:0},m=function(I,O){var C=(0,h.useBaseProps)(),T=C.prefixCls,j=C.multiple,x=C.searchValue,S=C.toggleOpen,R=C.open,A=C.notFoundContent,L=n.useContext(p.default),V=L.virtual,Z=L.listHeight,$=L.listItemHeight,J=L.listItemScrollOffset,b=L.treeData,d=L.fieldNames,X=L.onSelect,G=L.dropdownMatchSelectWidth,F=L.treeExpandAction,H=L.treeTitleRender,k=L.onPopupScroll,K=L.leftMaxCount,W=L.leafCountOnly,le=L.valueEntities,N=n.useContext(_.default),De=N.checkable,ie=N.checkedKeys,Oe=N.halfCheckedKeys,Ue=N.treeExpandedKeys,It=N.treeDefaultExpandAll,Pt=N.treeDefaultExpandedKeys,Ne=N.onTreeExpand,Be=N.treeIcon,nt=N.showTreeIcon,be=N.switcherIcon,Ct=N.treeLine,ve=N.treeNodeFilterProp,ue=N.loadData,Fe=N.treeLoadedKeys,ye=N.treeMotion,st=N.onTreeLoad,Ut=N.keyEntities,Te=n.useRef(),Ve=(0,a.default)(function(){return b},[R,b],function(B,Q){return Q[0]&&B[1]!==Q[1]}),lt=n.useMemo(function(){return De?{checked:ie,halfChecked:Oe}:null},[De,ie,Oe]);n.useEffect(function(){if(R&&!j&&ie.length){var B;(B=Te.current)===null||B===void 0||B.scrollTo({key:ie[0]})}},[R]);var Ze=function(Q){Q.preventDefault()},xe=function(Q,ae){var se=ae.node;De&&(0,D.isCheckDisabled)(se)||(X(se.key,{selected:!ie.includes(se.key)}),j||S(!1))},rt=n.useState(Pt),He=E()(rt,2),ge=He[0],at=He[1],ot=n.useState(null),ke=E()(ot,2),$e=ke[0],ze=ke[1],Ge=n.useMemo(function(){return Ue?r()(Ue):x?$e:ge},[ge,$e,Ue,x]),ct=function(Q){at(Q),ze(Q),Ne&&Ne(Q)},Ye=String(x).toLowerCase(),Je=function(Q){return Ye?String(Q[ve]).toLowerCase().includes(Ye):!1};n.useEffect(function(){x&&ze((0,D.getAllKeys)(b,d))},[x]);var Xe=n.useState(function(){return new Map}),Qe=E()(Xe,2),Ie=Qe[0],it=Qe[1];n.useEffect(function(){K&&it(new Map)},[K]);function _t(B){var Q=B[d.value];if(!Ie.has(Q)){var ae=le.get(Q),se=(ae.children||[]).length===0;if(se)Ie.set(Q,!1);else{var ce=ae.children.filter(function(q){return!q.node.disabled&&!q.node.disableCheckbox&&!ie.includes(q.node[d.value])}),oe=ce.length;Ie.set(Q,oe>K)}}return Ie.get(Q)}var we=(0,f.useEvent)(function(B){var Q=B[d.value];return ie.includes(Q)||K===null?!1:K<=0?!0:W&&K?_t(B):!1}),dt=function B(Q){var ae=P()(Q),se;try{for(ae.s();!(se=ae.n()).done;){var ce=se.value;if(!(ce.disabled||ce.selectable===!1)){if(x){if(Je(ce))return ce}else return ce;if(ce[d.children]){var oe=B(ce[d.children]);if(oe)return oe}}}}catch(q){ae.e(q)}finally{ae.f()}return null},ut=n.useState(null),Ee=E()(ut,2),je=Ee[0],pt=Ee[1],me=Ut[je];n.useEffect(function(){if(R){var B=null,Q=function(){var se=dt(Ve);return se?se[d.value]:null};!j&&ie.length&&!x?B=ie[0]:B=Q(),pt(B)}},[R,x]),n.useImperativeHandle(O,function(){var B;return{scrollTo:(B=Te.current)===null||B===void 0?void 0:B.scrollTo,onKeyDown:function(ae){var se,ce=ae.which;switch(ce){case s.default.UP:case s.default.DOWN:case s.default.LEFT:case s.default.RIGHT:(se=Te.current)===null||se===void 0||se.onKeyDown(ae);break;case s.default.ENTER:{if(me){var oe=we(me.node),q=(me==null?void 0:me.node)||{},Mt=q.selectable,vt=q.value,Et=q.disabled;Mt!==!1&&!Et&&!oe&&xe(null,{node:{key:je},selected:!ie.includes(vt)})}break}case s.default.ESC:S(!1)}},onKeyUp:function(){}}});var Le=(0,a.default)(function(){return!x},[x,Ue||ge],function(B,Q){var ae=E()(B,1),se=ae[0],ce=E()(Q,2),oe=ce[0],q=ce[1];return se!==oe&&!!(oe||q)}),Re=Le?ue:null;if(Ve.length===0)return n.createElement("div",{role:"listbox",className:"".concat(T,"-empty"),onMouseDown:Ze},A);var Ke={fieldNames:d};return Fe&&(Ke.loadedKeys=Fe),Ge&&(Ke.expandedKeys=Ge),n.createElement("div",{onMouseDown:Ze},me&&R&&n.createElement("span",{style:i,"aria-live":"assertive"},me.node.value),n.createElement(l.UnstableContext.Provider,{value:{nodeDisabled:we}},n.createElement(l.default,o()({ref:Te,focusable:!1,prefixCls:"".concat(T,"-tree"),treeData:Ve,height:Z,itemHeight:$,itemScrollOffset:J,virtual:V!==!1&&G!==!1,multiple:j,icon:Be,showIcon:nt,switcherIcon:be,showLine:Ct,loadData:Re,motion:ye,activeKey:je,checkable:De,checkStrictly:!0,checkedKeys:lt,selectedKeys:De?[]:ie,defaultExpandAll:It,titleRender:H},Ke,{onActiveChange:pt,onSelect:xe,onCheck:xe,onExpand:ct,onLoad:st,filterTreeNode:Je,expandAction:F,onScroll:k}))))},M=n.forwardRef(m);t.default=M},84670:function(v,t,e){e.r(t);var c=function(){return null};t.default=c},27165:function(v,t,e){e.r(t);var c=e(95582),o=e.n(c),u=e(37205),P=e.n(u),g=e(82242),r=e.n(g),U=e(79800),E=e.n(U),h=e(39647),l=e.n(h),s=e(31468),a=e.n(s),n=e(28853),_=e(83238),p=e(95094),D=e(99217),f=e(8399),i=e(96162),m=e(43647),M=e(89957),y=e(2971),I=e(41234),O=e(46077),C=e(76536),T=e(64803),j=e(87992),x=e(84670),S=e(19811),R=e(72770),A=e(56253),L=e(87643),V=e(47239),Z=["id","prefixCls","value","defaultValue","onChange","onSelect","onDeselect","searchValue","inputValue","onSearch","autoClearSearchValue","filterTreeNode","treeNodeFilterProp","showCheckedStrategy","treeNodeLabelProp","multiple","treeCheckable","treeCheckStrictly","labelInValue","maxCount","fieldNames","treeDataSimpleMode","treeData","children","loadData","treeLoadedKeys","onTreeLoad","treeDefaultExpandAll","treeExpandedKeys","treeDefaultExpandedKeys","onTreeExpand","treeExpandAction","virtual","listHeight","listItemHeight","listItemScrollOffset","onDropdownVisibleChange","dropdownMatchSelectWidth","treeLine","treeIcon","showTreeIcon","switcherIcon","treeMotion","treeTitleRender","onPopupScroll"];function $(d){return!d||a()(d)!=="object"}var J=i.forwardRef(function(d,X){var G=d.id,F=d.prefixCls,H=F===void 0?"rc-tree-select":F,k=d.value,K=d.defaultValue,W=d.onChange,le=d.onSelect,N=d.onDeselect,De=d.searchValue,ie=d.inputValue,Oe=d.onSearch,Ue=d.autoClearSearchValue,It=Ue===void 0?!0:Ue,Pt=d.filterTreeNode,Ne=d.treeNodeFilterProp,Be=Ne===void 0?"value":Ne,nt=d.showCheckedStrategy,be=d.treeNodeLabelProp,Ct=d.multiple,ve=d.treeCheckable,ue=d.treeCheckStrictly,Fe=d.labelInValue,ye=d.maxCount,st=d.fieldNames,Ut=d.treeDataSimpleMode,Te=d.treeData,Ve=d.children,lt=d.loadData,Ze=d.treeLoadedKeys,xe=d.onTreeLoad,rt=d.treeDefaultExpandAll,He=d.treeExpandedKeys,ge=d.treeDefaultExpandedKeys,at=d.onTreeExpand,ot=d.treeExpandAction,ke=d.virtual,$e=d.listHeight,ze=$e===void 0?200:$e,Ge=d.listItemHeight,ct=Ge===void 0?20:Ge,Ye=d.listItemScrollOffset,Je=Ye===void 0?0:Ye,Xe=d.onDropdownVisibleChange,Qe=d.dropdownMatchSelectWidth,Ie=Qe===void 0?!0:Qe,it=d.treeLine,_t=d.treeIcon,we=d.showTreeIcon,dt=d.switcherIcon,ut=d.treeMotion,Ee=d.treeTitleRender,je=d.onPopupScroll,pt=l()(d,Z),me=(0,_.default)(G),Le=ve&&!ue,Re=ve||ue,Ke=ue||Fe,B=Re||Ct,Q=(0,D.default)(K,{value:k}),ae=E()(Q,2),se=ae[0],ce=ae[1],oe=i.useMemo(function(){return ve?nt||A.SHOW_CHILD:A.SHOW_ALL},[nt,ve]),q=i.useMemo(function(){return(0,L.fillFieldNames)(st)},[JSON.stringify(st)]),Mt=(0,D.default)("",{value:De!==void 0?De:ie,postState:function(z){return z||""}}),vt=E()(Mt,2),Et=vt[0],xt=vt[1],bt=function(z){xt(z),Oe==null||Oe(z)},ht=(0,C.default)(Te,Ve,Ut),gt=(0,y.default)(ht,q),de=gt.keyEntities,fe=gt.valueEntities,jt=i.useCallback(function(Y){var z=[],w=[];return Y.forEach(function(ee){fe.has(ee)?w.push(ee):z.push(ee)}),{missingRawValues:z,existRawValues:w}},[fe]),Lt=(0,I.default)(ht,Et,{fieldNames:q,treeNodeFilterProp:Be,filterTreeNode:Pt}),Rt=i.useCallback(function(Y){if(Y){if(be)return Y[be];for(var z=q._title,w=0;w<z.length;w+=1){var ee=Y[z[w]];if(ee!==void 0)return ee}}},[q,be]),qe=i.useCallback(function(Y){var z=(0,L.toArray)(Y);return z.map(function(w){return $(w)?{value:w}:w})},[]),mt=i.useCallback(function(Y){var z=qe(Y);return z.map(function(w){var ee=w.label,pe=w.value,re=w.halfChecked,te,ne=fe.get(pe);if(ne){var _e;ee=Ee?Ee(ne.node):(_e=ee)!==null&&_e!==void 0?_e:Rt(ne.node),te=ne.node.disabled}else if(ee===void 0){var Pe=qe(se).find(function(et){return et.value===pe});ee=Pe.label}return{label:ee,value:pe,halfChecked:re,disabled:te}})},[fe,Rt,qe,se]),Kt=i.useMemo(function(){return qe(se===null?[]:se)},[qe,se]),Ft=i.useMemo(function(){var Y=[],z=[];return Kt.forEach(function(w){w.halfChecked?z.push(w):Y.push(w)}),[Y,z]},[Kt]),St=E()(Ft,2),Se=St[0],At=St[1],Wt=i.useMemo(function(){return Se.map(function(Y){return Y.value})},[Se]),Vt=(0,M.default)(Se,At,Le,de),Nt=E()(Vt,2),Ae=Nt[0],ft=Nt[1],Zt=i.useMemo(function(){var Y=(0,A.formatStrategyValues)(Ae,oe,de,q),z=Y.map(function(re){var te,ne;return(te=(ne=de[re])===null||ne===void 0||(ne=ne.node)===null||ne===void 0?void 0:ne[q.value])!==null&&te!==void 0?te:re}),w=z.map(function(re){var te=Se.find(function(_e){return _e.value===re}),ne=Fe?te==null?void 0:te.label:Ee==null?void 0:Ee(te);return{value:re,label:ne}}),ee=mt(w),pe=ee[0];return!B&&pe&&(0,L.isNil)(pe.value)&&(0,L.isNil)(pe.label)?[]:ee.map(function(re){var te;return r()(r()({},re),{},{label:(te=re.label)!==null&&te!==void 0?te:re.value})})},[q,B,Ae,Se,mt,oe,de]),Ht=(0,m.default)(Zt),kt=E()(Ht,1),Ot=kt[0],Bt=i.useMemo(function(){return B&&(oe==="SHOW_CHILD"||ue||!ve)?ye:null},[ye,B,ue,oe,ve]),Dt=(0,O.default)(function(Y,z,w){var ee=(0,A.formatStrategyValues)(Y,oe,de,q);if(!(Bt&&ee.length>Bt)){var pe=mt(Y);if(ce(pe),It&&xt(""),W){var re=Y;Le&&(re=ee.map(function(he){var Ce=fe.get(he);return Ce?Ce.node[q.value]:he}));var te=z||{triggerValue:void 0,selected:void 0},ne=te.triggerValue,_e=te.selected,Pe=re;if(ue){var et=At.filter(function(he){return!re.includes(he.value)});Pe=[].concat(P()(Pe),P()(et))}var tt=mt(Pe),Me={preValue:Se,triggerValue:ne},We=!0;(ue||w==="selection"&&!_e)&&(We=!1),(0,R.fillAdditionalInfo)(Me,ne,Y,ht,We,q),Re?Me.checked=_e:Me.selected=_e;var yt=Ke?tt:tt.map(function(he){return he.value});W(B?yt:yt[0],Ke?null:tt.map(function(he){return he.label}),Me)}}}),Tt=i.useCallback(function(Y,z){var w,ee=z.selected,pe=z.source,re=de[Y],te=re==null?void 0:re.node,ne=(w=te==null?void 0:te[q.value])!==null&&w!==void 0?w:Y;if(!B)Dt([ne],{selected:!0,triggerValue:ne},"option");else{var _e=ee?[].concat(P()(Wt),[ne]):Ae.filter(function(Ce){return Ce!==ne});if(Le){var Pe=jt(_e),et=Pe.missingRawValues,tt=Pe.existRawValues,Me=tt.map(function(Ce){return fe.get(Ce).key}),We;if(ee){var yt=(0,p.conductCheck)(Me,!0,de);We=yt.checkedKeys}else{var he=(0,p.conductCheck)(Me,{checked:!1,halfCheckedKeys:ft},de);We=he.checkedKeys}_e=[].concat(P()(et),P()(We.map(function(Ce){return de[Ce].node[q.value]})))}Dt(_e,{selected:ee,triggerValue:ne},pe||"option")}ee||!B?le==null||le(ne,(0,R.fillLegacyProps)(te)):N==null||N(ne,(0,R.fillLegacyProps)(te))},[jt,fe,de,q,B,Wt,Dt,Le,le,N,Ae,ft,ye]),$t=i.useCallback(function(Y){if(Xe){var z={};Object.defineProperty(z,"documentClickClose",{get:function(){return(0,f.default)(!1,"Second param of `onDropdownVisibleChange` has been removed."),!1}}),Xe(Y,z)}},[Xe]),zt=(0,O.default)(function(Y,z){var w=Y.map(function(ee){return ee.value});if(z.type==="clear"){Dt(w,{},"selection");return}z.values.length&&Tt(z.values[0].value,{selected:!1,source:"selection"})}),Gt=i.useMemo(function(){return{virtual:ke,dropdownMatchSelectWidth:Ie,listHeight:ze,listItemHeight:ct,listItemScrollOffset:Je,treeData:Lt,fieldNames:q,onSelect:Tt,treeExpandAction:ot,treeTitleRender:Ee,onPopupScroll:je,leftMaxCount:ye===void 0?null:ye-Ot.length,leafCountOnly:oe==="SHOW_CHILD"&&!ue&&!!ve,valueEntities:fe}},[ke,Ie,ze,ct,Je,Lt,q,Tt,ot,Ee,je,ye,Ot.length,oe,ue,ve,fe]),Yt=i.useMemo(function(){return{checkable:Re,loadData:lt,treeLoadedKeys:Ze,onTreeLoad:xe,checkedKeys:Ae,halfCheckedKeys:ft,treeDefaultExpandAll:rt,treeExpandedKeys:He,treeDefaultExpandedKeys:ge,onTreeExpand:at,treeIcon:_t,treeMotion:ut,showTreeIcon:we,switcherIcon:dt,treeLine:it,treeNodeFilterProp:Be,keyEntities:de}},[Re,lt,Ze,xe,Ae,ft,rt,He,ge,at,_t,ut,we,dt,it,Be,de]);return i.createElement(S.default.Provider,{value:Gt},i.createElement(T.default.Provider,{value:Yt},i.createElement(n.BaseSelect,o()({ref:X},pt,{id:me,prefixCls:H,mode:B?"multiple":void 0,displayValues:Ot,onDisplayValuesChange:zt,searchValue:Et,onSearch:bt,OptionList:j.default,emptyOptions:!ht.length,onDropdownVisibleChange:$t,dropdownMatchSelectWidth:Ie}))))}),b=J;b.TreeNode=x.default,b.SHOW_ALL=A.SHOW_ALL,b.SHOW_PARENT=A.SHOW_PARENT,b.SHOW_CHILD=A.SHOW_CHILD,t.default=b},19811:function(v,t,e){e.r(t);var c=e(96162),o=c.createContext(null);t.default=o},43647:function(v,t,e){e.r(t);var c=e(82242),o=e.n(c),u=e(96162);t.default=function(P){var g=u.useRef({valueLabels:new Map});return u.useMemo(function(){var r=g.current.valueLabels,U=new Map,E=P.map(function(h){var l=h.value,s=h.label,a=s!=null?s:r.get(l);return U.set(l,a),o()(o()({},h),{},{label:a})});return g.current.valueLabels=U,[E]},[P])}},89957:function(v,t,e){e.r(t);var c=e(37205),o=e.n(c),u=e(96162),P=e(95094),g=function(U,E,h,l){return u.useMemo(function(){var s=function(m){return m.map(function(M){var y=M.value;return y})},a=s(U),n=s(E),_=a.filter(function(i){return!l[i]}),p=a,D=n;if(h){var f=(0,P.conductCheck)(a,!0,l);p=f.checkedKeys,D=f.halfCheckedKeys}return[Array.from(new Set([].concat(o()(_),o()(p)))),D]},[U,E,h,l])};t.default=g},2971:function(v,t,e){e.r(t);var c=e(82242),o=e.n(c),u=e(96162),P=e(43880),g=e(8399);t.default=function(r,U){return u.useMemo(function(){var E=(0,P.convertDataToEntities)(r,{fieldNames:U,initWrapper:function(l){return o()(o()({},l),{},{valueEntities:new Map})},processEntity:function(l,s){var a=l.node[U.value];if(0)var n;s.valueEntities.set(a,l)}});return E},[r,U])}},41234:function(v,t,e){e.r(t);var c=e(85573),o=e.n(c),u=e(82242),P=e.n(u),g=e(96162),r=e(72770),U=function(h,l,s){var a=s.fieldNames,n=s.treeNodeFilterProp,_=s.filterTreeNode,p=a.children;return g.useMemo(function(){if(!l||_===!1)return h;var D=typeof _=="function"?_:function(i,m){return String(m[n]).toUpperCase().includes(l.toUpperCase())},f=function i(m){var M=arguments.length>1&&arguments[1]!==void 0?arguments[1]:!1;return m.reduce(function(y,I){var O=I[p],C=M||D(l,(0,r.fillLegacyProps)(I)),T=i(O||[],C);return(C||T.length)&&y.push(P()(P()({},I),{},o()({isLeaf:void 0},p,T))),y},[])};return f(h)},[h,l,p,n,_])};t.default=U},46077:function(v,t,e){e.r(t),e.d(t,{default:function(){return o}});var c=e(96162);function o(u){var P=c.useRef();P.current=u;var g=c.useCallback(function(){return P.current.apply(P,arguments)},[]);return g}},76536:function(v,t,e){e.r(t),e.d(t,{default:function(){return E}});var c=e(31468),o=e.n(c),u=e(82242),P=e.n(u),g=e(96162),r=e(72770);function U(h,l){var s=l.id,a=l.pId,n=l.rootPId,_=new Map,p=[];return h.forEach(function(D){var f=D[s],i=P()(P()({},D),{},{key:D.key||f});_.set(f,i)}),_.forEach(function(D){var f=D[a],i=_.get(f);i?(i.children=i.children||[],i.children.push(D)):(f===n||n===null)&&p.push(D)}),p}function E(h,l,s){return g.useMemo(function(){if(h){if(s){var a=P()({id:"id",pId:"pId",rootPId:null},o()(s)==="object"?s:{});return U(h,a)}return h}return(0,r.convertChildrenToData)(l)},[l,s,h])}},63484:function(v,t,e){e.r(t),e.d(t,{SHOW_ALL:function(){return u.SHOW_ALL},SHOW_CHILD:function(){return u.SHOW_CHILD},SHOW_PARENT:function(){return u.SHOW_PARENT},TreeNode:function(){return o.default}});var c=e(27165),o=e(84670),u=e(56253);t.default=c.default},72770:function(v,t,e){e.r(t),e.d(t,{convertChildrenToData:function(){return l},fillAdditionalInfo:function(){return a},fillLegacyProps:function(){return s}});var c=e(82242),o=e.n(c),u=e(39647),P=e.n(u),g=e(96162),r=e(29213),U=e(8399),E=e(84670),h=["children","value"];function l(n){return(0,r.default)(n).map(function(_){if(!g.isValidElement(_)||!_.type)return null;var p=_,D=p.key,f=p.props,i=f.children,m=f.value,M=P()(f,h),y=o()({key:D,value:m},M),I=l(i);return I.length&&(y.children=I),y}).filter(function(_){return _})}function s(n){if(!n)return n;var _=o()({},n);return"props"in _||Object.defineProperty(_,"props",{get:function(){return(0,U.default)(!1,"New `rc-tree-select` not support return node instance as argument anymore. Please consider to remove `props` access."),_}}),_}function a(n,_,p,D,f,i){var m=null,M=null;function y(){function I(O){var C=arguments.length>1&&arguments[1]!==void 0?arguments[1]:"0",T=arguments.length>2&&arguments[2]!==void 0?arguments[2]:!1;return O.map(function(j,x){var S="".concat(C,"-").concat(x),R=j[i.value],A=p.includes(R),L=I(j[i.children]||[],S,A),V=g.createElement(E.default,j,L.map(function($){return $.node}));if(_===R&&(m=V),A){var Z={pos:S,node:V,children:L};return T||M.push(Z),Z}return null}).filter(function(j){return j})}M||(M=[],I(D),M.sort(function(O,C){var T=O.node.props.value,j=C.node.props.value,x=p.indexOf(T),S=p.indexOf(j);return x-S}))}Object.defineProperty(n,"triggerNode",{get:function(){return(0,U.default)(!1,"`triggerNode` is deprecated. Please consider decoupling data with node."),y(),m}}),Object.defineProperty(n,"allCheckedNodes",{get:function(){return(0,U.default)(!1,"`allCheckedNodes` is deprecated. Please consider decoupling data with node."),y(),f?M:M.map(function(O){var C=O.node;return C})}})}},56253:function(v,t,e){e.r(t),e.d(t,{SHOW_ALL:function(){return o},SHOW_CHILD:function(){return P},SHOW_PARENT:function(){return u},formatStrategyValues:function(){return g}});var c=e(87643),o="SHOW_ALL",u="SHOW_PARENT",P="SHOW_CHILD";function g(r,U,E,h){var l=new Set(r);return U===P?r.filter(function(s){var a=E[s];return!a||!a.children||!a.children.some(function(n){var _=n.node;return l.has(_[h.value])})||!a.children.every(function(n){var _=n.node;return(0,c.isCheckDisabled)(_)||l.has(_[h.value])})}):U===u?r.filter(function(s){var a=E[s],n=a?a.parent:null;return!n||(0,c.isCheckDisabled)(n.node)||!l.has(n.key)}):r}},87643:function(v,t,e){e.r(t),e.d(t,{fillFieldNames:function(){return o},getAllKeys:function(){return P},isCheckDisabled:function(){return u},isNil:function(){return g},toArray:function(){return c}});var c=function(U){return Array.isArray(U)?U:U!==void 0?[U]:[]},o=function(U){var E=U||{},h=E.label,l=E.value,s=E.children;return{_title:h?[h]:["title","label"],value:l||"value",key:l||"value",children:s||"children"}},u=function(U){return!U||U.disabled||U.disableCheckbox||U.checkable===!1},P=function(U,E){var h=[],l=function s(a){a.forEach(function(n){var _=n[E.children];_&&(h.push(n[E.value]),s(_))})};return l(U),h},g=function(U){return U==null}},47239:function(v,t,e){e.r(t);var c=e(31468),o=e.n(c),u=e(8399),P=e(87643);function g(r){var U=r.searchPlaceholder,E=r.treeCheckStrictly,h=r.treeCheckable,l=r.labelInValue,s=r.value,a=r.multiple,n=r.showCheckedStrategy,_=r.maxCount;(0,u.default)(!U,"`searchPlaceholder` has been removed."),E&&l===!1&&(0,u.default)(!1,"`treeCheckStrictly` will force set `labelInValue` to `true`."),(l||E)&&(0,u.default)((0,P.toArray)(s).every(function(p){return p&&o()(p)==="object"&&"value"in p}),"Invalid prop `value` supplied to `TreeSelect`. You should use { label: string, value: string | number } or [{ label: string, value: string | number }] instead."),E||a||h?(0,u.default)(!s||Array.isArray(s),"`value` should be an array when `TreeSelect` is checkable or multiple."):(0,u.default)(!Array.isArray(s),"`value` should not be array when `TreeSelect` is single mode."),_&&(n==="SHOW_ALL"&&!E||n==="SHOW_PARENT")&&(0,u.default)(!1,"`maxCount` not work with `showCheckedStrategy=SHOW_ALL` (when `treeCheckStrictly=false`) or `showCheckedStrategy=SHOW_PARENT`.")}t.default=g},25240:function(v,t,e){e.r(t)},9450:function(v,t,e){e.r(t),e.d(t,{texts:function(){return o}});var c=e(80206);const o=[]},31479:function(v,t,e){e.r(t),e.d(t,{texts:function(){return o}});var c=e(2934);const o=[]},34334:function(v,t,e){e.r(t),e.d(t,{texts:function(){return o}});var c=e(86795);const o=[]},69557:function(v,t,e){e.r(t),e.d(t,{texts:function(){return o}});var c=e(32445);const o=[]},17948:function(v,t,e){e.r(t),e.d(t,{texts:function(){return o}});var c=e(55145);const o=[]},92463:function(v,t,e){e.r(t),e.d(t,{texts:function(){return o}});var c=e(94785);const o=[]},20137:function(v,t,e){e.r(t),e.d(t,{texts:function(){return o}});var c=e(547);const o=[]},90673:function(v,t,e){e.r(t),e.d(t,{texts:function(){return o}});var c=e(55230);const o=[]},43064:function(v,t,e){e.r(t),e.d(t,{texts:function(){return o}});var c=e(63166);const o=[]},33546:function(v,t,e){e.r(t),e.d(t,{texts:function(){return o}});var c=e(29272);const o=[]},48502:function(v,t,e){e.r(t),e.d(t,{texts:function(){return o}});var c=e(20189);const o=[]},59807:function(v,t,e){e.r(t),e.d(t,{texts:function(){return o}});var c=e(67928);const o=[]},80011:function(v,t,e){e.r(t),e.d(t,{texts:function(){return o}});var c=e(15316);const o=[]},99944:function(v,t,e){e.r(t),e.d(t,{texts:function(){return o}});var c=e(8210);const o=[{value:"React TreeSelect Component",paraId:0,tocIndex:0},{value:`npm install
npm start
`,paraId:1,tocIndex:2},{value:"http://localhost:8000/examples/",paraId:2,tocIndex:3},{value:"online example: ",paraId:3,tocIndex:3},{value:"https://tree-select-react-component.vercel.app/",paraId:3,tocIndex:3},{value:"name",paraId:4,tocIndex:6},{value:"description",paraId:4,tocIndex:6},{value:"type",paraId:4,tocIndex:6},{value:"default",paraId:4,tocIndex:6},{value:"className",paraId:4,tocIndex:6},{value:"additional css class of root dom node",paraId:4,tocIndex:6},{value:"String",paraId:4,tocIndex:6},{value:"''",paraId:4,tocIndex:6},{value:"prefixCls",paraId:4,tocIndex:6},{value:"prefix class",paraId:4,tocIndex:6},{value:"String",paraId:4,tocIndex:6},{value:"''",paraId:4,tocIndex:6},{value:"animation",paraId:4,tocIndex:6},{value:"dropdown animation name. only support slide-up now",paraId:4,tocIndex:6},{value:"String",paraId:4,tocIndex:6},{value:"''",paraId:4,tocIndex:6},{value:"transitionName",paraId:4,tocIndex:6},{value:"dropdown css animation name",paraId:4,tocIndex:6},{value:"String",paraId:4,tocIndex:6},{value:"''",paraId:4,tocIndex:6},{value:"choiceTransitionName",paraId:4,tocIndex:6},{value:"css animation name for selected items at multiple mode",paraId:4,tocIndex:6},{value:"String",paraId:4,tocIndex:6},{value:"''",paraId:4,tocIndex:6},{value:"dropdownMatchSelectWidth",paraId:4,tocIndex:6},{value:"whether dropdown's with is same with select. Default set ",paraId:4,tocIndex:6},{value:"min-width",paraId:4,tocIndex:6},{value:" same as input",paraId:4,tocIndex:6},{value:"bool",paraId:4,tocIndex:6},{value:"true",paraId:4,tocIndex:6},{value:"dropdownClassName",paraId:4,tocIndex:6},{value:"additional className applied to dropdown",paraId:4,tocIndex:6},{value:"String",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"dropdownStyle",paraId:4,tocIndex:6},{value:"additional style applied to dropdown",paraId:4,tocIndex:6},{value:"Object",paraId:4,tocIndex:6},{value:"{}",paraId:4,tocIndex:6},{value:"onDropdownVisibleChange",paraId:4,tocIndex:6},{value:"control dropdown visible",paraId:4,tocIndex:6},{value:"function",paraId:4,tocIndex:6},{value:"() => { return true; }",paraId:4,tocIndex:6},{value:"notFoundContent",paraId:4,tocIndex:6},{value:"specify content to show when no result matches.",paraId:4,tocIndex:6},{value:"String",paraId:4,tocIndex:6},{value:"'Not Found'",paraId:4,tocIndex:6},{value:"showSearch",paraId:4,tocIndex:6},{value:"whether show search input in single mode",paraId:4,tocIndex:6},{value:"bool",paraId:4,tocIndex:6},{value:"true",paraId:4,tocIndex:6},{value:"allowClear",paraId:4,tocIndex:6},{value:"whether allowClear",paraId:4,tocIndex:6},{value:"bool",paraId:4,tocIndex:6},{value:"false",paraId:4,tocIndex:6},{value:"maxTagTextLength",paraId:4,tocIndex:6},{value:"max tag text length to show",paraId:4,tocIndex:6},{value:"number",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"maxTagCount",paraId:4,tocIndex:6},{value:"max tag count to show",paraId:4,tocIndex:6},{value:"number",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"maxCount",paraId:4,tocIndex:6},{value:"Limit the maximum number of items that can be selected in multiple mode",paraId:4,tocIndex:6},{value:"number",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"maxTagPlaceholder",paraId:4,tocIndex:6},{value:"placeholder for omitted values",paraId:4,tocIndex:6},{value:"ReactNode/function(omittedValues)",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"multiple",paraId:4,tocIndex:6},{value:"whether multiple select (true when enable treeCheckable)",paraId:4,tocIndex:6},{value:"bool",paraId:4,tocIndex:6},{value:"false",paraId:4,tocIndex:6},{value:"disabled",paraId:4,tocIndex:6},{value:"whether disabled select",paraId:4,tocIndex:6},{value:"bool",paraId:4,tocIndex:6},{value:"false",paraId:4,tocIndex:6},{value:"searchValue",paraId:4,tocIndex:6},{value:"work with ",paraId:4,tocIndex:6},{value:"onSearch",paraId:4,tocIndex:6},{value:" to make search value controlled.",paraId:4,tocIndex:6},{value:"string",paraId:4,tocIndex:6},{value:"''",paraId:4,tocIndex:6},{value:"defaultValue",paraId:4,tocIndex:6},{value:"initial selected treeNode(s)",paraId:4,tocIndex:6},{value:"same as value type",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"value",paraId:4,tocIndex:6},{value:"current selected treeNode(s).",paraId:4,tocIndex:6},{value:"normal: String/Array",paraId:4,tocIndex:6},{value:". labelInValue: {value:String,label:React.Node}/Array<{value,label}>. treeCheckStrictly(halfChecked default false): {value:String,label:React.Node, halfChecked}/Array<{value,label,halfChecked}>.",paraId:5,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"labelInValue",paraId:4,tocIndex:6},{value:"whether to embed label in value, see above value type",paraId:4,tocIndex:6},{value:"Bool",paraId:4,tocIndex:6},{value:"false",paraId:4,tocIndex:6},{value:"onChange",paraId:4,tocIndex:6},{value:"called when select treeNode or input value change",paraId:4,tocIndex:6},{value:"function(value, label(null), extra)",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"onSelect",paraId:4,tocIndex:6},{value:"called when select treeNode",paraId:4,tocIndex:6},{value:"function(value, node, extra)",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"onSearch",paraId:4,tocIndex:6},{value:"called when input changed",paraId:4,tocIndex:6},{value:"function",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"onTreeExpand",paraId:4,tocIndex:6},{value:"called when tree node expand",paraId:4,tocIndex:6},{value:"function(expandedKeys)",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"onPopupScroll",paraId:4,tocIndex:6},{value:"called when popup scroll",paraId:4,tocIndex:6},{value:"function(event)",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"showCheckedStrategy",paraId:4,tocIndex:6},{value:"TreeSelect.SHOW_ALL",paraId:4,tocIndex:6},{value:": show all checked treeNodes (Include parent treeNode). ",paraId:4,tocIndex:6},{value:"TreeSelect.SHOW_PARENT",paraId:4,tocIndex:6},{value:": show checked treeNodes (Just show parent treeNode). Default just show child.",paraId:4,tocIndex:6},{value:"enum{TreeSelect.SHOW_ALL, TreeSelect.SHOW_PARENT, TreeSelect.SHOW_CHILD }",paraId:4,tocIndex:6},{value:"TreeSelect.SHOW_CHILD",paraId:4,tocIndex:6},{value:"treeIcon",paraId:4,tocIndex:6},{value:"show tree icon",paraId:4,tocIndex:6},{value:"bool",paraId:4,tocIndex:6},{value:"false",paraId:4,tocIndex:6},{value:"treeLine",paraId:4,tocIndex:6},{value:"show tree line",paraId:4,tocIndex:6},{value:"bool",paraId:4,tocIndex:6},{value:"false",paraId:4,tocIndex:6},{value:"treeDefaultExpandAll",paraId:4,tocIndex:6},{value:"default expand all treeNode",paraId:4,tocIndex:6},{value:"bool",paraId:4,tocIndex:6},{value:"false",paraId:4,tocIndex:6},{value:"treeDefaultExpandedKeys",paraId:4,tocIndex:6},{value:"default expanded treeNode keys",paraId:4,tocIndex:6},{value:"Array",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"treeExpandedKeys",paraId:4,tocIndex:6},{value:"set tree expanded keys",paraId:4,tocIndex:6},{value:"Array",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"treeExpandAction",paraId:4,tocIndex:6},{value:"Tree open logic, optional: false | ",paraId:4,tocIndex:6},{value:"click",paraId:4,tocIndex:6},{value:" | ",paraId:4,tocIndex:6},{value:"doubleClick",paraId:4,tocIndex:6},{value:", same as ",paraId:4,tocIndex:6},{value:"expandAction",paraId:4,tocIndex:6},{value:" of ",paraId:4,tocIndex:6},{value:"rc-tree",paraId:4,tocIndex:6},{value:"string | boolean",paraId:4,tocIndex:6},{value:"click",paraId:4,tocIndex:6},{value:"treeCheckable",paraId:4,tocIndex:6},{value:"whether tree show checkbox (select callback will not fire)",paraId:4,tocIndex:6},{value:"bool",paraId:4,tocIndex:6},{value:"false",paraId:4,tocIndex:6},{value:"treeCheckStrictly",paraId:4,tocIndex:6},{value:"check node precisely, parent and children nodes are not associated",paraId:4,tocIndex:6},{value:"bool",paraId:4,tocIndex:6},{value:"false",paraId:4,tocIndex:6},{value:"filterTreeNode",paraId:4,tocIndex:6},{value:"whether filter treeNodes by input value. default filter by treeNode's treeNodeFilterProp prop's value",paraId:4,tocIndex:6},{value:"bool/Function(inputValue:string, treeNode:TreeNode)",paraId:4,tocIndex:6},{value:"Function",paraId:4,tocIndex:6},{value:"treeNodeFilterProp",paraId:4,tocIndex:6},{value:"which prop value of treeNode will be used for filter if filterTreeNode return true",paraId:4,tocIndex:6},{value:"String",paraId:4,tocIndex:6},{value:"'value'",paraId:4,tocIndex:6},{value:"treeNodeLabelProp",paraId:4,tocIndex:6},{value:"which prop value of treeNode will render as content of select",paraId:4,tocIndex:6},{value:"String",paraId:4,tocIndex:6},{value:"'title'",paraId:4,tocIndex:6},{value:"treeData",paraId:4,tocIndex:6},{value:"treeNodes data Array, if set it then you need not to construct children TreeNode. (value should be unique across the whole array)",paraId:4,tocIndex:6},{value:"array<{value,label,children, [disabled,selectable]}>",paraId:4,tocIndex:6},{value:"[]",paraId:4,tocIndex:6},{value:"treeDataSimpleMode",paraId:4,tocIndex:6},{value:`enable simple mode of treeData.(treeData should be like this: [{id:1, pId:0, value:'1', label:"test1",...},...], `,paraId:4,tocIndex:6},{value:"pId",paraId:4,tocIndex:6},{value:" is parent node's id)",paraId:4,tocIndex:6},{value:"bool/object{id:'id', pId:'pId', rootPId:null}",paraId:4,tocIndex:6},{value:"false",paraId:4,tocIndex:6},{value:"treeTitleRender",paraId:4,tocIndex:6},{value:"Custom render nodes",paraId:4,tocIndex:6},{value:"(nodeData: OptionType) => ReactNode",paraId:4,tocIndex:6},{value:"loadData",paraId:4,tocIndex:6},{value:"load data asynchronously",paraId:4,tocIndex:6},{value:"function(node)",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"getPopupContainer",paraId:4,tocIndex:6},{value:"container which popup select menu rendered into",paraId:4,tocIndex:6},{value:"function(trigger:Node):Node",paraId:4,tocIndex:6},{value:"function(){return document.body;}",paraId:4,tocIndex:6},{value:"autoClearSearchValue",paraId:4,tocIndex:6},{value:"auto clear search input value when multiple select is selected/deselected",paraId:4,tocIndex:6},{value:"boolean",paraId:4,tocIndex:6},{value:"true",paraId:4,tocIndex:6},{value:"suffixIcon",paraId:4,tocIndex:6},{value:"specify the select arrow icon",paraId:4,tocIndex:6},{value:"ReactNode | (props: TreeProps) => ReactNode",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"clearIcon",paraId:4,tocIndex:6},{value:"specify the clear icon",paraId:4,tocIndex:6},{value:"ReactNode | (props: TreeProps) => ReactNode",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"removeIcon",paraId:4,tocIndex:6},{value:"specify the remove icon",paraId:4,tocIndex:6},{value:"ReactNode | (props: TreeProps) => ReactNode",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"switcherIcon",paraId:4,tocIndex:6},{value:"specify the switcher icon",paraId:4,tocIndex:6},{value:"ReactNode | (props: TreeProps) => ReactNode",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"virtual",paraId:4,tocIndex:6},{value:"Disable virtual when ",paraId:4,tocIndex:6},{value:"false",paraId:4,tocIndex:6},{value:"false",paraId:4,tocIndex:6},{value:"-",paraId:4,tocIndex:6},{value:"note: you'd better to use ",paraId:6,tocIndex:7},{value:"treeData",paraId:6,tocIndex:7},{value:" instead of using TreeNode.",paraId:6,tocIndex:7},{value:"name",paraId:7,tocIndex:7},{value:"description",paraId:7,tocIndex:7},{value:"type",paraId:7,tocIndex:7},{value:"default",paraId:7,tocIndex:7},{value:"disabled",paraId:7,tocIndex:7},{value:"disable treeNode",paraId:7,tocIndex:7},{value:"bool",paraId:7,tocIndex:7},{value:"false",paraId:7,tocIndex:7},{value:"key",paraId:7,tocIndex:7},{value:"it's value must be unique across the tree's all TreeNode, you must set it",paraId:7,tocIndex:7},{value:"String",paraId:7,tocIndex:7},{value:"-",paraId:7,tocIndex:7},{value:"value",paraId:7,tocIndex:7},{value:"default as treeNodeFilterProp (be unique across the tree's all TreeNode)",paraId:7,tocIndex:7},{value:"String",paraId:7,tocIndex:7},{value:"''",paraId:7,tocIndex:7},{value:"title",paraId:7,tocIndex:7},{value:"tree/subTree's title",paraId:7,tocIndex:7},{value:"String/element",paraId:7,tocIndex:7},{value:"'---'",paraId:7,tocIndex:7},{value:"isLeaf",paraId:7,tocIndex:7},{value:"whether it's leaf node",paraId:7,tocIndex:7},{value:"bool",paraId:7,tocIndex:7},{value:"false",paraId:7,tocIndex:7},{value:`Optimization tips(when there are large amounts of data, like more than 5000 nodes)
`,paraId:8,tocIndex:8},{value:"Do not Expand all nodes.",paraId:9,tocIndex:8},{value:"Recommend not exist many ",paraId:9,tocIndex:8},{value:"TreeSelect",paraId:9,tocIndex:8},{value:" components in a page at the same time.",paraId:9,tocIndex:8},{value:"Recommend not use ",paraId:9,tocIndex:8},{value:"treeCheckable",paraId:9,tocIndex:8},{value:" mode, or use ",paraId:9,tocIndex:8},{value:"treeCheckStrictly",paraId:9,tocIndex:8},{value:".",paraId:9,tocIndex:8},{value:"In ",paraId:8,tocIndex:8},{value:"treeCheckable",paraId:8,tocIndex:8},{value:" mode, It has the same effect when click ",paraId:8,tocIndex:8},{value:"x",paraId:8,tocIndex:8},{value:"(node in Selection box) or uncheck in the treeNode(in dropdown panel), but the essence is not the same. So, even if both of them trigger ",paraId:8,tocIndex:8},{value:"onChange",paraId:8,tocIndex:8},{value:" method, but the parameters (the third parameter) are different. \uFF08\u4E2D\u6587\uFF1A\u5728",paraId:8,tocIndex:8},{value:"treeCheckable",paraId:8,tocIndex:8},{value:"\u6A21\u5F0F\u4E0B\uFF0C\u5DF2\u9009\u62E9\u8282\u70B9\u4E0A\u7684",paraId:8,tocIndex:8},{value:"x",paraId:8,tocIndex:8},{value:"\u5220\u9664\u64CD\u4F5C\u3001\u548C\u76F8\u5E94 treeNode \u8282\u70B9\u4E0A checkbox \u7684 uncheck \u64CD\u4F5C\uFF0C\u6700\u7EC8\u6548\u679C\u76F8\u540C\uFF0C\u4F46\u672C\u8D28\u4E0D\u4E00\u6837\u3002\u524D\u8005\u8DDF\u5F39\u51FA\u7684 tree \u7EC4\u4EF6\u53EF\u4EE5\u201C\u6BEB\u65E0\u5173\u7CFB\u201D\uFF08\u4F8B\u5982 dropdown \u6CA1\u5C55\u5F00\u8FC7\uFF0Ctree \u4E5F\u5C31\u6CA1\u6E32\u67D3\u597D\uFF09\uFF0C\u800C\u540E\u8005\u662F tree \u7EC4\u4EF6\u4E0A\u7684\u8282\u70B9 uncheck \u4E8B\u4EF6\u3002\u6240\u4EE5\u3001\u5373\u4FBF\u4E24\u8005\u90FD\u4F1A\u89E6\u53D1",paraId:8,tocIndex:8},{value:"onChange",paraId:8,tocIndex:8},{value:"\u65B9\u6CD5\u3001\u4F46\u5B83\u4EEC\u7684\u53C2\u6570\uFF08\u7B2C\u4E09\u4E2A\u53C2\u6570\uFF09\u662F\u4E0D\u540C\u7684\u3002\uFF09",paraId:8,tocIndex:8},{value:"http://localhost:8000/tests/runner.html?coverage",paraId:10,tocIndex:9},{value:"http://localhost:8000/node_modules/rc-server/node_modules/node-jscover/lib/front-end/jscoverage.html?w=http://localhost:8000/tests/runner.html?coverage",paraId:11,tocIndex:10},{value:"rc-tree-select is released under the MIT license.",paraId:12,tocIndex:11}]},25866:function(v,t){t.Z=`@import "./select.less";
@import "./tree.less";
`},45745:function(v,t){t.Z=`import Dialog from 'rc-dialog';
import 'rc-dialog/assets/index.css';
import React from 'react';
import '../assets/index.less';
import TreeSelect, { SHOW_PARENT, TreeNode } from '../src';
import { gData } from './utils/dataUtil';

function isLeaf(value) {
  if (!value) {
    return false;
  }
  let queues = [...gData];
  while (queues.length) {
    // BFS
    const item = queues.shift();
    if (item.value === value) {
      if (!item.children) {
        return true;
      }
      return false;
    }
    if (item.children) {
      queues = queues.concat(item.children);
    }
  }
  return false;
}

function findPath(value, data) {
  const sel = [];
  function loop(selected, children) {
    for (let i = 0; i < children.length; i += 1) {
      const item = children[i];
      if (selected === item.value) {
        sel.push(item);
        return;
      }
      if (item.children) {
        loop(selected, item.children);
        if (sel.length) {
          sel.push(item);
          return;
        }
      }
    }
  }
  loop(value, data);
  return sel;
}

class Demo extends React.Component {
  state = {
    tsOpen: false,
    visible: false,
    searchValue: '0-0-0-label',
    value: '0-0-0-value1',
    // value: ['0-0-0-0-value', '0-0-0-1-value', '0-0-0-2-value'],
    lv: { value: '0-0-0-value', label: 'spe label' },
    multipleValue: [],
    simpleSearchValue: 'test111',
    simpleTreeData: [
      { key: 1, pId: 0, label: 'test1', value: 'test1' },
      { key: 121, pId: 0, label: 'test2', value: 'test2' },
      { key: 11, pId: 1, label: 'test11', value: 'test11' },
      { key: 12, pId: 1, label: 'test12', value: 'test12' },
      { key: 111, pId: 11, label: 'test111', value: 'test111' },
    ],
    treeDataSimpleMode: {
      id: 'key',
      rootPId: 0,
    },
  };

  onClick = () => {
    this.setState({
      visible: true,
    });
  };

  onClose = () => {
    this.setState({
      visible: false,
    });
  };

  onSearch = (value, ...args) => {
    console.log('Do Search:', value, ...args);
    this.setState({ searchValue: value });
  };

  onChange = (value, ...rest) => {
    console.log('onChange', value, ...rest);
    this.setState({ value });
  };

  onChangeChildren = (...args) => {
    const { value: preValue } = this.state;
    console.log('onChangeChildren', ...args);
    const value = args[0];
    const pre = value ? preValue : undefined;
    this.setState({ value: isLeaf(value) ? value : pre });
  };

  onChangeLV = (value, ...args) => {
    console.log('labelInValue', value, ...args);
    if (!value) {
      this.setState({ lv: undefined });
      return;
    }
    const path = findPath(value.value, gData)
      .map(i => i.label)
      .reverse()
      .join(' > ');
    this.setState({ lv: { value: value.value, label: path } });
  };

  onMultipleChange = value => {
    console.log('onMultipleChange', value);
    this.setState({ multipleValue: value });
  };

  onSelect = (...args) => {
    // use onChange instead
    console.log(args);
  };

  onDropdownVisibleChange = visible => {
    const { value } = this.state;
    console.log(visible, value);
    if (Array.isArray(value) && value.length > 1 && value.length < 3) {
      window.alert('please select more than two item or less than one item.');
      return false;
    }
    return true;
  };

  filterTreeNode = (input, child) => String(child.props.title).indexOf(input) === 0;

  render() {
    const {
      visible,
      value,
      searchValue,
      tsOpen,
      multipleValue,
      lv,
      simpleTreeData,
      simpleSearchValue,
      treeDataSimpleMode,
    } = this.state;
    return (
      <div style={{ margin: 20 }}>
        <h2>tree-select in dialog</h2>
        <button type="button" className="btn btn-primary" onClick={this.onClick}>
          show dialog
        </button>
        {visible ? (
          <Dialog
            visible={visible}
            animation="zoom"
            maskAnimation="fade"
            onClose={this.onClose}
            // style={{ width: 600, height: 400, overflow: 'auto' }}
          >
            <div style={{ height: 600, paddingTop: 100 }}>
              <TreeSelect
                getPopupContainer={triggerNode => triggerNode.parentNode}
                style={{ width: 300 }}
                transitionName="rc-tree-select-dropdown-slide-up"
                choiceTransitionName="rc-tree-select-selection__choice-zoom"
                // dropdownStyle={{ maxHeight: 200, overflow: 'auto', zIndex: 1500 }}
                placeholder={<i>\u8BF7\u4E0B\u62C9\u9009\u62E9</i>}
                showSearch
                allowClear
                treeLine
                value={value}
                treeData={gData}
                treeNodeFilterProp="label"
                filterTreeNode={false}
                onSearch={this.onSearch}
                onChange={this.onChange}
                onSelect={this.onSelect}
              />
            </div>
          </Dialog>
        ) : null}
        <h2>single select</h2>
        <TreeSelect
          style={{ width: 300 }}
          transitionName="rc-tree-select-dropdown-slide-up"
          choiceTransitionName="rc-tree-select-selection__choice-zoom"
          // dropdownStyle={{ maxHeight: 200, overflow: 'auto' }}
          placeholder={<i>\u8BF7\u4E0B\u62C9\u9009\u62E9</i>}
          showSearch
          allowClear
          treeLine
          searchValue={searchValue}
          value={value}
          treeData={gData}
          treeNodeFilterProp="label"
          filterTreeNode={false}
          onSearch={this.onSearch}
          open={tsOpen}
          onChange={(val, ...args) => {
            console.log('onChange', val, ...args);
            this.setState({ value: val });
          }}
          onDropdownVisibleChange={v => {
            console.log('single onDropdownVisibleChange', v);
            this.setState({
              tsOpen: v,
            });
          }}
          onSelect={this.onSelect}
          onPopupScroll={evt => {
            console.log('onPopupScroll:', evt.target);
          }}
        />

        <h2>single select (just select children)</h2>
        <TreeSelect
          style={{ width: 300 }}
          transitionName="rc-tree-select-dropdown-slide-up"
          choiceTransitionName="rc-tree-select-selection__choice-zoom"
          // dropdownStyle={{ maxHeight: 200, overflow: 'auto' }}
          placeholder={<i>\u8BF7\u4E0B\u62C9\u9009\u62E9</i>}
          showSearch
          allowClear
          treeLine
          value={value}
          treeData={gData}
          treeNodeFilterProp="label"
          filterTreeNode={false}
          onChange={this.onChangeChildren}
        />

        <h2>multiple select</h2>
        <TreeSelect
          style={{ width: 300 }}
          transitionName="rc-tree-select-dropdown-slide-up"
          choiceTransitionName="rc-tree-select-selection__choice-zoom"
          // dropdownStyle={{ maxHeight: 200, overflow: 'auto' }}
          placeholder={<i>\u8BF7\u4E0B\u62C9\u9009\u62E9</i>}
          multiple
          value={multipleValue}
          treeData={gData}
          treeNodeFilterProp="title"
          onChange={this.onMultipleChange}
          onSelect={this.onSelect}
          allowClear
        />

        <h2>check select</h2>
        <TreeSelect
          open
          allowClear
          className="check-select"
          transitionName="rc-tree-select-dropdown-slide-up"
          choiceTransitionName="rc-tree-select-selection__choice-zoom"
          style={{ width: 300 }}
          // dropdownStyle={{ height: 200, overflow: 'auto' }}
          popupAlign={{
            overflow: { adjustY: 0, adjustX: 0 },
            offset: [0, 2],
          }}
          onDropdownVisibleChange={this.onDropdownVisibleChange}
          placeholder={<i>\u8BF7\u4E0B\u62C9\u9009\u62E9</i>}
          treeLine
          maxTagTextLength={10}
          value={value}
          autoClearSearchValue
          treeData={gData}
          treeNodeFilterProp="title"
          treeCheckable
          showCheckedStrategy={SHOW_PARENT}
          onChange={this.onChange}
          onSelect={this.onSelect}
          maxTagCount="responsive"
          maxTagPlaceholder={valueList => {
            // console.log('Max Tag Rest Value:', valueList);
            return \`\${valueList.length} rest...\`;
          }}
        />

        <h2>labelInValue & show path</h2>
        <TreeSelect
          style={{ width: 500 }}
          transitionName="rc-tree-select-dropdown-slide-up"
          choiceTransitionName="rc-tree-select-selection__choice-zoom"
          // dropdownStyle={{ maxHeight: 200, overflow: 'auto' }}
          placeholder={<i>\u8BF7\u4E0B\u62C9\u9009\u62E9</i>}
          showSearch
          allowClear
          treeLine
          value={lv}
          labelInValue
          treeData={gData}
          treeNodeFilterProp="label"
          filterTreeNode={false}
          onChange={this.onChangeLV}
        />

        <h2>use treeDataSimpleMode</h2>
        <TreeSelect
          style={{ width: 300 }}
          // dropdownStyle={{ maxHeight: 200, overflow: 'auto' }}
          placeholder={<i>\u8BF7\u4E0B\u62C9\u9009\u62E9</i>}
          // treeLine
          maxTagTextLength={10}
          searchValue={simpleSearchValue}
          onSearch={val => {
            this.setState({ simpleSearchValue: val });
          }}
          value={value}
          treeData={simpleTreeData}
          treeNodeFilterProp="title"
          treeDataSimpleMode={treeDataSimpleMode}
          treeCheckable
          showCheckedStrategy={SHOW_PARENT}
          onChange={this.onChange}
          onSelect={(...args) => {
            this.setState({ simpleSearchValue: '' });
            this.onSelect(...args);
          }}
        />

        <h2>Testing in extreme conditions (Boundary conditions test) </h2>
        <TreeSelect
          style={{ width: 200 }}
          popupStyle={{ maxHeight: 200, overflow: 'auto' }}
          defaultValue="leaf1"
          multiple
          treeCheckable
          showCheckedStrategy={SHOW_PARENT}
          treeDefaultExpandAll
          treeData={[
            { key: '', value: '', label: 'empty value', children: [] },
            {
              key: '0',
              value: '0',
              label: '0 label',
              children: [
                { key: '00', value: '00', label: '00 label', children: [] },
                { key: '01', value: '01', label: '01 label', children: [] },
              ],
            },
          ]}
          onChange={(val, ...args) => console.log(val, ...args)}
        />

        <h2>use TreeNode Component (not recommend)</h2>
        <TreeSelect
          style={{ width: 200 }}
          // dropdownStyle={{ maxHeight: 200, overflow: 'auto' }}
          defaultValue="leaf1"
          treeDefaultExpandAll
          treeNodeFilterProp="title"
          filterTreeNode={this.filterTreeNode}
          onChange={(val, ...args) => console.log(val, ...args)}
        >
          <TreeNode value="" title="parent 1" key="">
            <TreeNode value="parent 1-0" title="parent 1-0" key="0-1-0">
              <TreeNode value="leaf1" title="my leaf" key="random" />
              <TreeNode value="leaf2" title="your leaf" key="random1" disabled />
            </TreeNode>
            <TreeNode value="parent 1-1" title="parent 1-1" key="0-1-1">
              <TreeNode
                value="sss"
                title={<span style={{ color: 'red' }}>sss</span>}
                key="random3"
              />
              <TreeNode value="same value1" title="same txtle" key="0-1-1-1">
                <TreeNode
                  value="same value10"
                  title="same titlexd"
                  key="0-1-1-1-0"
                  style={{ color: 'red', background: 'green' }}
                />
              </TreeNode>
            </TreeNode>
          </TreeNode>
          <TreeNode value="same value2" title="same title" key="0-2">
            <TreeNode value="2same value" title="2same title" key="0-2-0" />
          </TreeNode>
          <TreeNode value="same value3" title="same title" key="0-3" />
        </TreeSelect>

        <h2>title render</h2>
        <TreeSelect<{ label: string }>
          open
          style={{ width: 300 }}
          treeData={gData}
          treeTitleRender={node => node.label + 'ok'}
        />
      </div>
    );
  }
}

export default Demo;
`},13371:function(v,t){t.Z=`import '../assets/index.less';
import React from 'react';
import TreeSelect, { SHOW_PARENT } from '../src';
import Gen from './utils/big-data-generator';

class Demo extends React.Component {
  state = {
    gData: [],
    gData1: [],
    value: '',
    value1: '',
  };

  onChange = value => {
    console.log('onChange', value);
    this.setState({ value });
  };

  onChangeStrictly = value1 => {
    console.log('onChangeStrictly', value1);
    const ind = parseInt(\`\${Math.random() * 3}\`, 10);
    value1.push({
      value: \`0-0-0-\${ind}-value\`,
      label: \`0-0-0-\${ind}-label\`,
      halfChecked: true,
    });
    this.setState({
      value1,
    });
  };

  onGen = data => {
    this.setState({
      gData: data,
      gData1: [...data],
      value: '0-0-0-value',
      value1: [
        { value: '0-0-value', label: '0-0-label', halfChecked: true },
        { value: '0-0-0-value', label: '0-0-0-label' },
      ],
    });
  };

  render() {
    const { value1, gData1, value, gData } = this.state;

    // console.log('>>>', gData, gData1, value1);

    return (
      <div style={{ padding: '0 20px' }}>
        <Gen onGen={this.onGen} />
        <div style={{ display: 'flex' }}>
          <div style={{ marginRight: 20 }}>
            <h3>normal check</h3>
            <TreeSelect
              style={{ width: 300 }}
              // dropdownStyle={{ maxHeight: 200, overflow: 'auto' }}
              treeData={gData}
              treeLine
              value={value}
              placeholder={<i>\u8BF7\u4E0B\u62C9\u9009\u62E9</i>}
              treeCheckable
              showCheckedStrategy={SHOW_PARENT}
              onChange={this.onChange}
            />
          </div>
          <div>
            <h3>checkStrictly</h3>
            <TreeSelect
              style={{ width: 300 }}
              // dropdownStyle={{ maxHeight: 200, overflow: 'auto' }}
              treeData={gData1}
              treeLine
              value={value1}
              placeholder={<i>\u8BF7\u4E0B\u62C9\u9009\u62E9</i>}
              treeCheckable
              treeCheckStrictly
              showCheckedStrategy={SHOW_PARENT}
              onChange={this.onChangeStrictly}
            />
          </div>
        </div>
      </div>
    );
  }
}

export default Demo;
`},19760:function(v,t){t.Z=`import '../assets/index.less';
import React from 'react';
import 'rc-dialog/assets/index.css';
import TreeSelect, { TreeNode } from '../src';

class Demo extends React.Component {
  state = {
    treeExpandedKeys: [],
  };

  onTreeExpand = treeExpandedKeys => {
    this.setState({
      treeExpandedKeys,
    });
  };

  treeExpandedKeys = () => {
    this.setState({
      treeExpandedKeys: ['000', '0-1-0'],
    });
  };

  render() {
    const { treeExpandedKeys } = this.state;

    return (
      <div>
        <h2>Conrolled treeExpandedKeys</h2>
        <TreeSelect
          style={{ width: 200 }}
          // dropdownStyle={{ maxHeight: 200, overflow: 'auto' }}
          treeExpandedKeys={treeExpandedKeys}
          onTreeExpand={this.onTreeExpand}
        >
          <TreeNode value="" title="parent 1" key="000">
            <TreeNode value="parent 1-0" title="parent 1-0" key="0-1-0">
              <TreeNode value="leaf1" title="my leaf" key="random" />
              <TreeNode
                value="leaf2"
                title="your leaf"
                key="random1"
                disabled
              />
            </TreeNode>
            <TreeNode value="parent 1-1" title="parent 1-1" key="0-1-1">
              <TreeNode
                value="sss"
                title={<span style={{ color: 'red' }}>sss</span>}
                key="random3"
              />
              <TreeNode value="same value1" title="same txtle" key="0-1-1-1">
                <TreeNode
                  value="same value10"
                  title="same titlexd"
                  key="0-1-1-1-0"
                  style={{ color: 'red', background: 'green' }}
                />
              </TreeNode>
            </TreeNode>
          </TreeNode>
          <TreeNode value="same value2" title="same title" key="0-2">
            <TreeNode value="2same value" title="2same title" key="0-2-0" />
          </TreeNode>
          <TreeNode value="same value3" title="same title" key="0-3" />
        </TreeSelect>
        <button type="button" onClick={this.treeExpandedKeys}>
          Set treeExpandedKeys
        </button>
      </div>
    );
  }
}

export default Demo;
`},79898:function(v,t){t.Z=`import '../assets/index.less';
import React from 'react';
import 'rc-dialog/assets/index.css';
import TreeSelect from '../src';
import { gData } from './utils/dataUtil';

const bubblePath =
  'M632 888H392c-4.4 0-8 3.6-8 8v32c0 ' +
  '17.7 14.3 32 32 32h192c17.7 0 32-14.3 32-32v-3' +
  '2c0-4.4-3.6-8-8-8zM512 64c-181.1 0-328 146.9-3' +
  '28 328 0 121.4 66 227.4 164 284.1V792c0 17.7 1' +
  '4.3 32 32 32h264c17.7 0 32-14.3 32-32V676.1c98' +
  '-56.7 164-162.7 164-284.1 0-181.1-146.9-328-32' +
  '8-328z m127.9 549.8L604 634.6V752H420V634.6l-3' +
  '5.9-20.8C305.4 568.3 256 484.5 256 392c0-141.4' +
  ' 114.6-256 256-256s256 114.6 256 256c0 92.5-49' +
  '.4 176.3-128.1 221.8z';

const clearPath =
  'M793 242H366v-74c0-6.7-7.7-10.4-12.9' +
  '-6.3l-142 112c-4.1 3.2-4.1 9.4 0 12.6l142 112c' +
  '5.2 4.1 12.9 0.4 12.9-6.3v-74h415v470H175c-4.4' +
  ' 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h618c35.3 0 64-' +
  '28.7 64-64V306c0-35.3-28.7-64-64-64z';

const arrowPath =
  'M765.7 486.8L314.9 134.7c-5.3-4.1' +
  '-12.9-0.4-12.9 6.3v77.3c0 4.9 2.3 9.6 6.1 12.6l36' +
  '0 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6' +
  '.7 7.7 10.4 12.9 6.3l450.8-352.1c16.4-12.8 16.4-3' +
  '7.6 0-50.4z';

const getSvg = (path, iStyle = {}, style = {}) => (
  <i style={iStyle}>
    <svg
      viewBox="0 0 1024 1024"
      width="1em"
      height="1em"
      fill="currentColor"
      style={{ verticalAlign: '-.125em', ...style }}
    >
      <path d={path} />
    </svg>
  </i>
);

const switcherIcon = obj => {
  if (obj.isLeaf) {
    return getSvg(
      arrowPath,
      { cursor: 'pointer', backgroundColor: 'white' },
      { transform: 'rotate(270deg)' },
    );
  }
  return getSvg(
    arrowPath,
    { cursor: 'pointer', backgroundColor: 'white' },
    { transform: \`rotate(\${obj.expanded ? 90 : 0}deg)\` },
  );
};

const suffixIcon = getSvg(bubblePath);
const clearIcon = getSvg(clearPath);
const removeIcon = getSvg(clearPath);

const iconProps = {
  suffixIcon,
  clearIcon,
  removeIcon,
  switcherIcon,
};

const iconPropsFunction = {
  suffixIcon: () => suffixIcon,
  clearIcon: () => clearIcon,
  removeIcon: () => removeIcon,
  switcherIcon,
};

function Demo() {
  return (
    <div className="custom-icon-demo">
      <h2>Single</h2>
      <TreeSelect
        treeData={gData}
        placeholder={<span>Please Select</span>}
        transitionName="rc-tree-select-dropdown-slide-up"
        style={{ width: 300 }}
        popupStyle={{ maxHeight: 200, overflow: 'auto', zIndex: 1500 }}
        showSearch
        allowClear
        {...iconProps}
      />
      <br />
      <h2>Multiple</h2>
      <TreeSelect
        treeData={gData}
        multiple
        placeholder={<span>Please Select</span>}
        transitionName="rc-tree-select-dropdown-slide-up"
        style={{ width: 300 }}
        popupStyle={{ maxHeight: 200, overflow: 'auto', zIndex: 1500 }}
        showSearch
        allowClear
        {...iconPropsFunction}
      />
    </div>
  );
}

export default Demo;
`},91938:function(v,t){t.Z=`import React from 'react';
import TreeSelect from '../src';
import '../assets/index.less';

const treeData = [
  { key: '0', value: '0', title: 'label0' },
  { key: '1', value: '1', title: 'label1' },
];

const createSelect = props => <TreeSelect treeData={treeData} labelInValue {...props} />;

export default () =>
  createSelect({
    maxTagCount: 1,
    defaultValue: {
      value: '0',
      label: '2333',
    },
  });
`},73633:function(v,t){t.Z=`import '../assets/index.less';
import React from 'react';
import TreeSelect from '../src';

const { SHOW_PARENT } = TreeSelect;

const treeData = [
  {
    label: 'Node1',
    value: '0-0',
    key: '0-0',
    children: [
      {
        label: 'Child Node1',
        value: '0-0-0',
        key: '0-0-0',
      },
    ],
  },
  {
    label: 'Node2',
    value: '0-1',
    key: '0-1',
    children: [
      {
        label: 'Child Node3',
        value: '0-1-0',
        key: '0-1-0',
      },
      {
        label: 'Child Node4',
        value: '0-1-1',
        key: '0-1-1',
      },
      {
        label: 'Child Node5',
        value: '0-1-2',
        key: '0-1-2',
      },
    ],
  },
];

class Demo extends React.Component {
  state = {
    value: ['0-0-0'],
    disabled: false,
  };

  onChange = (value, ...args) => {
    console.log('onChange ', value, args);
    this.setState({ value });
  };

  switch = checked => {
    this.setState({ disabled: checked });
  };

  render() {
    const { disabled, value } = this.state;
    const tProps = {
      treeData,
      disabled,
      value,
      onChange: this.onChange,
      multiple: true,
      allowClear: true,
      treeCheckable: true,
      showCheckedStrategy: SHOW_PARENT,
      style: {
        width: 300,
      },
    };
    return (
      <div>
        <TreeSelect {...tProps} />
        <input type="checkbox" onChange={e => this.switch(e.target.checked)} /> \u7981\u7528
      </div>
    );
  }
}

export default Demo;
/* eslint-enable */
`},86272:function(v,t){t.Z=`import '../assets/index.less';
import React from 'react';
import TreeSelect from '../src';
import { getNewTreeData, generateTreeNodes } from './utils/dataUtil';

function getTreeData() {
  return [
    { label: 'pNode 01', value: '0-0', key: '0-0' },
    { label: 'pNode 02', value: '0-1', key: '0-1' },
    { label: 'pNode 03', value: '0-2', key: '0-2', isLeaf: true },
  ];
}

class Demo extends React.Component {
  static propTypes = {};

  state = {
    treeData: getTreeData(),
    // value: '0-0',
    value: { value: '0-0-0-value', label: '0-0-0-label' },
    loadedKeys: [],
  };

  onChange = value => {
    console.log(value);
    this.setState({
      value,
    });
  };

  loadData = treeNode => {
    console.log('trigger load:', treeNode);
    return new Promise<void>(resolve => {
      setTimeout(() => {
        let { treeData } = this.state;
        treeData = treeData.slice();
        getNewTreeData(
          treeData,
          treeNode.props.eventKey,
          generateTreeNodes(treeNode),
          2,
        );
        this.setState({ treeData });
        resolve();
      }, 500);
    });
  };

  onTreeLoad = loadedKeys => {
    this.setState({ loadedKeys });
  };

  onResetTree = () => {
    this.setState({
      treeData: getTreeData(),
    });
  };

  onResetLoadedKeys = () => {
    this.setState({
      loadedKeys: [],
    });
  };

  render() {
    const { treeData, value, loadedKeys } = this.state;
    return (
      <div style={{ padding: '10px 30px' }}>
        <h2>dynamic render</h2>
        <TreeSelect
          style={{ width: 300 }}
          treeData={treeData}
          labelInValue
          value={value}
          onChange={this.onChange}
          loadData={this.loadData}
        />
        <h2>Controlled</h2>
        <TreeSelect
          style={{ width: 300 }}
          treeData={treeData}
          labelInValue
          showSearch
          value={value}
          treeLoadedKeys={loadedKeys}
          onChange={this.onChange}
          loadData={this.loadData}
          onTreeLoad={this.onTreeLoad}
        />

        <button type="button" onClick={this.onResetTree}>
          Reset Tree
        </button>
        <button type="button" onClick={this.onResetLoadedKeys}>
          Reset LoadedKeys
        </button>
      </div>
    );
  }
}

export default Demo;
`},14610:function(v,t){t.Z=`import '../assets/index.less';
import React from 'react';
import 'rc-dialog/assets/index.css';
import TreeSelect from '../src';

export default () => {
  return (
    <div>
      <h2>basic</h2>
      <TreeSelect
        treeDefaultExpandAll
        treeData={[
          {
            myLabel: 'Parent',
            myValue: 'parent',
            myChildren: [
              {
                myLabel: 'Sub 1',
                myValue: 'sub_1',
              },
              {
                myLabel: 'Sub 2',
                myValue: 'sub_2',
              },
            ],
          },
        ]}
        fieldNames={{
          label: 'myLabel',
          value: 'myValue',
          children: 'myChildren',
        }}
      />

      <h2>title render</h2>
      <TreeSelect
        treeDefaultExpandAll
        treeTitleRender={node => <span>{node.myLabel}</span>}
        treeData={[
          {
            myLabel: 'Parent',
            myValue: 'parent',
            myChildren: [
              {
                myLabel: 'Sub 1',
                myValue: 'sub_1',
              },
              {
                myLabel: 'Sub 2',
                myValue: 'sub_2',
              },
            ],
          },
        ]}
        fieldNames={{
          label: 'myLabel',
          value: 'myValue',
          children: 'myChildren',
        }}
      />
    </div>
  );
};
`},33073:function(v,t){t.Z=`import '../assets/index.less';
import React from 'react';
import TreeSelect, { SHOW_PARENT } from '../src';
import { gData } from './utils/dataUtil';

console.log('TreeData:', gData);

class Demo extends React.Component {
  state = {
    value: '11',
    // value: ['0-0-0-0-value', '0-0-0-1-value', '0-0-0-2-value'],
    simpleTreeData: [
      { key: 1, pId: 0, label: 'a', value: 'a' },
      { key: 11, pId: 1, label: 'a12', value: 'a12', disabled: true },
      { key: 111, pId: 11, label: 'a00', value: 'a00', selectable: false },
      { key: 2, pId: 0, label: 'b', value: 'b' },
      { key: 20, pId: 2, label: 'b10', value: 'b10' },
      { key: 21, pId: 2, label: 'b1', value: 'b1' },
      { key: 22, pId: 2, label: 'b12', value: 'b12' },
    ],
    treeDataSimpleMode: {
      id: 'key',
      rootPId: 0,
    },
  };

  onChange = value => {
    const { simpleTreeData } = this.state;
    if (value.length === 1) {
      // return;
    }
    console.log('onChange', value, simpleTreeData);
    this.setState({ value });
  };

  onSelect = () => {
    // use onChange instead
    // console.log(arguments);
  };

  onDataChange = () => {
    const { simpleTreeData } = this.state;
    const data = simpleTreeData.slice();
    data.forEach(i => {
      if (i.key === 11) {
        // eslint-disable-next-line no-param-reassign
        delete i.disabled;
      }
      if (i.key === 20) {
        // eslint-disable-next-line no-param-reassign
        i.disabled = true;
      }
    });
    this.setState({ simpleTreeData: data });
  };

  render() {
    const { value, simpleTreeData, treeDataSimpleMode } = this.state;
    return (
      <div style={{ margin: 20 }}>
        <h2>check select</h2>
        <TreeSelect
          style={{ width: 300 }}
          transitionName="rc-tree-select-dropdown-slide-up"
          choiceTransitionName="rc-tree-select-selection__choice-zoom"
          // dropdownStyle={{ height: 200, overflow: 'auto' }}
          // dropdownPopupAlign={{
          //   overflow: { adjustY: 0, adjustX: 0 },
          //   offset: [0, 2],
          // }}
          placeholder={<i>\u8BF7\u4E0B\u62C9\u9009\u62E9</i>}
          treeLine
          maxTagTextLength={10}
          value={value}
          treeData={gData}
          treeNodeFilterProp="title"
          treeCheckable
          onChange={this.onChange}
          onSelect={this.onSelect}
        />

        <h2>use treeDataSimpleMode</h2>
        <TreeSelect
          style={{ width: 300 }}
          popupStyle={{ maxHeight: 200, overflow: 'auto' }}
          placeholder={<i>\u8BF7\u4E0B\u62C9\u9009\u62E9</i>}
          treeLine
          maxTagTextLength={10}
          inputValue={null}
          value={value}
          treeData={simpleTreeData}
          treeDefaultExpandAll
          treeNodeFilterProp="title"
          treeDataSimpleMode={treeDataSimpleMode}
          treeCheckable
          showCheckedStrategy={SHOW_PARENT}
          onChange={this.onChange}
          onSelect={this.onSelect}
        />
        <button type="button" onClick={this.onDataChange}>
          change data
        </button>
      </div>
    );
  }
}

export default Demo;
`},43335:function(v,t){t.Z=`import React, { Component } from 'react';
import Select from '@rc-component/select';
import Form, { useForm, Field } from 'rc-field-form';
import TreeSelect from '../src';
import '@rc-component/select/assets/index.less';
import '../assets/index.less';
import { gData } from './utils/dataUtil';

const { Option } = Select;

const regionStyle = {
  border: '1px solid red',
  marginTop: 10,
  padding: 10,
};

const errorStyle = {
  color: 'red',
  marginTop: 10,
  padding: 10,
};

class TreeSelectInput extends Component<{
  onChange?: (value: string[]) => void;
  style: React.CSSProperties;
}> {
  onChange = (value, ...args) => {
    console.log(value, ...args);
    const { props } = this;
    if (props.onChange) {
      props.onChange(value);
    }
  };

  render() {
    return <TreeSelect {...this.props} onChange={this.onChange} />;
  }
}

const Demo = () => {
  const [form] = useForm();

  const onFinish = values => {
    console.log('Submit:', values);
  };

  const onReset = () => {
    form.resetFields();
  };

  const tProps = {
    multiple: true,
    treeData: gData,
    treeCheckable: true,
  };

  return (
    <div style={{ margin: 20 }}>
      <h2>validity</h2>
      <Form
        form={form}
        onFinish={onFinish}
        initialValues={{
          'tree-select': ['0-0-0-value'],
          'tree-select1': ['0-0-0-value'],
          select: ['jack'],
        }}
      >
        <div style={regionStyle}>
          <div>
            <p style={{ color: 'blue' }}>no onChange</p>
            <Field
              name="tree-select"
              rules={[
                {
                  required: true,
                  type: 'array',
                  message: 'tree-select \u9700\u8981\u5FC5\u586B',
                },
              ]}
            >
              {(control, { errors }) => (
                <div>
                  <TreeSelect {...tProps} {...control} style={{ width: 300 }} />

                  <p style={errorStyle}>{errors.join(',')}</p>
                </div>
              )}
            </Field>
          </div>
        </div>

        <div style={regionStyle}>
          <div>
            <p style={{ color: 'blue' }}>custom onChange</p>
            <Field
              name="tree-select1"
              rules={[
                {
                  required: true,
                  type: 'array',
                  message: 'tree-select1 \u9700\u8981\u5FC5\u586B',
                },
              ]}
            >
              {(control, { errors }) => (
                <div>
                  <TreeSelectInput {...tProps} {...control} style={{ width: 300 }} />

                  <p style={errorStyle}>{errors.join(',')}</p>
                </div>
              )}
            </Field>
          </div>
        </div>

        <div style={regionStyle}>
          <div>
            <Field
              name="select"
              rules={[{ required: true, type: 'array', message: 'select \u9700\u8981\u5FC5\u586B' }]}
            >
              {(control, { errors }) => (
                <div>
                  <Select style={{ width: 200 }} {...control} allowClear mode="multiple">
                    <Option value="jack">jack</Option>
                    <Option value="lucy">lucy</Option>
                    <Option value="disabled" disabled>
                      disabled
                    </Option>
                    <Option value="yiminghe">yiminghe</Option>
                  </Select>

                  <p style={errorStyle}>{errors.join(',')}</p>
                </div>
              )}
            </Field>
          </div>
        </div>

        <div style={regionStyle}>
          <button type="button" onClick={onReset}>
            reset
          </button>
          &nbsp;
          <input type="submit" value="submit" />
        </div>
      </Form>
    </div>
  );
};

export default Demo;
`},67959:function(v,t){t.Z=`import React, { useState } from 'react';
import TreeSelect from '../src';

export default () => {
  const [value, setValue] = useState<string[]>(['1']);
  const [checkValue, setCheckValue] = useState<string[]>(['1']);

  const treeData = [
    {
      key: '1',
      value: '1',
      title: '1',
      children: [
        {
          key: '1-1',
          value: '1-1',
          title: '1-1',
        },
        {
          key: '1-2',
          value: '1-2',
          title: '1-2',
          disabled: true,
          children: [
            {
              key: '1-2-1',
              value: '1-2-1',
              title: '1-2-1',
              disabled: true,
            },
            {
              key: '1-2-2',
              value: '1-2-2',
              title: '1-2-2',
            },
          ],
        },
        {
          key: '1-3',
          value: '1-3',
          title: '1-3',
        },
      ],
    },
    {
      key: '2',
      value: '2',
      title: '2',
    },
    {
      key: '3',
      value: '3',
      title: '3',
    },
    {
      key: '4',
      value: '4',
      title: '4',
    },
  ];

  const onChange = (val: string[]) => {
    setValue(val);
  };

  const onCheckChange = (val: string[]) => {
    setCheckValue(val);
  };

  return (
    <>
      <h2>multiple with maxCount</h2>
      <TreeSelect
        style={{ width: 300 }}
        fieldNames={{ value: 'value', label: 'title' }}
        multiple
        maxCount={3}
        treeData={treeData}
      />
      <h2>checkable with maxCount</h2>
      <TreeSelect
        style={{ width: 300 }}
        treeCheckable
        // showCheckedStrategy="SHOW_ALL"
        // showCheckedStrategy="SHOW_PARENT"
        maxCount={4}
        treeData={treeData}
        onChange={onChange}
        value={value}
      />
      <h2>checkable with maxCount and treeCheckStrictly</h2>
      <TreeSelect
        style={{ width: 300 }}
        multiple
        treeCheckable
        treeCheckStrictly
        maxCount={3}
        treeData={treeData}
        onChange={onCheckChange}
        value={checkValue}
      />
    </>
  );
};
`},73007:function(v,t){t.Z=`import '../assets/index.less';
import React from 'react';
import TreeSelect, { TreeNode } from '../src';

const treeData = [
  {
    title: 'a list is option only',
    showTitle: 'Node2',
    value: '0-1',
  },
];

function Demo() {
  return (
    <>
      <TreeSelect
        style={{ width: '100%' }}
        treeDefaultExpandAll
        treeData={treeData}
        treeNodeLabelProp="showTitle"
      />
      <TreeSelect style={{ width: '100%' }} treeDefaultExpandAll treeNodeLabelProp="showTitle">
        <TreeNode value="0-0" title="a list is option only" showTitle="Node2" />
      </TreeSelect>
    </>
  );
}

export default Demo;
`},30390:function(v,t){t.Z=`import React from 'react';
import { generateData, calcTotal } from './dataUtil';

interface GenProps {
  onGen: (data: any[]) => void;
  x: number;
  y: number;
  z: number;
}

class Gen extends React.Component<GenProps> {
  static defaultProps = {
    onGen: () => {},
    x: 20,
    y: 18,
    z: 1,
  };

  xRef = React.createRef<HTMLInputElement>();

  yRef = React.createRef<HTMLInputElement>();

  zRef = React.createRef<HTMLInputElement>();

  state = {
    nums: '',
  };

  componentDidMount() {
    const { onGen } = this.props;
    const vals = this.getVals();
    onGen(generateData(vals.x, vals.y, vals.z));
  }

  onGen = e => {
    const { onGen } = this.props;
    e.preventDefault();
    const vals = this.getVals();
    onGen(generateData(vals.x, vals.y, vals.z));
    this.setState({
      nums: calcTotal(vals.x, vals.y, vals.z),
    });
  };

  getVals = () => ({
    x: parseInt(this.xRef.current.value, 10),
    y: parseInt(this.yRef.current.value, 10),
    z: parseInt(this.zRef.current.value, 10),
  });

  getDefaultValue = (key: string) => {
    if (key in this.props) {
      return String(this.props[key]);
    }
    return null;
  };

  render() {
    const { nums } = this.state;
    const { x, y, z } = this.props;
    return (
      <div style={{ padding: '0 20px' }}>
        <h2>big data generator</h2>
        <form onSubmit={this.onGen}>
          <span style={{ marginRight: 10 }}>
            x:{' '}
            <input
              ref={this.xRef}
              defaultValue={this.getDefaultValue('x')}
              type="number"
              min="1"
              required
              style={{ width: 50 }}
            />
          </span>
          <span style={{ marginRight: 10 }}>
            y:{' '}
            <input
              ref={this.yRef}
              defaultValue={this.getDefaultValue('y')}
              type="number"
              min="1"
              required
              style={{ width: 50 }}
            />
          </span>
          <span style={{ marginRight: 10 }}>
            z:{' '}
            <input
              ref={this.zRef}
              defaultValue={this.getDefaultValue('z')}
              type="number"
              min="1"
              required
              style={{ width: 50 }}
            />
          </span>
          <button type="submit">Generate</button>
          <p>total nodes: {nums || calcTotal(x, y, z)}</p>
        </form>
        <p style={{ fontSize: 12 }}>
          x\uFF1A\u6BCF\u4E00\u7EA7\u4E0B\u7684\u8282\u70B9\u603B\u6570\u3002y\uFF1A\u6BCF\u7EA7\u8282\u70B9\u91CC\u6709y\u4E2A\u8282\u70B9\u3001\u5B58\u5728\u5B50\u8282\u70B9\u3002z\uFF1A\u6811\u7684level\u5C42\u7EA7\u6570\uFF080\u8868\u793A\u4E00\u7EA7\uFF09
        </p>
      </div>
    );
  }
}

export default Gen;
`},37756:function(v,t){t.Z=`/* eslint-disable no-plusplus, no-mixed-operators, no-underscore-dangle */
export function generateData(x = 3, y = 2, z = 1, gData = []) {
  // x\uFF1A\u6BCF\u4E00\u7EA7\u4E0B\u7684\u8282\u70B9\u603B\u6570\u3002y\uFF1A\u6BCF\u7EA7\u8282\u70B9\u91CC\u6709y\u4E2A\u8282\u70B9\u3001\u5B58\u5728\u5B50\u8282\u70B9\u3002z\uFF1A\u6811\u7684level\u5C42\u7EA7\u6570\uFF080\u8868\u793A\u4E00\u7EA7\uFF09
  function _loop(_level, _preKey?, _tns?) {
    const preKey = _preKey || '0';
    const tns = _tns || gData;

    const children = [];
    for (let i = 0; i < x; i++) {
      const key = \`\${preKey}-\${i}\`;
      tns.push({
        label: \`\${key}-label\`,
        value: \`\${key}-value\`,
        key,
        disabled: key === '0-0-0-1' || false,
      });
      if (i < y) {
        children.push(key);
      }
    }
    if (_level < 0) {
      return tns;
    }
    const __level = _level - 1;
    children.forEach((key, index) => {
      tns[index].children = [];
      return _loop(__level, key, tns[index].children);
    });

    return null;
  }
  _loop(z);
  return gData;
}
export function calcTotal(x = 3, y = 2, z = 1) {
  /* eslint no-param-reassign:0 */
  const rec = n => (n >= 0 ? x * y ** n-- + rec(n) : 0);
  return rec(z + 1);
}
console.log('\u603B\u8282\u70B9\u6570\uFF08\u5355\u4E2Atree\uFF09\uFF1A', calcTotal());
export const gData = generateData();

export function generateTreeNodes(treeNode) {
  const arr = [];
  const key = treeNode.props.eventKey;
  for (let i = 0; i < 3; i++) {
    arr.push({ label: \`\${key}-\${i}-label\`, value: \`\${key}-\${i}-value\`, key: \`\${key}-\${i}\` });
  }
  return arr;
}

function setLeaf(treeData, curKey, level) {
  const loopLeaf = (data, lev) => {
    const l = lev - 1;
    data.forEach(item => {
      if (
        item.key.length > curKey.length
          ? item.key.indexOf(curKey) !== 0
          : curKey.indexOf(item.key) !== 0
      ) {
        return;
      }
      if (item.children) {
        loopLeaf(item.children, l);
      } else if (l < 1) {
        item.isLeaf = true;
      }
    });
  };
  loopLeaf(treeData, level + 1);
}

export function getNewTreeData(treeData, curKey, child, level) {
  const loop = data => {
    if (level < 1 || curKey.length - 3 > level * 2) return;
    data.forEach(item => {
      if (curKey.indexOf(item.key) === 0) {
        if (item.children) {
          loop(item.children);
        } else {
          item.children = child;
        }
      }
    });
  };
  loop(treeData);
  setLeaf(treeData, curKey, level);
}

function loopData(data, callback) {
  const loop = (d, level = '0') => {
    d.forEach((item, index) => {
      const pos = \`\${level}-\${index}\`;
      if (item.children) {
        loop(item.children, pos);
      }
      callback(item, index, pos);
    });
  };
  loop(data);
}

function isPositionPrefix(smallPos, bigPos) {
  if (bigPos.length < smallPos.length) {
    return false;
  }
  // attention: "0-0-1" "0-0-10"
  if (bigPos.length > smallPos.length && bigPos.charAt(smallPos.length) !== '-') {
    return false;
  }
  return bigPos.substr(0, smallPos.length) === smallPos;
}
// console.log(isPositionPrefix("0-1", "0-10-1"));

export function getFilterValue(val, sVal, delVal) {
  const allPos = [];
  const delPos = [];
  loopData(gData, (item, index, pos) => {
    if (sVal.indexOf(item.value) > -1) {
      allPos.push(pos);
    }
    if (delVal.indexOf(item.value) > -1) {
      delPos.push(pos);
    }
  });
  const newPos = [];
  delPos.forEach(item => {
    allPos.forEach(i => {
      if (isPositionPrefix(item, i) || isPositionPrefix(i, item)) {
        // \u8FC7\u6EE4\u6389 \u7236\u7EA7\u8282\u70B9 \u548C \u6240\u6709\u5B50\u8282\u70B9\u3002
        // \u56E0\u4E3A node\u8282\u70B9 \u4E0D\u9009\u65F6\uFF0C\u5176 \u7236\u7EA7\u8282\u70B9 \u548C \u6240\u6709\u5B50\u8282\u70B9 \u90FD\u4E0D\u9009\u3002
        return;
      }
      newPos.push(i);
    });
  });
  const newVal = [];
  if (newPos.length) {
    loopData(gData, (item, index, pos) => {
      if (newPos.indexOf(pos) > -1) {
        newVal.push(item.value);
      }
    });
  }
  return newVal;
}
`},33149:function(v,t){t.Z=`import '../assets/index.less';
import React from 'react';
import TreeSelect, { TreeNode } from '../src';

function Demo() {
  return (
    <TreeSelect style={{ width: 120 }} dropdownMatchSelectWidth={false} treeDefaultExpandAll>
      <TreeNode value="parent 1" title="parent 1">
        <TreeNode value="parent 1-0 sdfsdfsdsdfsd" title="parent 1-0 sdfsdfsd">
          <TreeNode value="leaf1  sdfsdf" title="leaf1" />
          <TreeNode value="leaf2 sdfsdfsdf" title="leaf2" />
        </TreeNode>
        <TreeNode value="parent 1-1 sdfsdfsdf" title="parent 1-1 sdfsdfsd">
          <TreeNode value="leaf3" title={<b style={{ color: '#08c' }}>leaf3</b>} />
        </TreeNode>
        <TreeNode value="parent 1-2 sdfsdfsdf" title="parent 1-2 sdfsdfsd">
          <TreeNode value="leaf3" title={<b style={{ color: '#08c' }}>leaf3</b>} />
        </TreeNode>
        <TreeNode value="parent 1-3 sdfsdfsdf" title="parent 1-2 sdfsdfsd">
          <TreeNode value="leaf3" title={<b style={{ color: '#08c' }}>leaf3</b>} />
        </TreeNode>
        <TreeNode value="parent 1-4 sdfsdfsdf" title="parent 1-4 sdfsdfsd">
          <TreeNode value="leaf3" title={<b style={{ color: '#08c' }}>leaf3</b>} />
        </TreeNode>
        <TreeNode value="parent 1-5 2sdfsdfsdf" title="parent 1-5 sdfsdfsd">
          <TreeNode value="leaf3" title={<b style={{ color: '#08c' }}>leaf3</b>} />
        </TreeNode>
      </TreeNode>
    </TreeSelect>
  );
}

export default Demo;
`},28458:function(v,t){t.Z=`import * as React from 'react';
import type { DataEntity, IconType } from 'rc-tree/lib/interface';
import type { LegacyDataNode, SafeKey, Key } from './interface';

interface LegacyContextProps {
  checkable: boolean | React.ReactNode;
  checkedKeys: Key[];
  halfCheckedKeys: Key[];
  treeExpandedKeys: Key[];
  treeDefaultExpandedKeys: Key[];
  onTreeExpand: (keys: Key[]) => void;
  treeDefaultExpandAll: boolean;
  treeIcon: IconType;
  showTreeIcon: boolean;
  switcherIcon: IconType;
  treeLine: boolean;
  treeNodeFilterProp: string;
  treeLoadedKeys: Key[];
  treeMotion: any;
  loadData: (treeNode: LegacyDataNode) => Promise<unknown>;
  onTreeLoad: (loadedKeys: Key[]) => void;

  keyEntities: Record<SafeKey, DataEntity<any>>;
}

const LegacySelectContext = React.createContext<LegacyContextProps>(null);

export default LegacySelectContext;
`},49597:function(v,t){t.Z=`import { useBaseProps } from '@rc-component/select';
import type { RefOptionListProps } from '@rc-component/select/lib/OptionList';
import type { TreeProps } from 'rc-tree';
import Tree from 'rc-tree';
import { UnstableContext } from 'rc-tree';
import type { EventDataNode, ScrollTo } from 'rc-tree/lib/interface';
import KeyCode from '@rc-component/util/lib/KeyCode';
import useMemo from '@rc-component/util/lib/hooks/useMemo';
import * as React from 'react';
import LegacyContext from './LegacyContext';
import TreeSelectContext from './TreeSelectContext';
import type { DataNode, Key, SafeKey } from './interface';
import { getAllKeys, isCheckDisabled } from './utils/valueUtil';
import { useEvent } from '@rc-component/util';

const HIDDEN_STYLE = {
  width: 0,
  height: 0,
  display: 'flex',
  overflow: 'hidden',
  opacity: 0,
  border: 0,
  padding: 0,
  margin: 0,
};

interface TreeEventInfo {
  node: { key: Key };
  selected?: boolean;
  checked?: boolean;
}

type ReviseRefOptionListProps = Omit<RefOptionListProps, 'scrollTo'> & { scrollTo: ScrollTo };

const OptionList: React.ForwardRefRenderFunction<ReviseRefOptionListProps> = (_, ref) => {
  const { prefixCls, multiple, searchValue, toggleOpen, open, notFoundContent } = useBaseProps();

  const {
    virtual,
    listHeight,
    listItemHeight,
    listItemScrollOffset,
    treeData,
    fieldNames,
    onSelect,
    dropdownMatchSelectWidth,
    treeExpandAction,
    treeTitleRender,
    onPopupScroll,
    leftMaxCount,
    leafCountOnly,
    valueEntities,
  } = React.useContext(TreeSelectContext);

  const {
    checkable,
    checkedKeys,
    halfCheckedKeys,
    treeExpandedKeys,
    treeDefaultExpandAll,
    treeDefaultExpandedKeys,
    onTreeExpand,
    treeIcon,
    showTreeIcon,
    switcherIcon,
    treeLine,
    treeNodeFilterProp,
    loadData,
    treeLoadedKeys,
    treeMotion,
    onTreeLoad,
    keyEntities,
  } = React.useContext(LegacyContext);

  const treeRef = React.useRef<Tree>();

  const memoTreeData = useMemo(
    () => treeData,
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [open, treeData],
    (prev, next) => next[0] && prev[1] !== next[1],
  );

  // ========================== Values ==========================
  const mergedCheckedKeys = React.useMemo(() => {
    if (!checkable) {
      return null;
    }

    return {
      checked: checkedKeys,
      halfChecked: halfCheckedKeys,
    };
  }, [checkable, checkedKeys, halfCheckedKeys]);

  // ========================== Scroll ==========================
  React.useEffect(() => {
    // Single mode should scroll to current key
    if (open && !multiple && checkedKeys.length) {
      treeRef.current?.scrollTo({ key: checkedKeys[0] });
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [open]);

  // ========================== Events ==========================
  const onListMouseDown: React.MouseEventHandler<HTMLDivElement> = event => {
    event.preventDefault();
  };

  const onInternalSelect = (__: Key[], info: TreeEventInfo) => {
    const { node } = info;

    if (checkable && isCheckDisabled(node)) {
      return;
    }

    onSelect(node.key, {
      selected: !checkedKeys.includes(node.key),
    });

    if (!multiple) {
      toggleOpen(false);
    }
  };

  // =========================== Keys ===========================
  const [expandedKeys, setExpandedKeys] = React.useState<Key[]>(treeDefaultExpandedKeys);
  const [searchExpandedKeys, setSearchExpandedKeys] = React.useState<Key[]>(null);

  const mergedExpandedKeys = React.useMemo(() => {
    if (treeExpandedKeys) {
      return [...treeExpandedKeys];
    }
    return searchValue ? searchExpandedKeys : expandedKeys;
  }, [expandedKeys, searchExpandedKeys, treeExpandedKeys, searchValue]);

  const onInternalExpand = (keys: Key[]) => {
    setExpandedKeys(keys);
    setSearchExpandedKeys(keys);

    if (onTreeExpand) {
      onTreeExpand(keys);
    }
  };

  // ========================== Search ==========================
  const lowerSearchValue = String(searchValue).toLowerCase();
  const filterTreeNode = (treeNode: EventDataNode<any>) => {
    if (!lowerSearchValue) {
      return false;
    }
    return String(treeNode[treeNodeFilterProp]).toLowerCase().includes(lowerSearchValue);
  };

  React.useEffect(() => {
    if (searchValue) {
      setSearchExpandedKeys(getAllKeys(treeData, fieldNames));
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [searchValue]);

  // ========================= Disabled =========================
  // Cache disabled states in React state to ensure re-render when cache updates
  const [disabledCache, setDisabledCache] = React.useState<Map<string, boolean>>(() => new Map());

  React.useEffect(() => {
    if (leftMaxCount) {
      setDisabledCache(new Map());
    }
  }, [leftMaxCount]);

  function getDisabledWithCache(node: DataNode) {
    const value = node[fieldNames.value];
    if (!disabledCache.has(value)) {
      const entity = valueEntities.get(value);
      const isLeaf = (entity.children || []).length === 0;

      if (!isLeaf) {
        const checkableChildren = entity.children.filter(
          childTreeNode =>
            !childTreeNode.node.disabled &&
            !childTreeNode.node.disableCheckbox &&
            !checkedKeys.includes(childTreeNode.node[fieldNames.value]),
        );

        const checkableChildrenCount = checkableChildren.length;
        disabledCache.set(value, checkableChildrenCount > leftMaxCount);
      } else {
        disabledCache.set(value, false);
      }
    }
    return disabledCache.get(value);
  }

  const nodeDisabled = useEvent((node: DataNode) => {
    const nodeValue = node[fieldNames.value];

    if (checkedKeys.includes(nodeValue)) {
      return false;
    }

    if (leftMaxCount === null) {
      return false;
    }

    if (leftMaxCount <= 0) {
      return true;
    }

    // This is a low performance calculation
    if (leafCountOnly && leftMaxCount) {
      return getDisabledWithCache(node);
    }

    return false;
  });

  // ========================== Get First Selectable Node ==========================
  const getFirstMatchingNode = (nodes: EventDataNode<any>[]): EventDataNode<any> | null => {
    for (const node of nodes) {
      if (node.disabled || node.selectable === false) {
        continue;
      }

      if (searchValue) {
        if (filterTreeNode(node)) {
          return node;
        }
      } else {
        return node;
      }

      if (node[fieldNames.children]) {
        const matchInChildren = getFirstMatchingNode(node[fieldNames.children]);
        if (matchInChildren) {
          return matchInChildren;
        }
      }
    }
    return null;
  };

  // ========================== Active ==========================
  const [activeKey, setActiveKey] = React.useState<Key>(null);
  const activeEntity = keyEntities[activeKey as SafeKey];

  React.useEffect(() => {
    if (!open) {
      return;
    }
    let nextActiveKey = null;

    const getFirstNode = () => {
      const firstNode = getFirstMatchingNode(memoTreeData);
      return firstNode ? firstNode[fieldNames.value] : null;
    };

    // single mode active first checked node
    if (!multiple && checkedKeys.length && !searchValue) {
      nextActiveKey = checkedKeys[0];
    } else {
      nextActiveKey = getFirstNode();
    }

    setActiveKey(nextActiveKey);
  }, [open, searchValue]);

  // ========================= Keyboard =========================
  React.useImperativeHandle(ref, () => ({
    scrollTo: treeRef.current?.scrollTo,
    onKeyDown: event => {
      const { which } = event;
      switch (which) {
        // >>> Arrow keys
        case KeyCode.UP:
        case KeyCode.DOWN:
        case KeyCode.LEFT:
        case KeyCode.RIGHT:
          treeRef.current?.onKeyDown(event as React.KeyboardEvent<HTMLDivElement>);
          break;

        // >>> Select item
        case KeyCode.ENTER: {
          if (activeEntity) {
            const isNodeDisabled = nodeDisabled(activeEntity.node);
            const { selectable, value, disabled } = activeEntity?.node || {};
            if (selectable !== false && !disabled && !isNodeDisabled) {
              onInternalSelect(null, {
                node: { key: activeKey },
                selected: !checkedKeys.includes(value),
              });
            }
          }
          break;
        }

        // >>> Close
        case KeyCode.ESC: {
          toggleOpen(false);
        }
      }
    },
    onKeyUp: () => {},
  }));

  const hasLoadDataFn = useMemo(
    () => (searchValue ? false : true),
    [searchValue, treeExpandedKeys || expandedKeys],
    ([preSearchValue], [nextSearchValue, nextExcludeSearchExpandedKeys]) =>
      preSearchValue !== nextSearchValue && !!(nextSearchValue || nextExcludeSearchExpandedKeys),
  );

  const syncLoadData = hasLoadDataFn ? loadData : null;

  // ========================== Render ==========================
  if (memoTreeData.length === 0) {
    return (
      <div role="listbox" className={\`\${prefixCls}-empty\`} onMouseDown={onListMouseDown}>
        {notFoundContent}
      </div>
    );
  }

  const treeProps: Partial<TreeProps> = {
    fieldNames,
  };
  if (treeLoadedKeys) {
    treeProps.loadedKeys = treeLoadedKeys;
  }
  if (mergedExpandedKeys) {
    treeProps.expandedKeys = mergedExpandedKeys;
  }

  return (
    <div onMouseDown={onListMouseDown}>
      {activeEntity && open && (
        <span style={HIDDEN_STYLE} aria-live="assertive">
          {activeEntity.node.value}
        </span>
      )}
      <UnstableContext.Provider value={{ nodeDisabled }}>
        <Tree
          ref={treeRef}
          focusable={false}
          prefixCls={\`\${prefixCls}-tree\`}
          treeData={memoTreeData}
          height={listHeight}
          itemHeight={listItemHeight}
          itemScrollOffset={listItemScrollOffset}
          virtual={virtual !== false && dropdownMatchSelectWidth !== false}
          multiple={multiple}
          icon={treeIcon}
          showIcon={showTreeIcon}
          switcherIcon={switcherIcon}
          showLine={treeLine}
          loadData={syncLoadData}
          motion={treeMotion}
          activeKey={activeKey}
          // We handle keys by out instead tree self
          checkable={checkable}
          checkStrictly
          checkedKeys={mergedCheckedKeys}
          selectedKeys={!checkable ? checkedKeys : []}
          defaultExpandAll={treeDefaultExpandAll}
          titleRender={treeTitleRender}
          {...treeProps}
          // Proxy event out
          onActiveChange={setActiveKey}
          onSelect={onInternalSelect}
          onCheck={onInternalSelect}
          onExpand={onInternalExpand}
          onLoad={onTreeLoad}
          filterTreeNode={filterTreeNode}
          expandAction={treeExpandAction}
          onScroll={onPopupScroll}
        />
      </UnstableContext.Provider>
    </div>
  );
};

const RefOptionList = React.forwardRef<ReviseRefOptionListProps>(OptionList);

if (process.env.NODE_ENV !== 'production') {
  RefOptionList.displayName = 'OptionList';
}

export default RefOptionList;
`},41849:function(v,t){t.Z=`/* istanbul ignore file */
import type * as React from 'react';
import type { DataNode, Key } from './interface';

export interface TreeNodeProps extends Omit<DataNode, 'children'> {
  value: Key;
  children?: React.ReactNode;
}

/** This is a placeholder, not real render in dom */
const TreeNode: React.FC<TreeNodeProps> = () => null;

export default TreeNode;
`},51184:function(v,t){t.Z=`import type { BaseSelectPropsWithoutPrivate, BaseSelectRef } from '@rc-component/select';
import { BaseSelect } from '@rc-component/select';
import useId from '@rc-component/select/lib/hooks/useId';
import type { IconType } from 'rc-tree/lib/interface';
import type { ExpandAction } from 'rc-tree/lib/Tree';
import { conductCheck } from 'rc-tree/lib/utils/conductUtil';
import useMergedState from '@rc-component/util/lib/hooks/useMergedState';
import warning from '@rc-component/util/lib/warning';
import * as React from 'react';
import useCache from './hooks/useCache';
import useCheckedKeys from './hooks/useCheckedKeys';
import useDataEntities from './hooks/useDataEntities';
import useFilterTreeData from './hooks/useFilterTreeData';
import useRefFunc from './hooks/useRefFunc';
import useTreeData from './hooks/useTreeData';
import LegacyContext from './LegacyContext';
import OptionList from './OptionList';
import TreeNode from './TreeNode';
import type { TreeSelectContextProps } from './TreeSelectContext';
import TreeSelectContext from './TreeSelectContext';
import { fillAdditionalInfo, fillLegacyProps } from './utils/legacyUtil';
import type { CheckedStrategy } from './utils/strategyUtil';
import { formatStrategyValues, SHOW_ALL, SHOW_CHILD, SHOW_PARENT } from './utils/strategyUtil';
import { fillFieldNames, isNil, toArray } from './utils/valueUtil';
import warningProps from './utils/warningPropsUtil';
import type {
  LabeledValueType,
  SafeKey,
  Key,
  DataNode,
  SimpleModeConfig,
  ChangeEventExtra,
  SelectSource,
  DefaultValueType,
  FieldNames,
  LegacyDataNode,
} from './interface';

export interface TreeSelectProps<ValueType = any, OptionType extends DataNode = DataNode>
  extends Omit<BaseSelectPropsWithoutPrivate, 'mode'> {
  prefixCls?: string;
  id?: string;
  children?: React.ReactNode;

  // >>> Value
  value?: ValueType;
  defaultValue?: ValueType;
  onChange?: (value: ValueType, labelList: React.ReactNode[], extra: ChangeEventExtra) => void;

  // >>> Search
  searchValue?: string;
  /** @deprecated Use \`searchValue\` instead */
  inputValue?: string;
  onSearch?: (value: string) => void;
  autoClearSearchValue?: boolean;
  filterTreeNode?: boolean | ((inputValue: string, treeNode: DataNode) => boolean);
  treeNodeFilterProp?: string;

  // >>> Select
  onSelect?: (value: ValueType, option: OptionType) => void;
  onDeselect?: (value: ValueType, option: OptionType) => void;

  // >>> Selector
  showCheckedStrategy?: CheckedStrategy;
  treeNodeLabelProp?: string;

  // >>> Field Names
  fieldNames?: FieldNames;

  // >>> Mode
  multiple?: boolean;
  treeCheckable?: boolean | React.ReactNode;
  treeCheckStrictly?: boolean;
  labelInValue?: boolean;
  maxCount?: number;

  // >>> Data
  treeData?: OptionType[];
  treeDataSimpleMode?: boolean | SimpleModeConfig;
  loadData?: (dataNode: LegacyDataNode) => Promise<unknown>;
  treeLoadedKeys?: SafeKey[];
  onTreeLoad?: (loadedKeys: SafeKey[]) => void;

  // >>> Expanded
  treeDefaultExpandAll?: boolean;
  treeExpandedKeys?: SafeKey[];
  treeDefaultExpandedKeys?: SafeKey[];
  onTreeExpand?: (expandedKeys: SafeKey[]) => void;
  treeExpandAction?: ExpandAction;

  // >>> Options
  virtual?: boolean;
  listHeight?: number;
  listItemHeight?: number;
  listItemScrollOffset?: number;
  onDropdownVisibleChange?: (open: boolean) => void;
  treeTitleRender?: (node: OptionType) => React.ReactNode;

  // >>> Tree
  treeLine?: boolean;
  treeIcon?: IconType;
  showTreeIcon?: boolean;
  switcherIcon?: IconType;
  treeMotion?: any;
}

function isRawValue(value: SafeKey | LabeledValueType): value is SafeKey {
  return !value || typeof value !== 'object';
}

const TreeSelect = React.forwardRef<BaseSelectRef, TreeSelectProps>((props, ref) => {
  const {
    id,
    prefixCls = 'rc-tree-select',

    // Value
    value,
    defaultValue,
    onChange,
    onSelect,
    onDeselect,

    // Search
    searchValue,
    inputValue,
    onSearch,
    autoClearSearchValue = true,
    filterTreeNode,
    treeNodeFilterProp = 'value',

    // Selector
    showCheckedStrategy,
    treeNodeLabelProp,

    //  Mode
    multiple,
    treeCheckable,
    treeCheckStrictly,
    labelInValue,
    maxCount,

    // FieldNames
    fieldNames,

    // Data
    treeDataSimpleMode,
    treeData,
    children,
    loadData,
    treeLoadedKeys,
    onTreeLoad,

    // Expanded
    treeDefaultExpandAll,
    treeExpandedKeys,
    treeDefaultExpandedKeys,
    onTreeExpand,
    treeExpandAction,

    // Options
    virtual,
    listHeight = 200,
    listItemHeight = 20,
    listItemScrollOffset = 0,

    onDropdownVisibleChange,
    dropdownMatchSelectWidth = true,

    // Tree
    treeLine,
    treeIcon,
    showTreeIcon,
    switcherIcon,
    treeMotion,
    treeTitleRender,

    onPopupScroll,
    ...restProps
  } = props;

  const mergedId = useId(id);
  const treeConduction = treeCheckable && !treeCheckStrictly;
  const mergedCheckable = treeCheckable || treeCheckStrictly;
  const mergedLabelInValue = treeCheckStrictly || labelInValue;
  const mergedMultiple = mergedCheckable || multiple;

  const [internalValue, setInternalValue] = useMergedState(defaultValue, { value });

  // \`multiple\` && \`!treeCheckable\` should be show all
  const mergedShowCheckedStrategy = React.useMemo(() => {
    if (!treeCheckable) {
      return SHOW_ALL;
    }

    return showCheckedStrategy || SHOW_CHILD;
  }, [showCheckedStrategy, treeCheckable]);

  // ========================== Warning ===========================
  if (process.env.NODE_ENV !== 'production') {
    warningProps(props);
  }

  // ========================= FieldNames =========================
  const mergedFieldNames: FieldNames = React.useMemo(
    () => fillFieldNames(fieldNames),
    /* eslint-disable react-hooks/exhaustive-deps */
    [JSON.stringify(fieldNames)],
    /* eslint-enable react-hooks/exhaustive-deps */
  );

  // =========================== Search ===========================
  const [mergedSearchValue, setSearchValue] = useMergedState('', {
    value: searchValue !== undefined ? searchValue : inputValue,
    postState: search => search || '',
  });

  const onInternalSearch = searchText => {
    setSearchValue(searchText);
    onSearch?.(searchText);
  };

  // ============================ Data ============================
  // \`useTreeData\` only do convert of \`children\` or \`simpleMode\`.
  // Else will return origin \`treeData\` for perf consideration.
  // Do not do anything to loop the data.
  const mergedTreeData = useTreeData(treeData, children, treeDataSimpleMode);

  const { keyEntities, valueEntities } = useDataEntities(mergedTreeData, mergedFieldNames);

  /** Get \`missingRawValues\` which not exist in the tree yet */
  const splitRawValues = React.useCallback(
    (newRawValues: SafeKey[]) => {
      const missingRawValues = [];
      const existRawValues = [];

      // Keep missing value in the cache
      newRawValues.forEach(val => {
        if (valueEntities.has(val)) {
          existRawValues.push(val);
        } else {
          missingRawValues.push(val);
        }
      });

      return { missingRawValues, existRawValues };
    },
    [valueEntities],
  );

  // Filtered Tree
  const filteredTreeData = useFilterTreeData(mergedTreeData, mergedSearchValue, {
    fieldNames: mergedFieldNames,
    treeNodeFilterProp,
    filterTreeNode,
  });

  // =========================== Label ============================
  const getLabel = React.useCallback(
    (item: DataNode) => {
      if (item) {
        if (treeNodeLabelProp) {
          return item[treeNodeLabelProp];
        }

        // Loop from fieldNames
        const { _title: titleList } = mergedFieldNames;

        for (let i = 0; i < titleList.length; i += 1) {
          const title = item[titleList[i]];
          if (title !== undefined) {
            return title;
          }
        }
      }
    },
    [mergedFieldNames, treeNodeLabelProp],
  );

  // ========================= Wrap Value =========================
  const toLabeledValues = React.useCallback((draftValues: DefaultValueType) => {
    const values = toArray(draftValues);

    return values.map(val => {
      if (isRawValue(val)) {
        return { value: val };
      }
      return val;
    });
  }, []);

  const convert2LabelValues = React.useCallback(
    (draftValues: DefaultValueType) => {
      const values = toLabeledValues(draftValues);

      return values.map(item => {
        let { label: rawLabel } = item;
        const { value: rawValue, halfChecked: rawHalfChecked } = item;

        let rawDisabled: boolean | undefined;

        const entity = valueEntities.get(rawValue);

        // Fill missing label & status
        if (entity) {
          rawLabel = treeTitleRender
            ? treeTitleRender(entity.node)
            : (rawLabel ?? getLabel(entity.node));
          rawDisabled = entity.node.disabled;
        } else if (rawLabel === undefined) {
          // We try to find in current \`labelInValue\` value
          const labelInValueItem = toLabeledValues(internalValue).find(
            labeledItem => labeledItem.value === rawValue,
          );
          rawLabel = labelInValueItem.label;
        }
        return {
          label: rawLabel,
          value: rawValue,
          halfChecked: rawHalfChecked,
          disabled: rawDisabled,
        };
      });
    },
    [valueEntities, getLabel, toLabeledValues, internalValue],
  );

  // =========================== Values ===========================
  const rawMixedLabeledValues = React.useMemo(
    () => toLabeledValues(internalValue === null ? [] : internalValue),
    [toLabeledValues, internalValue],
  );

  // Split value into full check and half check
  const [rawLabeledValues, rawHalfLabeledValues] = React.useMemo(() => {
    const fullCheckValues: LabeledValueType[] = [];
    const halfCheckValues: LabeledValueType[] = [];

    rawMixedLabeledValues.forEach(item => {
      if (item.halfChecked) {
        halfCheckValues.push(item);
      } else {
        fullCheckValues.push(item);
      }
    });

    return [fullCheckValues, halfCheckValues];
  }, [rawMixedLabeledValues]);

  // const [mergedValues] = useCache(rawLabeledValues);
  const rawValues = React.useMemo(
    () => rawLabeledValues.map(item => item.value),
    [rawLabeledValues],
  );

  // Convert value to key. Will fill missed keys for conduct check.
  const [rawCheckedValues, rawHalfCheckedValues] = useCheckedKeys(
    rawLabeledValues,
    rawHalfLabeledValues,
    treeConduction,
    keyEntities,
  );

  // Convert rawCheckedKeys to check strategy related values
  const displayValues = React.useMemo(() => {
    // Collect keys which need to show
    const displayKeys = formatStrategyValues(
      rawCheckedValues as SafeKey[],
      mergedShowCheckedStrategy,
      keyEntities,
      mergedFieldNames,
    );

    // Convert to value and filled with label
    const values = displayKeys.map(key => keyEntities[key]?.node?.[mergedFieldNames.value] ?? key);

    // Back fill with origin label
    const labeledValues = values.map(val => {
      const targetItem = rawLabeledValues.find(item => item.value === val);
      const label = labelInValue ? targetItem?.label : treeTitleRender?.(targetItem);
      return {
        value: val,
        label,
      };
    });

    const rawDisplayValues = convert2LabelValues(labeledValues);

    const firstVal = rawDisplayValues[0];

    if (!mergedMultiple && firstVal && isNil(firstVal.value) && isNil(firstVal.label)) {
      return [];
    }

    return rawDisplayValues.map(item => ({
      ...item,
      label: item.label ?? item.value,
    }));
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [
    mergedFieldNames,
    mergedMultiple,
    rawCheckedValues,
    rawLabeledValues,
    convert2LabelValues,
    mergedShowCheckedStrategy,
    keyEntities,
  ]);

  const [cachedDisplayValues] = useCache(displayValues);

  // ========================== MaxCount ==========================
  const mergedMaxCount = React.useMemo(() => {
    if (
      mergedMultiple &&
      (mergedShowCheckedStrategy === 'SHOW_CHILD' || treeCheckStrictly || !treeCheckable)
    ) {
      return maxCount;
    }
    return null;
  }, [maxCount, mergedMultiple, treeCheckStrictly, mergedShowCheckedStrategy, treeCheckable]);

  // =========================== Change ===========================
  const triggerChange = useRefFunc(
    (
      newRawValues: SafeKey[],
      extra: { triggerValue?: SafeKey; selected?: boolean },
      source: SelectSource,
    ) => {
      const formattedKeyList = formatStrategyValues(
        newRawValues,
        mergedShowCheckedStrategy,
        keyEntities,
        mergedFieldNames,
      );

      // Not allow pass with \`maxCount\`
      if (mergedMaxCount && formattedKeyList.length > mergedMaxCount) {
        return;
      }

      const labeledValues = convert2LabelValues(newRawValues);
      setInternalValue(labeledValues);

      // Clean up if needed
      if (autoClearSearchValue) {
        setSearchValue('');
      }

      // Generate rest parameters is costly, so only do it when necessary
      if (onChange) {
        let eventValues: SafeKey[] = newRawValues;
        if (treeConduction) {
          eventValues = formattedKeyList.map(key => {
            const entity = valueEntities.get(key);
            return entity ? entity.node[mergedFieldNames.value] : key;
          });
        }

        const { triggerValue, selected } = extra || {
          triggerValue: undefined,
          selected: undefined,
        };

        let returnRawValues: (LabeledValueType | SafeKey)[] = eventValues;

        // We need fill half check back
        if (treeCheckStrictly) {
          const halfValues = rawHalfLabeledValues.filter(item => !eventValues.includes(item.value));

          returnRawValues = [...returnRawValues, ...halfValues];
        }

        const returnLabeledValues = convert2LabelValues(returnRawValues);
        const additionalInfo = {
          // [Legacy] Always return as array contains label & value
          preValue: rawLabeledValues,
          triggerValue,
        } as ChangeEventExtra;

        // [Legacy] Fill legacy data if user query.
        // This is expansive that we only fill when user query
        // https://github.com/react-component/tree-select/blob/fe33eb7c27830c9ac70cd1fdb1ebbe7bc679c16a/src/Select.jsx
        let showPosition = true;
        if (treeCheckStrictly || (source === 'selection' && !selected)) {
          showPosition = false;
        }

        fillAdditionalInfo(
          additionalInfo,
          triggerValue,
          newRawValues,
          mergedTreeData,
          showPosition,
          mergedFieldNames,
        );

        if (mergedCheckable) {
          additionalInfo.checked = selected;
        } else {
          additionalInfo.selected = selected;
        }

        const returnValues = mergedLabelInValue
          ? returnLabeledValues
          : returnLabeledValues.map(item => item.value);

        onChange(
          mergedMultiple ? returnValues : returnValues[0],
          mergedLabelInValue ? null : returnLabeledValues.map(item => item.label),
          additionalInfo,
        );
      }
    },
  );

  // ========================== Options ===========================
  /** Trigger by option list */
  const onOptionSelect = React.useCallback(
    (selectedKey: SafeKey, { selected, source }: { selected: boolean; source: SelectSource }) => {
      const entity = keyEntities[selectedKey];
      const node = entity?.node;
      const selectedValue = node?.[mergedFieldNames.value] ?? selectedKey;

      // Never be falsy but keep it safe
      if (!mergedMultiple) {
        // Single mode always set value
        triggerChange([selectedValue], { selected: true, triggerValue: selectedValue }, 'option');
      } else {
        let newRawValues = selected
          ? [...rawValues, selectedValue]
          : rawCheckedValues.filter(v => v !== selectedValue);

        // Add keys if tree conduction
        if (treeConduction) {
          // Should keep missing values
          const { missingRawValues, existRawValues } = splitRawValues(newRawValues);
          const keyList = existRawValues.map(val => valueEntities.get(val).key);

          // Conduction by selected or not
          let checkedKeys: Key[];
          if (selected) {
            ({ checkedKeys } = conductCheck(keyList, true, keyEntities));
          } else {
            ({ checkedKeys } = conductCheck(
              keyList,
              { checked: false, halfCheckedKeys: rawHalfCheckedValues },
              keyEntities,
            ));
          }

          // Fill back of keys
          newRawValues = [
            ...missingRawValues,
            ...checkedKeys.map(key => keyEntities[key as SafeKey].node[mergedFieldNames.value]),
          ];
        }
        triggerChange(newRawValues, { selected, triggerValue: selectedValue }, source || 'option');
      }

      // Trigger select event
      if (selected || !mergedMultiple) {
        onSelect?.(selectedValue, fillLegacyProps(node));
      } else {
        onDeselect?.(selectedValue, fillLegacyProps(node));
      }
    },
    [
      splitRawValues,
      valueEntities,
      keyEntities,
      mergedFieldNames,
      mergedMultiple,
      rawValues,
      triggerChange,
      treeConduction,
      onSelect,
      onDeselect,
      rawCheckedValues,
      rawHalfCheckedValues,
      maxCount,
    ],
  );

  // ========================== Dropdown ==========================
  const onInternalDropdownVisibleChange = React.useCallback(
    (open: boolean) => {
      if (onDropdownVisibleChange) {
        const legacyParam = {};

        Object.defineProperty(legacyParam, 'documentClickClose', {
          get() {
            warning(false, 'Second param of \`onDropdownVisibleChange\` has been removed.');
            return false;
          },
        });

        (onDropdownVisibleChange as any)(open, legacyParam);
      }
    },
    [onDropdownVisibleChange],
  );

  // ====================== Display Change ========================
  const onDisplayValuesChange = useRefFunc((newValues, info) => {
    const newRawValues = newValues.map(item => item.value);

    if (info.type === 'clear') {
      triggerChange(newRawValues, {}, 'selection');
      return;
    }

    // TreeSelect only have multiple mode which means display change only has remove
    if (info.values.length) {
      onOptionSelect(info.values[0].value, { selected: false, source: 'selection' });
    }
  });

  // ========================== Context ===========================
  const treeSelectContext = React.useMemo<TreeSelectContextProps>(() => {
    return {
      virtual,
      dropdownMatchSelectWidth,
      listHeight,
      listItemHeight,
      listItemScrollOffset,
      treeData: filteredTreeData,
      fieldNames: mergedFieldNames,
      onSelect: onOptionSelect,
      treeExpandAction,
      treeTitleRender,
      onPopupScroll,
      leftMaxCount: maxCount === undefined ? null : maxCount - cachedDisplayValues.length,
      leafCountOnly:
        mergedShowCheckedStrategy === 'SHOW_CHILD' && !treeCheckStrictly && !!treeCheckable,
      valueEntities,
    };
  }, [
    virtual,
    dropdownMatchSelectWidth,
    listHeight,
    listItemHeight,
    listItemScrollOffset,
    filteredTreeData,
    mergedFieldNames,
    onOptionSelect,
    treeExpandAction,
    treeTitleRender,
    onPopupScroll,
    maxCount,
    cachedDisplayValues.length,
    mergedShowCheckedStrategy,
    treeCheckStrictly,
    treeCheckable,
    valueEntities,
  ]);

  // ======================= Legacy Context =======================
  const legacyContext = React.useMemo(
    () => ({
      checkable: mergedCheckable,
      loadData,
      treeLoadedKeys,
      onTreeLoad,
      checkedKeys: rawCheckedValues,
      halfCheckedKeys: rawHalfCheckedValues,
      treeDefaultExpandAll,
      treeExpandedKeys,
      treeDefaultExpandedKeys,
      onTreeExpand,
      treeIcon,
      treeMotion,
      showTreeIcon,
      switcherIcon,
      treeLine,
      treeNodeFilterProp,
      keyEntities,
    }),
    [
      mergedCheckable,
      loadData,
      treeLoadedKeys,
      onTreeLoad,
      rawCheckedValues,
      rawHalfCheckedValues,
      treeDefaultExpandAll,
      treeExpandedKeys,
      treeDefaultExpandedKeys,
      onTreeExpand,
      treeIcon,
      treeMotion,
      showTreeIcon,
      switcherIcon,
      treeLine,
      treeNodeFilterProp,
      keyEntities,
    ],
  );

  // =========================== Render ===========================
  return (
    <TreeSelectContext.Provider value={treeSelectContext}>
      <LegacyContext.Provider value={legacyContext}>
        <BaseSelect
          ref={ref}
          {...restProps}
          // >>> MISC
          id={mergedId}
          prefixCls={prefixCls}
          mode={mergedMultiple ? 'multiple' : undefined}
          // >>> Display Value
          displayValues={cachedDisplayValues}
          onDisplayValuesChange={onDisplayValuesChange}
          // >>> Search
          searchValue={mergedSearchValue}
          onSearch={onInternalSearch}
          // >>> Options
          OptionList={OptionList}
          emptyOptions={!mergedTreeData.length}
          onDropdownVisibleChange={onInternalDropdownVisibleChange}
          dropdownMatchSelectWidth={dropdownMatchSelectWidth}
        />
      </LegacyContext.Provider>
    </TreeSelectContext.Provider>
  );
});

// Assign name for Debug
if (process.env.NODE_ENV !== 'production') {
  TreeSelect.displayName = 'TreeSelect';
}

const GenericTreeSelect = TreeSelect as unknown as (<
  ValueType = any,
  OptionType extends DataNode = DataNode,
>(
  props: React.PropsWithChildren<TreeSelectProps<ValueType, OptionType>> & {
    ref?: React.Ref<BaseSelectRef>;
  },
) => React.ReactElement) & {
  TreeNode: typeof TreeNode;
  SHOW_ALL: typeof SHOW_ALL;
  SHOW_PARENT: typeof SHOW_PARENT;
  SHOW_CHILD: typeof SHOW_CHILD;
};

GenericTreeSelect.TreeNode = TreeNode;
GenericTreeSelect.SHOW_ALL = SHOW_ALL;
GenericTreeSelect.SHOW_PARENT = SHOW_PARENT;
GenericTreeSelect.SHOW_CHILD = SHOW_CHILD;

export default GenericTreeSelect;
`},29285:function(v,t){t.Z=`import * as React from 'react';
import type { ExpandAction } from 'rc-tree/lib/Tree';
import type { DataNode, FieldNames, Key } from './interface';
import type useDataEntities from './hooks/useDataEntities';

export interface TreeSelectContextProps {
  virtual?: boolean;
  dropdownMatchSelectWidth?: boolean | number;
  listHeight: number;
  listItemHeight: number;
  listItemScrollOffset?: number;
  treeData: DataNode[];
  fieldNames: FieldNames;
  onSelect: (value: Key, info: { selected: boolean }) => void;
  treeExpandAction?: ExpandAction;
  treeTitleRender?: (node: any) => React.ReactNode;
  onPopupScroll?: React.UIEventHandler<HTMLDivElement>;

  // For \`maxCount\` usage
  leftMaxCount: number | null;
  /** When \`true\`, only take leaf node as count, or take all as count with \`maxCount\` limitation */
  leafCountOnly: boolean;
  valueEntities: ReturnType<typeof useDataEntities>['valueEntities'];
}

const TreeSelectContext = React.createContext<TreeSelectContextProps>(null as any);

export default TreeSelectContext;
`},95108:function(v,t){t.Z=`import * as React from 'react';
import type { LabeledValueType, SafeKey } from '../interface';

/**
 * This function will try to call requestIdleCallback if available to save performance.
 * No need \`getLabel\` here since already fetch on \`rawLabeledValue\`.
 */
export default (values: LabeledValueType[]): [LabeledValueType[]] => {
  const cacheRef = React.useRef({
    valueLabels: new Map<SafeKey, React.ReactNode>(),
  });

  return React.useMemo(() => {
    const { valueLabels } = cacheRef.current;
    const valueLabelsCache = new Map<SafeKey, React.ReactNode>();

    const filledValues = values.map(item => {
      const { value, label } = item;
      const mergedLabel = label ?? valueLabels.get(value);

      // Save in cache
      valueLabelsCache.set(value, mergedLabel);

      return {
        ...item,
        label: mergedLabel,
      };
    });

    cacheRef.current.valueLabels = valueLabelsCache;

    return [filledValues];
  }, [values]);
};
`},7118:function(v,t){t.Z=`import * as React from 'react';
import type { DataEntity } from 'rc-tree/lib/interface';
import { conductCheck } from 'rc-tree/lib/utils/conductUtil';
import type { LabeledValueType, SafeKey, Key } from '../interface';

const useCheckedKeys = (
  rawLabeledValues: LabeledValueType[],
  rawHalfCheckedValues: LabeledValueType[],
  treeConduction: boolean,
  keyEntities: Record<SafeKey, DataEntity>,
) => {
  return React.useMemo(() => {
    const extractValues = (values: LabeledValueType[]): Key[] => values.map(({ value }) => value);

    const checkedKeys = extractValues(rawLabeledValues);
    const halfCheckedKeys = extractValues(rawHalfCheckedValues);

    const missingValues = checkedKeys.filter(key => !keyEntities[key as SafeKey]);

    let finalCheckedKeys = checkedKeys;
    let finalHalfCheckedKeys = halfCheckedKeys;

    if (treeConduction) {
      const conductResult = conductCheck(checkedKeys, true, keyEntities);
      finalCheckedKeys = conductResult.checkedKeys;
      finalHalfCheckedKeys = conductResult.halfCheckedKeys;
    }

    return [Array.from(new Set([...missingValues, ...finalCheckedKeys])), finalHalfCheckedKeys];
  }, [rawLabeledValues, rawHalfCheckedValues, treeConduction, keyEntities]);
};

export default useCheckedKeys;
`},31367:function(v,t){t.Z=`import * as React from 'react';
import { convertDataToEntities } from 'rc-tree/lib/utils/treeUtil';
import type { DataEntity } from 'rc-tree/lib/interface';
import type { SafeKey, FieldNames } from '../interface';
import warning from '@rc-component/util/lib/warning';
import { isNil } from '../utils/valueUtil';

export default (treeData: any, fieldNames: FieldNames) =>
  React.useMemo<{
    valueEntities: Map<SafeKey, DataEntity>;
    keyEntities: Record<string, DataEntity>;
  }>(() => {
    const collection = convertDataToEntities(treeData, {
      fieldNames,
      initWrapper: wrapper => ({
        ...wrapper,
        valueEntities: new Map(),
      }),
      processEntity: (entity, wrapper: any) => {
        const val = entity.node[fieldNames.value];

        // Check if exist same value
        if (process.env.NODE_ENV !== 'production') {
          const key = entity.node.key;

          warning(!isNil(val), 'TreeNode \`value\` is invalidate: undefined');
          warning(!wrapper.valueEntities.has(val), \`Same \\\`value\\\` exist in the tree: \${val}\`);
          warning(
            !key || String(key) === String(val),
            \`\\\`key\\\` or \\\`value\\\` with TreeNode must be the same or you can remove one of them. key: \${key}, value: \${val}.\`,
          );
        }
        wrapper.valueEntities.set(val, entity);
      },
    });

    return collection as any;
  }, [treeData, fieldNames]);
`},67685:function(v,t){t.Z=`import * as React from 'react';
import type { TreeSelectProps } from '../TreeSelect';
import type { DataNode, FieldNames } from '../interface';
import { fillLegacyProps } from '../utils/legacyUtil';

type FilterFn = NonNullable<TreeSelectProps['filterTreeNode']>;

const useFilterTreeData = (
  treeData: DataNode[],
  searchValue: string,
  options: {
    fieldNames: FieldNames;
    treeNodeFilterProp: string;
    filterTreeNode: TreeSelectProps['filterTreeNode'];
  },
) => {
  const { fieldNames, treeNodeFilterProp, filterTreeNode } = options;
  const { children: fieldChildren } = fieldNames;

  return React.useMemo(() => {
    if (!searchValue || filterTreeNode === false) {
      return treeData;
    }

    const filterOptionFunc: FilterFn =
      typeof filterTreeNode === 'function'
        ? filterTreeNode
        : (_, dataNode) =>
            String(dataNode[treeNodeFilterProp]).toUpperCase().includes(searchValue.toUpperCase());

    const filterTreeNodes = (nodes: DataNode[], keepAll = false): DataNode[] =>
      nodes.reduce<DataNode[]>((filtered, node) => {
        const children = node[fieldChildren];
        const isMatch = keepAll || filterOptionFunc(searchValue, fillLegacyProps(node));
        const filteredChildren = filterTreeNodes(children || [], isMatch);

        if (isMatch || filteredChildren.length) {
          filtered.push({
            ...node,
            isLeaf: undefined,
            [fieldChildren]: filteredChildren,
          });
        }
        return filtered;
      }, []);

    return filterTreeNodes(treeData);
  }, [treeData, searchValue, fieldChildren, treeNodeFilterProp, filterTreeNode]);
};

export default useFilterTreeData;
`},60133:function(v,t){t.Z=`import * as React from 'react';

/**
 * Same as \`React.useCallback\` but always return a memoized function
 * but redirect to real function.
 */
export default function useRefFunc<T extends (...args: any[]) => any>(callback: T): T {
  const funcRef = React.useRef<T>();
  funcRef.current = callback;

  const cacheFn = React.useCallback((...args: any[]) => {
    return funcRef.current(...args);
  }, []);

  return cacheFn as any;
}
`},4997:function(v,t){t.Z=`import * as React from 'react';
import type { DataNode, SimpleModeConfig } from '../interface';
import { convertChildrenToData } from '../utils/legacyUtil';

function buildTreeStructure(nodes: DataNode[], config: SimpleModeConfig): DataNode[] {
  const { id, pId, rootPId } = config;
  const nodeMap = new Map<string, DataNode>();
  const rootNodes: DataNode[] = [];

  nodes.forEach(node => {
    const nodeKey = node[id];
    const clonedNode = { ...node, key: node.key || nodeKey };
    nodeMap.set(nodeKey, clonedNode);
  });

  nodeMap.forEach(node => {
    const parentKey = node[pId];
    const parent = nodeMap.get(parentKey);

    if (parent) {
      parent.children = parent.children || [];
      parent.children.push(node);
    } else if (parentKey === rootPId || rootPId === null) {
      rootNodes.push(node);
    }
  });

  return rootNodes;
}

/**
 * \u5C06 \`treeData\` \u6216 \`children\` \u8F6C\u6362\u4E3A\u683C\u5F0F\u5316\u7684 \`treeData\`\u3002
 * \u5982\u679C \`treeData\` \u6216 \`children\` \u6CA1\u6709\u53D8\u5316\uFF0C\u5219\u4E0D\u4F1A\u91CD\u65B0\u8BA1\u7B97\u3002
 */
export default function useTreeData(
  treeData: DataNode[],
  children: React.ReactNode,
  simpleMode: boolean | SimpleModeConfig,
): DataNode[] {
  return React.useMemo(() => {
    if (treeData) {
      if (simpleMode) {
        const config: SimpleModeConfig = {
          id: 'id',
          pId: 'pId',
          rootPId: null,
          ...(typeof simpleMode === 'object' ? simpleMode : {}),
        };
        return buildTreeStructure(treeData, config);
      }
      return treeData;
    }

    return convertChildrenToData(children);
  }, [children, simpleMode, treeData]);
}
`},42103:function(v,t){t.Z=`import TreeSelect from './TreeSelect';
import type { TreeSelectProps } from './TreeSelect';
import TreeNode from './TreeNode';
import { SHOW_ALL, SHOW_CHILD, SHOW_PARENT } from './utils/strategyUtil';

export { TreeNode, SHOW_ALL, SHOW_CHILD, SHOW_PARENT };
export type { TreeSelectProps };
export default TreeSelect;
`},29035:function(v,t){t.Z=`import * as React from 'react';
import toArray from '@rc-component/util/lib/Children/toArray';
import warning from '@rc-component/util/lib/warning';
import type {
  DataNode,
  ChangeEventExtra,
  SafeKey,
  LegacyCheckedNode,
  FieldNames,
} from '../interface';
import TreeNode from '../TreeNode';

export function convertChildrenToData(nodes: React.ReactNode): DataNode[] {
  return toArray(nodes)
    .map((node: React.ReactElement) => {
      if (!React.isValidElement(node) || !node.type) {
        return null;
      }

      const {
        key,
        props: { children, value, ...restProps },
      } = node as React.ReactElement;

      const data = {
        key,
        value,
        ...restProps,
      };

      const childData = convertChildrenToData(children);
      if (childData.length) {
        data.children = childData;
      }

      return data;
    })
    .filter(data => data);
}

export function fillLegacyProps(dataNode: DataNode) {
  if (!dataNode) {
    return dataNode;
  }

  const cloneNode = { ...dataNode };

  if (!('props' in cloneNode)) {
    Object.defineProperty(cloneNode, 'props', {
      get() {
        warning(
          false,
          'New \`rc-tree-select\` not support return node instance as argument anymore. Please consider to remove \`props\` access.',
        );
        return cloneNode;
      },
    });
  }

  return cloneNode;
}

export function fillAdditionalInfo(
  extra: ChangeEventExtra,
  triggerValue: SafeKey,
  checkedValues: SafeKey[],
  treeData: DataNode[],
  showPosition: boolean,
  fieldNames: FieldNames,
) {
  let triggerNode: React.ReactNode = null;
  let nodeList: LegacyCheckedNode[] = null;

  function generateMap() {
    function dig(list: DataNode[], level = '0', parentIncluded = false) {
      return list
        .map((option, index) => {
          const pos = \`\${level}-\${index}\`;
          const value = option[fieldNames.value];
          const included = checkedValues.includes(value);
          const children = dig(option[fieldNames.children] || [], pos, included);
          const node = (
            <TreeNode {...(option as Required<DataNode>)}>
              {children.map(child => child.node)}
            </TreeNode>
          );

          // Link with trigger node
          if (triggerValue === value) {
            triggerNode = node;
          }

          if (included) {
            const checkedNode: LegacyCheckedNode = {
              pos,
              node,
              children,
            };

            if (!parentIncluded) {
              nodeList.push(checkedNode);
            }

            return checkedNode;
          }
          return null;
        })
        .filter(node => node);
    }

    if (!nodeList) {
      nodeList = [];

      dig(treeData);

      // Sort to keep the checked node length
      nodeList.sort(
        (
          {
            node: {
              props: { value: val1 },
            },
          },
          {
            node: {
              props: { value: val2 },
            },
          },
        ) => {
          const index1 = checkedValues.indexOf(val1);
          const index2 = checkedValues.indexOf(val2);
          return index1 - index2;
        },
      );
    }
  }

  Object.defineProperty(extra, 'triggerNode', {
    get() {
      warning(false, '\`triggerNode\` is deprecated. Please consider decoupling data with node.');
      generateMap();

      return triggerNode;
    },
  });
  Object.defineProperty(extra, 'allCheckedNodes', {
    get() {
      warning(false, '\`allCheckedNodes\` is deprecated. Please consider decoupling data with node.');
      generateMap();

      if (showPosition) {
        return nodeList;
      }

      return nodeList.map(({ node }) => node);
    },
  });
}
`},76742:function(v,t){t.Z=`import type { DataEntity } from 'rc-tree/lib/interface';
import type { SafeKey, FieldNames } from '../interface';
import { isCheckDisabled } from './valueUtil';

export const SHOW_ALL = 'SHOW_ALL';
export const SHOW_PARENT = 'SHOW_PARENT';
export const SHOW_CHILD = 'SHOW_CHILD';

export type CheckedStrategy = typeof SHOW_ALL | typeof SHOW_PARENT | typeof SHOW_CHILD;

export function formatStrategyValues(
  values: SafeKey[],
  strategy: CheckedStrategy,
  keyEntities: Record<SafeKey, DataEntity>,
  fieldNames: FieldNames,
): SafeKey[] {
  const valueSet = new Set(values);

  if (strategy === SHOW_CHILD) {
    return values.filter(key => {
      const entity = keyEntities[key];
      return (
        !entity ||
        !entity.children ||
        !entity.children.some(({ node }) => valueSet.has(node[fieldNames.value])) ||
        !entity.children.every(
          ({ node }) => isCheckDisabled(node) || valueSet.has(node[fieldNames.value]),
        )
      );
    });
  }
  if (strategy === SHOW_PARENT) {
    return values.filter(key => {
      const entity = keyEntities[key];
      const parent = entity ? entity.parent : null;
      return !parent || isCheckDisabled(parent.node) || !valueSet.has(parent.key as SafeKey);
    });
  }
  return values;
}
`},4370:function(v,t){t.Z=`import type { DataNode, FieldNames, SafeKey } from '../interface';

export const toArray = <T>(value: T | T[]): T[] =>
  Array.isArray(value) ? value : value !== undefined ? [value] : [];

export const fillFieldNames = (fieldNames?: FieldNames) => {
  const { label, value, children } = fieldNames || {};
  return {
    _title: label ? [label] : ['title', 'label'],
    value: value || 'value',
    key: value || 'value',
    children: children || 'children',
  };
};

export const isCheckDisabled = (node: DataNode): boolean =>
  !node || node.disabled || node.disableCheckbox || node.checkable === false;

export const getAllKeys = (treeData: DataNode[], fieldNames: FieldNames): SafeKey[] => {
  const keys: SafeKey[] = [];

  const dig = (list: DataNode[]): void => {
    list.forEach(item => {
      const children = item[fieldNames.children];
      if (children) {
        keys.push(item[fieldNames.value]);
        dig(children);
      }
    });
  };

  dig(treeData);

  return keys;
};

export const isNil = (val: any): boolean => val === null || val === undefined;
`},29162:function(v,t){t.Z=`import warning from '@rc-component/util/lib/warning';
import type { TreeSelectProps } from '../TreeSelect';
import { toArray } from './valueUtil';

function warningProps(props: TreeSelectProps & { searchPlaceholder?: string }) {
  const {
    searchPlaceholder,
    treeCheckStrictly,
    treeCheckable,
    labelInValue,
    value,
    multiple,
    showCheckedStrategy,
    maxCount,
  } = props;

  warning(!searchPlaceholder, '\`searchPlaceholder\` has been removed.');

  if (treeCheckStrictly && labelInValue === false) {
    warning(false, '\`treeCheckStrictly\` will force set \`labelInValue\` to \`true\`.');
  }

  if (labelInValue || treeCheckStrictly) {
    warning(
      toArray(value).every(val => val && typeof val === 'object' && 'value' in val),
      'Invalid prop \`value\` supplied to \`TreeSelect\`. You should use { label: string, value: string | number } or [{ label: string, value: string | number }] instead.',
    );
  }

  if (treeCheckStrictly || multiple || treeCheckable) {
    warning(
      !value || Array.isArray(value),
      '\`value\` should be an array when \`TreeSelect\` is checkable or multiple.',
    );
  } else {
    warning(!Array.isArray(value), '\`value\` should not be array when \`TreeSelect\` is single mode.');
  }

  if (
    maxCount &&
    ((showCheckedStrategy === 'SHOW_ALL' && !treeCheckStrictly) ||
      showCheckedStrategy === 'SHOW_PARENT')
  ) {
    warning(
      false,
      '\`maxCount\` not work with \`showCheckedStrategy=SHOW_ALL\` (when \`treeCheckStrictly=false\`) or \`showCheckedStrategy=SHOW_PARENT\`.',
    );
  }
}

export default warningProps;
`}}]);
